/* Generated by Cython 0.25.2 */

/* BEGIN: Cython Metadata
{
    "distutils": {},
    "module_name": "svelter_sv.readme"
}
END: Cython Metadata */

#define PY_SSIZE_T_CLEAN
#include "Python.h"
#ifndef Py_PYTHON_H
    #error Python headers needed to compile C extensions, please install development version of Python.
#elif PY_VERSION_HEX < 0x02060000 || (0x03000000 <= PY_VERSION_HEX && PY_VERSION_HEX < 0x03020000)
    #error Cython requires Python 2.6+ or Python 3.2+.
#else
#define CYTHON_ABI "0_25_2"
#include <stddef.h>
#ifndef offsetof
  #define offsetof(type, member) ( (size_t) & ((type*)0) -> member )
#endif
#if !defined(WIN32) && !defined(MS_WINDOWS)
  #ifndef __stdcall
    #define __stdcall
  #endif
  #ifndef __cdecl
    #define __cdecl
  #endif
  #ifndef __fastcall
    #define __fastcall
  #endif
#endif
#ifndef DL_IMPORT
  #define DL_IMPORT(t) t
#endif
#ifndef DL_EXPORT
  #define DL_EXPORT(t) t
#endif
#ifndef HAVE_LONG_LONG
  #if PY_VERSION_HEX >= 0x03030000 || (PY_MAJOR_VERSION == 2 && PY_VERSION_HEX >= 0x02070000)
    #define HAVE_LONG_LONG
  #endif
#endif
#ifndef PY_LONG_LONG
  #define PY_LONG_LONG LONG_LONG
#endif
#ifndef Py_HUGE_VAL
  #define Py_HUGE_VAL HUGE_VAL
#endif
#ifdef PYPY_VERSION
  #define CYTHON_COMPILING_IN_PYPY 1
  #define CYTHON_COMPILING_IN_PYSTON 0
  #define CYTHON_COMPILING_IN_CPYTHON 0
  #undef CYTHON_USE_TYPE_SLOTS
  #define CYTHON_USE_TYPE_SLOTS 0
  #undef CYTHON_USE_ASYNC_SLOTS
  #define CYTHON_USE_ASYNC_SLOTS 0
  #undef CYTHON_USE_PYLIST_INTERNALS
  #define CYTHON_USE_PYLIST_INTERNALS 0
  #undef CYTHON_USE_UNICODE_INTERNALS
  #define CYTHON_USE_UNICODE_INTERNALS 0
  #undef CYTHON_USE_UNICODE_WRITER
  #define CYTHON_USE_UNICODE_WRITER 0
  #undef CYTHON_USE_PYLONG_INTERNALS
  #define CYTHON_USE_PYLONG_INTERNALS 0
  #undef CYTHON_AVOID_BORROWED_REFS
  #define CYTHON_AVOID_BORROWED_REFS 1
  #undef CYTHON_ASSUME_SAFE_MACROS
  #define CYTHON_ASSUME_SAFE_MACROS 0
  #undef CYTHON_UNPACK_METHODS
  #define CYTHON_UNPACK_METHODS 0
  #undef CYTHON_FAST_THREAD_STATE
  #define CYTHON_FAST_THREAD_STATE 0
  #undef CYTHON_FAST_PYCALL
  #define CYTHON_FAST_PYCALL 0
#elif defined(PYSTON_VERSION)
  #define CYTHON_COMPILING_IN_PYPY 0
  #define CYTHON_COMPILING_IN_PYSTON 1
  #define CYTHON_COMPILING_IN_CPYTHON 0
  #ifndef CYTHON_USE_TYPE_SLOTS
    #define CYTHON_USE_TYPE_SLOTS 1
  #endif
  #undef CYTHON_USE_ASYNC_SLOTS
  #define CYTHON_USE_ASYNC_SLOTS 0
  #undef CYTHON_USE_PYLIST_INTERNALS
  #define CYTHON_USE_PYLIST_INTERNALS 0
  #ifndef CYTHON_USE_UNICODE_INTERNALS
    #define CYTHON_USE_UNICODE_INTERNALS 1
  #endif
  #undef CYTHON_USE_UNICODE_WRITER
  #define CYTHON_USE_UNICODE_WRITER 0
  #undef CYTHON_USE_PYLONG_INTERNALS
  #define CYTHON_USE_PYLONG_INTERNALS 0
  #ifndef CYTHON_AVOID_BORROWED_REFS
    #define CYTHON_AVOID_BORROWED_REFS 0
  #endif
  #ifndef CYTHON_ASSUME_SAFE_MACROS
    #define CYTHON_ASSUME_SAFE_MACROS 1
  #endif
  #ifndef CYTHON_UNPACK_METHODS
    #define CYTHON_UNPACK_METHODS 1
  #endif
  #undef CYTHON_FAST_THREAD_STATE
  #define CYTHON_FAST_THREAD_STATE 0
  #undef CYTHON_FAST_PYCALL
  #define CYTHON_FAST_PYCALL 0
#else
  #define CYTHON_COMPILING_IN_PYPY 0
  #define CYTHON_COMPILING_IN_PYSTON 0
  #define CYTHON_COMPILING_IN_CPYTHON 1
  #ifndef CYTHON_USE_TYPE_SLOTS
    #define CYTHON_USE_TYPE_SLOTS 1
  #endif
  #if PY_MAJOR_VERSION < 3
    #undef CYTHON_USE_ASYNC_SLOTS
    #define CYTHON_USE_ASYNC_SLOTS 0
  #elif !defined(CYTHON_USE_ASYNC_SLOTS)
    #define CYTHON_USE_ASYNC_SLOTS 1
  #endif
  #if PY_VERSION_HEX < 0x02070000
    #undef CYTHON_USE_PYLONG_INTERNALS
    #define CYTHON_USE_PYLONG_INTERNALS 0
  #elif !defined(CYTHON_USE_PYLONG_INTERNALS)
    #define CYTHON_USE_PYLONG_INTERNALS 1
  #endif
  #ifndef CYTHON_USE_PYLIST_INTERNALS
    #define CYTHON_USE_PYLIST_INTERNALS 1
  #endif
  #ifndef CYTHON_USE_UNICODE_INTERNALS
    #define CYTHON_USE_UNICODE_INTERNALS 1
  #endif
  #if PY_VERSION_HEX < 0x030300F0
    #undef CYTHON_USE_UNICODE_WRITER
    #define CYTHON_USE_UNICODE_WRITER 0
  #elif !defined(CYTHON_USE_UNICODE_WRITER)
    #define CYTHON_USE_UNICODE_WRITER 1
  #endif
  #ifndef CYTHON_AVOID_BORROWED_REFS
    #define CYTHON_AVOID_BORROWED_REFS 0
  #endif
  #ifndef CYTHON_ASSUME_SAFE_MACROS
    #define CYTHON_ASSUME_SAFE_MACROS 1
  #endif
  #ifndef CYTHON_UNPACK_METHODS
    #define CYTHON_UNPACK_METHODS 1
  #endif
  #ifndef CYTHON_FAST_THREAD_STATE
    #define CYTHON_FAST_THREAD_STATE 1
  #endif
  #ifndef CYTHON_FAST_PYCALL
    #define CYTHON_FAST_PYCALL 1
  #endif
#endif
#if !defined(CYTHON_FAST_PYCCALL)
#define CYTHON_FAST_PYCCALL  (CYTHON_FAST_PYCALL && PY_VERSION_HEX >= 0x030600B1)
#endif
#if CYTHON_USE_PYLONG_INTERNALS
  #include "longintrepr.h"
  #undef SHIFT
  #undef BASE
  #undef MASK
#endif
#if CYTHON_COMPILING_IN_PYPY && PY_VERSION_HEX < 0x02070600 && !defined(Py_OptimizeFlag)
  #define Py_OptimizeFlag 0
#endif
#define __PYX_BUILD_PY_SSIZE_T "n"
#define CYTHON_FORMAT_SSIZE_T "z"
#if PY_MAJOR_VERSION < 3
  #define __Pyx_BUILTIN_MODULE_NAME "__builtin__"
  #define __Pyx_PyCode_New(a, k, l, s, f, code, c, n, v, fv, cell, fn, name, fline, lnos)\
          PyCode_New(a+k, l, s, f, code, c, n, v, fv, cell, fn, name, fline, lnos)
  #define __Pyx_DefaultClassType PyClass_Type
#else
  #define __Pyx_BUILTIN_MODULE_NAME "builtins"
  #define __Pyx_PyCode_New(a, k, l, s, f, code, c, n, v, fv, cell, fn, name, fline, lnos)\
          PyCode_New(a, k, l, s, f, code, c, n, v, fv, cell, fn, name, fline, lnos)
  #define __Pyx_DefaultClassType PyType_Type
#endif
#ifndef Py_TPFLAGS_CHECKTYPES
  #define Py_TPFLAGS_CHECKTYPES 0
#endif
#ifndef Py_TPFLAGS_HAVE_INDEX
  #define Py_TPFLAGS_HAVE_INDEX 0
#endif
#ifndef Py_TPFLAGS_HAVE_NEWBUFFER
  #define Py_TPFLAGS_HAVE_NEWBUFFER 0
#endif
#ifndef Py_TPFLAGS_HAVE_FINALIZE
  #define Py_TPFLAGS_HAVE_FINALIZE 0
#endif
#ifndef METH_FASTCALL
  #define METH_FASTCALL 0x80
  typedef PyObject *(*__Pyx_PyCFunctionFast) (PyObject *self, PyObject **args,
                                              Py_ssize_t nargs, PyObject *kwnames);
#else
  #define __Pyx_PyCFunctionFast _PyCFunctionFast
#endif
#if CYTHON_FAST_PYCCALL
#define __Pyx_PyFastCFunction_Check(func)\
    ((PyCFunction_Check(func) && (METH_FASTCALL == (PyCFunction_GET_FLAGS(func) & ~(METH_CLASS | METH_STATIC | METH_COEXIST)))))
#else
#define __Pyx_PyFastCFunction_Check(func) 0
#endif
#if PY_VERSION_HEX > 0x03030000 && defined(PyUnicode_KIND)
  #define CYTHON_PEP393_ENABLED 1
  #define __Pyx_PyUnicode_READY(op)       (likely(PyUnicode_IS_READY(op)) ?\
                                              0 : _PyUnicode_Ready((PyObject *)(op)))
  #define __Pyx_PyUnicode_GET_LENGTH(u)   PyUnicode_GET_LENGTH(u)
  #define __Pyx_PyUnicode_READ_CHAR(u, i) PyUnicode_READ_CHAR(u, i)
  #define __Pyx_PyUnicode_MAX_CHAR_VALUE(u)   PyUnicode_MAX_CHAR_VALUE(u)
  #define __Pyx_PyUnicode_KIND(u)         PyUnicode_KIND(u)
  #define __Pyx_PyUnicode_DATA(u)         PyUnicode_DATA(u)
  #define __Pyx_PyUnicode_READ(k, d, i)   PyUnicode_READ(k, d, i)
  #define __Pyx_PyUnicode_WRITE(k, d, i, ch)  PyUnicode_WRITE(k, d, i, ch)
  #define __Pyx_PyUnicode_IS_TRUE(u)      (0 != (likely(PyUnicode_IS_READY(u)) ? PyUnicode_GET_LENGTH(u) : PyUnicode_GET_SIZE(u)))
#else
  #define CYTHON_PEP393_ENABLED 0
  #define PyUnicode_1BYTE_KIND  1
  #define PyUnicode_2BYTE_KIND  2
  #define PyUnicode_4BYTE_KIND  4
  #define __Pyx_PyUnicode_READY(op)       (0)
  #define __Pyx_PyUnicode_GET_LENGTH(u)   PyUnicode_GET_SIZE(u)
  #define __Pyx_PyUnicode_READ_CHAR(u, i) ((Py_UCS4)(PyUnicode_AS_UNICODE(u)[i]))
  #define __Pyx_PyUnicode_MAX_CHAR_VALUE(u)   ((sizeof(Py_UNICODE) == 2) ? 65535 : 1114111)
  #define __Pyx_PyUnicode_KIND(u)         (sizeof(Py_UNICODE))
  #define __Pyx_PyUnicode_DATA(u)         ((void*)PyUnicode_AS_UNICODE(u))
  #define __Pyx_PyUnicode_READ(k, d, i)   ((void)(k), (Py_UCS4)(((Py_UNICODE*)d)[i]))
  #define __Pyx_PyUnicode_WRITE(k, d, i, ch)  (((void)(k)), ((Py_UNICODE*)d)[i] = ch)
  #define __Pyx_PyUnicode_IS_TRUE(u)      (0 != PyUnicode_GET_SIZE(u))
#endif
#if CYTHON_COMPILING_IN_PYPY
  #define __Pyx_PyUnicode_Concat(a, b)      PyNumber_Add(a, b)
  #define __Pyx_PyUnicode_ConcatSafe(a, b)  PyNumber_Add(a, b)
#else
  #define __Pyx_PyUnicode_Concat(a, b)      PyUnicode_Concat(a, b)
  #define __Pyx_PyUnicode_ConcatSafe(a, b)  ((unlikely((a) == Py_None) || unlikely((b) == Py_None)) ?\
      PyNumber_Add(a, b) : __Pyx_PyUnicode_Concat(a, b))
#endif
#if CYTHON_COMPILING_IN_PYPY && !defined(PyUnicode_Contains)
  #define PyUnicode_Contains(u, s)  PySequence_Contains(u, s)
#endif
#if CYTHON_COMPILING_IN_PYPY && !defined(PyByteArray_Check)
  #define PyByteArray_Check(obj)  PyObject_TypeCheck(obj, &PyByteArray_Type)
#endif
#if CYTHON_COMPILING_IN_PYPY && !defined(PyObject_Format)
  #define PyObject_Format(obj, fmt)  PyObject_CallMethod(obj, "__format__", "O", fmt)
#endif
#if CYTHON_COMPILING_IN_PYPY && !defined(PyObject_Malloc)
  #define PyObject_Malloc(s)   PyMem_Malloc(s)
  #define PyObject_Free(p)     PyMem_Free(p)
  #define PyObject_Realloc(p)  PyMem_Realloc(p)
#endif
#if CYTHON_COMPILING_IN_PYSTON
  #define __Pyx_PyCode_HasFreeVars(co)  PyCode_HasFreeVars(co)
  #define __Pyx_PyFrame_SetLineNumber(frame, lineno) PyFrame_SetLineNumber(frame, lineno)
#else
  #define __Pyx_PyCode_HasFreeVars(co)  (PyCode_GetNumFree(co) > 0)
  #define __Pyx_PyFrame_SetLineNumber(frame, lineno)  (frame)->f_lineno = (lineno)
#endif
#define __Pyx_PyString_FormatSafe(a, b)   ((unlikely((a) == Py_None)) ? PyNumber_Remainder(a, b) : __Pyx_PyString_Format(a, b))
#define __Pyx_PyUnicode_FormatSafe(a, b)  ((unlikely((a) == Py_None)) ? PyNumber_Remainder(a, b) : PyUnicode_Format(a, b))
#if PY_MAJOR_VERSION >= 3
  #define __Pyx_PyString_Format(a, b)  PyUnicode_Format(a, b)
#else
  #define __Pyx_PyString_Format(a, b)  PyString_Format(a, b)
#endif
#if PY_MAJOR_VERSION < 3 && !defined(PyObject_ASCII)
  #define PyObject_ASCII(o)            PyObject_Repr(o)
#endif
#if PY_MAJOR_VERSION >= 3
  #define PyBaseString_Type            PyUnicode_Type
  #define PyStringObject               PyUnicodeObject
  #define PyString_Type                PyUnicode_Type
  #define PyString_Check               PyUnicode_Check
  #define PyString_CheckExact          PyUnicode_CheckExact
#endif
#if PY_MAJOR_VERSION >= 3
  #define __Pyx_PyBaseString_Check(obj) PyUnicode_Check(obj)
  #define __Pyx_PyBaseString_CheckExact(obj) PyUnicode_CheckExact(obj)
#else
  #define __Pyx_PyBaseString_Check(obj) (PyString_Check(obj) || PyUnicode_Check(obj))
  #define __Pyx_PyBaseString_CheckExact(obj) (PyString_CheckExact(obj) || PyUnicode_CheckExact(obj))
#endif
#ifndef PySet_CheckExact
  #define PySet_CheckExact(obj)        (Py_TYPE(obj) == &PySet_Type)
#endif
#define __Pyx_TypeCheck(obj, type) PyObject_TypeCheck(obj, (PyTypeObject *)type)
#define __Pyx_PyException_Check(obj) __Pyx_TypeCheck(obj, PyExc_Exception)
#if PY_MAJOR_VERSION >= 3
  #define PyIntObject                  PyLongObject
  #define PyInt_Type                   PyLong_Type
  #define PyInt_Check(op)              PyLong_Check(op)
  #define PyInt_CheckExact(op)         PyLong_CheckExact(op)
  #define PyInt_FromString             PyLong_FromString
  #define PyInt_FromUnicode            PyLong_FromUnicode
  #define PyInt_FromLong               PyLong_FromLong
  #define PyInt_FromSize_t             PyLong_FromSize_t
  #define PyInt_FromSsize_t            PyLong_FromSsize_t
  #define PyInt_AsLong                 PyLong_AsLong
  #define PyInt_AS_LONG                PyLong_AS_LONG
  #define PyInt_AsSsize_t              PyLong_AsSsize_t
  #define PyInt_AsUnsignedLongMask     PyLong_AsUnsignedLongMask
  #define PyInt_AsUnsignedLongLongMask PyLong_AsUnsignedLongLongMask
  #define PyNumber_Int                 PyNumber_Long
#endif
#if PY_MAJOR_VERSION >= 3
  #define PyBoolObject                 PyLongObject
#endif
#if PY_MAJOR_VERSION >= 3 && CYTHON_COMPILING_IN_PYPY
  #ifndef PyUnicode_InternFromString
    #define PyUnicode_InternFromString(s) PyUnicode_FromString(s)
  #endif
#endif
#if PY_VERSION_HEX < 0x030200A4
  typedef long Py_hash_t;
  #define __Pyx_PyInt_FromHash_t PyInt_FromLong
  #define __Pyx_PyInt_AsHash_t   PyInt_AsLong
#else
  #define __Pyx_PyInt_FromHash_t PyInt_FromSsize_t
  #define __Pyx_PyInt_AsHash_t   PyInt_AsSsize_t
#endif
#if PY_MAJOR_VERSION >= 3
  #define __Pyx_PyMethod_New(func, self, klass) ((self) ? PyMethod_New(func, self) : PyInstanceMethod_New(func))
#else
  #define __Pyx_PyMethod_New(func, self, klass) PyMethod_New(func, self, klass)
#endif
#if CYTHON_USE_ASYNC_SLOTS
  #if PY_VERSION_HEX >= 0x030500B1
    #define __Pyx_PyAsyncMethodsStruct PyAsyncMethods
    #define __Pyx_PyType_AsAsync(obj) (Py_TYPE(obj)->tp_as_async)
  #else
    typedef struct {
        unaryfunc am_await;
        unaryfunc am_aiter;
        unaryfunc am_anext;
    } __Pyx_PyAsyncMethodsStruct;
    #define __Pyx_PyType_AsAsync(obj) ((__Pyx_PyAsyncMethodsStruct*) (Py_TYPE(obj)->tp_reserved))
  #endif
#else
  #define __Pyx_PyType_AsAsync(obj) NULL
#endif
#ifndef CYTHON_RESTRICT
  #if defined(__GNUC__)
    #define CYTHON_RESTRICT __restrict__
  #elif defined(_MSC_VER) && _MSC_VER >= 1400
    #define CYTHON_RESTRICT __restrict
  #elif defined (__STDC_VERSION__) && __STDC_VERSION__ >= 199901L
    #define CYTHON_RESTRICT restrict
  #else
    #define CYTHON_RESTRICT
  #endif
#endif
#ifndef CYTHON_UNUSED
# if defined(__GNUC__)
#   if !(defined(__cplusplus)) || (__GNUC__ > 3 || (__GNUC__ == 3 && __GNUC_MINOR__ >= 4))
#     define CYTHON_UNUSED __attribute__ ((__unused__))
#   else
#     define CYTHON_UNUSED
#   endif
# elif defined(__ICC) || (defined(__INTEL_COMPILER) && !defined(_MSC_VER))
#   define CYTHON_UNUSED __attribute__ ((__unused__))
# else
#   define CYTHON_UNUSED
# endif
#endif
#ifndef CYTHON_MAYBE_UNUSED_VAR
#  if defined(__cplusplus)
     template<class T> void CYTHON_MAYBE_UNUSED_VAR( const T& ) { }
#  else
#    define CYTHON_MAYBE_UNUSED_VAR(x) (void)(x)
#  endif
#endif
#ifndef CYTHON_NCP_UNUSED
# if CYTHON_COMPILING_IN_CPYTHON
#  define CYTHON_NCP_UNUSED
# else
#  define CYTHON_NCP_UNUSED CYTHON_UNUSED
# endif
#endif
#define __Pyx_void_to_None(void_result) ((void)(void_result), Py_INCREF(Py_None), Py_None)

#ifndef CYTHON_INLINE
  #if defined(__clang__)
    #define CYTHON_INLINE __inline__ __attribute__ ((__unused__))
  #elif defined(__GNUC__)
    #define CYTHON_INLINE __inline__
  #elif defined(_MSC_VER)
    #define CYTHON_INLINE __inline
  #elif defined (__STDC_VERSION__) && __STDC_VERSION__ >= 199901L
    #define CYTHON_INLINE inline
  #else
    #define CYTHON_INLINE
  #endif
#endif

#if defined(WIN32) || defined(MS_WINDOWS)
  #define _USE_MATH_DEFINES
#endif
#include <math.h>
#ifdef NAN
#define __PYX_NAN() ((float) NAN)
#else
static CYTHON_INLINE float __PYX_NAN() {
  float value;
  memset(&value, 0xFF, sizeof(value));
  return value;
}
#endif
#if defined(__CYGWIN__) && defined(_LDBL_EQ_DBL)
#define __Pyx_truncl trunc
#else
#define __Pyx_truncl truncl
#endif


#define __PYX_ERR(f_index, lineno, Ln_error) \
{ \
  __pyx_filename = __pyx_f[f_index]; __pyx_lineno = lineno; __pyx_clineno = __LINE__; goto Ln_error; \
}

#if PY_MAJOR_VERSION >= 3
  #define __Pyx_PyNumber_Divide(x,y)         PyNumber_TrueDivide(x,y)
  #define __Pyx_PyNumber_InPlaceDivide(x,y)  PyNumber_InPlaceTrueDivide(x,y)
#else
  #define __Pyx_PyNumber_Divide(x,y)         PyNumber_Divide(x,y)
  #define __Pyx_PyNumber_InPlaceDivide(x,y)  PyNumber_InPlaceDivide(x,y)
#endif

#ifndef __PYX_EXTERN_C
  #ifdef __cplusplus
    #define __PYX_EXTERN_C extern "C"
  #else
    #define __PYX_EXTERN_C extern
  #endif
#endif

#define __PYX_HAVE__svelter_sv__readme
#define __PYX_HAVE_API__svelter_sv__readme
#ifdef _OPENMP
#include <omp.h>
#endif /* _OPENMP */

#ifdef PYREX_WITHOUT_ASSERTIONS
#define CYTHON_WITHOUT_ASSERTIONS
#endif

typedef struct {PyObject **p; const char *s; const Py_ssize_t n; const char* encoding;
                const char is_unicode; const char is_str; const char intern; } __Pyx_StringTabEntry;

#define __PYX_DEFAULT_STRING_ENCODING_IS_ASCII 0
#define __PYX_DEFAULT_STRING_ENCODING_IS_DEFAULT 0
#define __PYX_DEFAULT_STRING_ENCODING ""
#define __Pyx_PyObject_FromString __Pyx_PyBytes_FromString
#define __Pyx_PyObject_FromStringAndSize __Pyx_PyBytes_FromStringAndSize
#define __Pyx_uchar_cast(c) ((unsigned char)c)
#define __Pyx_long_cast(x) ((long)x)
#define __Pyx_fits_Py_ssize_t(v, type, is_signed)  (\
    (sizeof(type) < sizeof(Py_ssize_t))  ||\
    (sizeof(type) > sizeof(Py_ssize_t) &&\
          likely(v < (type)PY_SSIZE_T_MAX ||\
                 v == (type)PY_SSIZE_T_MAX)  &&\
          (!is_signed || likely(v > (type)PY_SSIZE_T_MIN ||\
                                v == (type)PY_SSIZE_T_MIN)))  ||\
    (sizeof(type) == sizeof(Py_ssize_t) &&\
          (is_signed || likely(v < (type)PY_SSIZE_T_MAX ||\
                               v == (type)PY_SSIZE_T_MAX)))  )
#if defined (__cplusplus) && __cplusplus >= 201103L
    #include <cstdlib>
    #define __Pyx_sst_abs(value) std::abs(value)
#elif SIZEOF_INT >= SIZEOF_SIZE_T
    #define __Pyx_sst_abs(value) abs(value)
#elif SIZEOF_LONG >= SIZEOF_SIZE_T
    #define __Pyx_sst_abs(value) labs(value)
#elif defined (_MSC_VER) && defined (_M_X64)
    #define __Pyx_sst_abs(value) _abs64(value)
#elif defined (__STDC_VERSION__) && __STDC_VERSION__ >= 199901L
    #define __Pyx_sst_abs(value) llabs(value)
#elif defined (__GNUC__)
    #define __Pyx_sst_abs(value) __builtin_llabs(value)
#else
    #define __Pyx_sst_abs(value) ((value<0) ? -value : value)
#endif
static CYTHON_INLINE char* __Pyx_PyObject_AsString(PyObject*);
static CYTHON_INLINE char* __Pyx_PyObject_AsStringAndSize(PyObject*, Py_ssize_t* length);
#define __Pyx_PyByteArray_FromString(s) PyByteArray_FromStringAndSize((const char*)s, strlen((const char*)s))
#define __Pyx_PyByteArray_FromStringAndSize(s, l) PyByteArray_FromStringAndSize((const char*)s, l)
#define __Pyx_PyBytes_FromString        PyBytes_FromString
#define __Pyx_PyBytes_FromStringAndSize PyBytes_FromStringAndSize
static CYTHON_INLINE PyObject* __Pyx_PyUnicode_FromString(const char*);
#if PY_MAJOR_VERSION < 3
    #define __Pyx_PyStr_FromString        __Pyx_PyBytes_FromString
    #define __Pyx_PyStr_FromStringAndSize __Pyx_PyBytes_FromStringAndSize
#else
    #define __Pyx_PyStr_FromString        __Pyx_PyUnicode_FromString
    #define __Pyx_PyStr_FromStringAndSize __Pyx_PyUnicode_FromStringAndSize
#endif
#define __Pyx_PyObject_AsSString(s)    ((signed char*) __Pyx_PyObject_AsString(s))
#define __Pyx_PyObject_AsUString(s)    ((unsigned char*) __Pyx_PyObject_AsString(s))
#define __Pyx_PyObject_FromCString(s)  __Pyx_PyObject_FromString((const char*)s)
#define __Pyx_PyBytes_FromCString(s)   __Pyx_PyBytes_FromString((const char*)s)
#define __Pyx_PyByteArray_FromCString(s)   __Pyx_PyByteArray_FromString((const char*)s)
#define __Pyx_PyStr_FromCString(s)     __Pyx_PyStr_FromString((const char*)s)
#define __Pyx_PyUnicode_FromCString(s) __Pyx_PyUnicode_FromString((const char*)s)
#if PY_MAJOR_VERSION < 3
static CYTHON_INLINE size_t __Pyx_Py_UNICODE_strlen(const Py_UNICODE *u)
{
    const Py_UNICODE *u_end = u;
    while (*u_end++) ;
    return (size_t)(u_end - u - 1);
}
#else
#define __Pyx_Py_UNICODE_strlen Py_UNICODE_strlen
#endif
#define __Pyx_PyUnicode_FromUnicode(u)       PyUnicode_FromUnicode(u, __Pyx_Py_UNICODE_strlen(u))
#define __Pyx_PyUnicode_FromUnicodeAndLength PyUnicode_FromUnicode
#define __Pyx_PyUnicode_AsUnicode            PyUnicode_AsUnicode
#define __Pyx_NewRef(obj) (Py_INCREF(obj), obj)
#define __Pyx_Owned_Py_None(b) __Pyx_NewRef(Py_None)
#define __Pyx_PyBool_FromLong(b) ((b) ? __Pyx_NewRef(Py_True) : __Pyx_NewRef(Py_False))
static CYTHON_INLINE int __Pyx_PyObject_IsTrue(PyObject*);
static CYTHON_INLINE PyObject* __Pyx_PyNumber_IntOrLong(PyObject* x);
static CYTHON_INLINE Py_ssize_t __Pyx_PyIndex_AsSsize_t(PyObject*);
static CYTHON_INLINE PyObject * __Pyx_PyInt_FromSize_t(size_t);
#if CYTHON_ASSUME_SAFE_MACROS
#define __pyx_PyFloat_AsDouble(x) (PyFloat_CheckExact(x) ? PyFloat_AS_DOUBLE(x) : PyFloat_AsDouble(x))
#else
#define __pyx_PyFloat_AsDouble(x) PyFloat_AsDouble(x)
#endif
#define __pyx_PyFloat_AsFloat(x) ((float) __pyx_PyFloat_AsDouble(x))
#if PY_MAJOR_VERSION >= 3
#define __Pyx_PyNumber_Int(x) (PyLong_CheckExact(x) ? __Pyx_NewRef(x) : PyNumber_Long(x))
#else
#define __Pyx_PyNumber_Int(x) (PyInt_CheckExact(x) ? __Pyx_NewRef(x) : PyNumber_Int(x))
#endif
#define __Pyx_PyNumber_Float(x) (PyFloat_CheckExact(x) ? __Pyx_NewRef(x) : PyNumber_Float(x))
#if PY_MAJOR_VERSION < 3 && __PYX_DEFAULT_STRING_ENCODING_IS_ASCII
static int __Pyx_sys_getdefaultencoding_not_ascii;
static int __Pyx_init_sys_getdefaultencoding_params(void) {
    PyObject* sys;
    PyObject* default_encoding = NULL;
    PyObject* ascii_chars_u = NULL;
    PyObject* ascii_chars_b = NULL;
    const char* default_encoding_c;
    sys = PyImport_ImportModule("sys");
    if (!sys) goto bad;
    default_encoding = PyObject_CallMethod(sys, (char*) "getdefaultencoding", NULL);
    Py_DECREF(sys);
    if (!default_encoding) goto bad;
    default_encoding_c = PyBytes_AsString(default_encoding);
    if (!default_encoding_c) goto bad;
    if (strcmp(default_encoding_c, "ascii") == 0) {
        __Pyx_sys_getdefaultencoding_not_ascii = 0;
    } else {
        char ascii_chars[128];
        int c;
        for (c = 0; c < 128; c++) {
            ascii_chars[c] = c;
        }
        __Pyx_sys_getdefaultencoding_not_ascii = 1;
        ascii_chars_u = PyUnicode_DecodeASCII(ascii_chars, 128, NULL);
        if (!ascii_chars_u) goto bad;
        ascii_chars_b = PyUnicode_AsEncodedString(ascii_chars_u, default_encoding_c, NULL);
        if (!ascii_chars_b || !PyBytes_Check(ascii_chars_b) || memcmp(ascii_chars, PyBytes_AS_STRING(ascii_chars_b), 128) != 0) {
            PyErr_Format(
                PyExc_ValueError,
                "This module compiled with c_string_encoding=ascii, but default encoding '%.200s' is not a superset of ascii.",
                default_encoding_c);
            goto bad;
        }
        Py_DECREF(ascii_chars_u);
        Py_DECREF(ascii_chars_b);
    }
    Py_DECREF(default_encoding);
    return 0;
bad:
    Py_XDECREF(default_encoding);
    Py_XDECREF(ascii_chars_u);
    Py_XDECREF(ascii_chars_b);
    return -1;
}
#endif
#if __PYX_DEFAULT_STRING_ENCODING_IS_DEFAULT && PY_MAJOR_VERSION >= 3
#define __Pyx_PyUnicode_FromStringAndSize(c_str, size) PyUnicode_DecodeUTF8(c_str, size, NULL)
#else
#define __Pyx_PyUnicode_FromStringAndSize(c_str, size) PyUnicode_Decode(c_str, size, __PYX_DEFAULT_STRING_ENCODING, NULL)
#if __PYX_DEFAULT_STRING_ENCODING_IS_DEFAULT
static char* __PYX_DEFAULT_STRING_ENCODING;
static int __Pyx_init_sys_getdefaultencoding_params(void) {
    PyObject* sys;
    PyObject* default_encoding = NULL;
    char* default_encoding_c;
    sys = PyImport_ImportModule("sys");
    if (!sys) goto bad;
    default_encoding = PyObject_CallMethod(sys, (char*) (const char*) "getdefaultencoding", NULL);
    Py_DECREF(sys);
    if (!default_encoding) goto bad;
    default_encoding_c = PyBytes_AsString(default_encoding);
    if (!default_encoding_c) goto bad;
    __PYX_DEFAULT_STRING_ENCODING = (char*) malloc(strlen(default_encoding_c));
    if (!__PYX_DEFAULT_STRING_ENCODING) goto bad;
    strcpy(__PYX_DEFAULT_STRING_ENCODING, default_encoding_c);
    Py_DECREF(default_encoding);
    return 0;
bad:
    Py_XDECREF(default_encoding);
    return -1;
}
#endif
#endif


/* Test for GCC > 2.95 */
#if defined(__GNUC__)     && (__GNUC__ > 2 || (__GNUC__ == 2 && (__GNUC_MINOR__ > 95)))
  #define likely(x)   __builtin_expect(!!(x), 1)
  #define unlikely(x) __builtin_expect(!!(x), 0)
#else /* !__GNUC__ or GCC < 2.95 */
  #define likely(x)   (x)
  #define unlikely(x) (x)
#endif /* __GNUC__ */

static PyObject *__pyx_m;
static PyObject *__pyx_d;
static PyObject *__pyx_b;
static PyObject *__pyx_empty_tuple;
static PyObject *__pyx_empty_bytes;
static PyObject *__pyx_empty_unicode;
static int __pyx_lineno;
static int __pyx_clineno = 0;
static const char * __pyx_cfilenm= __FILE__;
static const char *__pyx_filename;


static const char *__pyx_f[] = {
  "svelter_sv/readme.pyx",
};

/*--- Type declarations ---*/

/* --- Runtime support code (head) --- */
/* Refnanny.proto */
#ifndef CYTHON_REFNANNY
  #define CYTHON_REFNANNY 0
#endif
#if CYTHON_REFNANNY
  typedef struct {
    void (*INCREF)(void*, PyObject*, int);
    void (*DECREF)(void*, PyObject*, int);
    void (*GOTREF)(void*, PyObject*, int);
    void (*GIVEREF)(void*, PyObject*, int);
    void* (*SetupContext)(const char*, int, const char*);
    void (*FinishContext)(void**);
  } __Pyx_RefNannyAPIStruct;
  static __Pyx_RefNannyAPIStruct *__Pyx_RefNanny = NULL;
  static __Pyx_RefNannyAPIStruct *__Pyx_RefNannyImportAPI(const char *modname);
  #define __Pyx_RefNannyDeclarations void *__pyx_refnanny = NULL;
#ifdef WITH_THREAD
  #define __Pyx_RefNannySetupContext(name, acquire_gil)\
          if (acquire_gil) {\
              PyGILState_STATE __pyx_gilstate_save = PyGILState_Ensure();\
              __pyx_refnanny = __Pyx_RefNanny->SetupContext((name), __LINE__, __FILE__);\
              PyGILState_Release(__pyx_gilstate_save);\
          } else {\
              __pyx_refnanny = __Pyx_RefNanny->SetupContext((name), __LINE__, __FILE__);\
          }
#else
  #define __Pyx_RefNannySetupContext(name, acquire_gil)\
          __pyx_refnanny = __Pyx_RefNanny->SetupContext((name), __LINE__, __FILE__)
#endif
  #define __Pyx_RefNannyFinishContext()\
          __Pyx_RefNanny->FinishContext(&__pyx_refnanny)
  #define __Pyx_INCREF(r)  __Pyx_RefNanny->INCREF(__pyx_refnanny, (PyObject *)(r), __LINE__)
  #define __Pyx_DECREF(r)  __Pyx_RefNanny->DECREF(__pyx_refnanny, (PyObject *)(r), __LINE__)
  #define __Pyx_GOTREF(r)  __Pyx_RefNanny->GOTREF(__pyx_refnanny, (PyObject *)(r), __LINE__)
  #define __Pyx_GIVEREF(r) __Pyx_RefNanny->GIVEREF(__pyx_refnanny, (PyObject *)(r), __LINE__)
  #define __Pyx_XINCREF(r)  do { if((r) != NULL) {__Pyx_INCREF(r); }} while(0)
  #define __Pyx_XDECREF(r)  do { if((r) != NULL) {__Pyx_DECREF(r); }} while(0)
  #define __Pyx_XGOTREF(r)  do { if((r) != NULL) {__Pyx_GOTREF(r); }} while(0)
  #define __Pyx_XGIVEREF(r) do { if((r) != NULL) {__Pyx_GIVEREF(r);}} while(0)
#else
  #define __Pyx_RefNannyDeclarations
  #define __Pyx_RefNannySetupContext(name, acquire_gil)
  #define __Pyx_RefNannyFinishContext()
  #define __Pyx_INCREF(r) Py_INCREF(r)
  #define __Pyx_DECREF(r) Py_DECREF(r)
  #define __Pyx_GOTREF(r)
  #define __Pyx_GIVEREF(r)
  #define __Pyx_XINCREF(r) Py_XINCREF(r)
  #define __Pyx_XDECREF(r) Py_XDECREF(r)
  #define __Pyx_XGOTREF(r)
  #define __Pyx_XGIVEREF(r)
#endif
#define __Pyx_XDECREF_SET(r, v) do {\
        PyObject *tmp = (PyObject *) r;\
        r = v; __Pyx_XDECREF(tmp);\
    } while (0)
#define __Pyx_DECREF_SET(r, v) do {\
        PyObject *tmp = (PyObject *) r;\
        r = v; __Pyx_DECREF(tmp);\
    } while (0)
#define __Pyx_CLEAR(r)    do { PyObject* tmp = ((PyObject*)(r)); r = NULL; __Pyx_DECREF(tmp);} while(0)
#define __Pyx_XCLEAR(r)   do { if((r) != NULL) {PyObject* tmp = ((PyObject*)(r)); r = NULL; __Pyx_DECREF(tmp);}} while(0)

/* PyObjectGetAttrStr.proto */
#if CYTHON_USE_TYPE_SLOTS
static CYTHON_INLINE PyObject* __Pyx_PyObject_GetAttrStr(PyObject* obj, PyObject* attr_name) {
    PyTypeObject* tp = Py_TYPE(obj);
    if (likely(tp->tp_getattro))
        return tp->tp_getattro(obj, attr_name);
#if PY_MAJOR_VERSION < 3
    if (likely(tp->tp_getattr))
        return tp->tp_getattr(obj, PyString_AS_STRING(attr_name));
#endif
    return PyObject_GetAttr(obj, attr_name);
}
#else
#define __Pyx_PyObject_GetAttrStr(o,n) PyObject_GetAttr(o,n)
#endif

/* GetBuiltinName.proto */
static PyObject *__Pyx_GetBuiltinName(PyObject *name);

/* PyObjectCall.proto */
#if CYTHON_COMPILING_IN_CPYTHON
static CYTHON_INLINE PyObject* __Pyx_PyObject_Call(PyObject *func, PyObject *arg, PyObject *kw);
#else
#define __Pyx_PyObject_Call(func, arg, kw) PyObject_Call(func, arg, kw)
#endif

/* CodeObjectCache.proto */
typedef struct {
    PyCodeObject* code_object;
    int code_line;
} __Pyx_CodeObjectCacheEntry;
struct __Pyx_CodeObjectCache {
    int count;
    int max_count;
    __Pyx_CodeObjectCacheEntry* entries;
};
static struct __Pyx_CodeObjectCache __pyx_code_cache = {0,0,NULL};
static int __pyx_bisect_code_objects(__Pyx_CodeObjectCacheEntry* entries, int count, int code_line);
static PyCodeObject *__pyx_find_code_object(int code_line);
static void __pyx_insert_code_object(int code_line, PyCodeObject* code_object);

/* AddTraceback.proto */
static void __Pyx_AddTraceback(const char *funcname, int c_line,
                               int py_line, const char *filename);

/* CIntToPy.proto */
static CYTHON_INLINE PyObject* __Pyx_PyInt_From_long(long value);

/* CIntFromPy.proto */
static CYTHON_INLINE long __Pyx_PyInt_As_long(PyObject *);

/* CIntFromPy.proto */
static CYTHON_INLINE int __Pyx_PyInt_As_int(PyObject *);

/* CheckBinaryVersion.proto */
static int __Pyx_check_binary_version(void);

/* InitStrings.proto */
static int __Pyx_InitStrings(__Pyx_StringTabEntry *t);


/* Module declarations from 'svelter_sv.readme' */
#define __Pyx_MODULE_NAME "svelter_sv.readme"
int __pyx_module_is_main_svelter_sv__readme = 0;

/* Implementation of 'svelter_sv.readme' */
static PyObject *__pyx_builtin_print;
static const char __pyx_k__2[] = "";
static const char __pyx_k__5[] = " ";
static const char __pyx_k_main[] = "__main__";
static const char __pyx_k_test[] = "__test__";
static const char __pyx_k_Clean[] = "    Clean";
static const char __pyx_k_Setup[] = "    Setup";
static const char __pyx_k_print[] = "print";
static const char __pyx_k_Options[] = "Options:";
static const char __pyx_k_BPSearch[] = "    BPSearch";
static const char __pyx_k_NullModel[] = "    NullModel";
static const char __pyx_k_SVPredict[] = "    SVPredict";
static const char __pyx_k_BPIntegrate[] = "    BPIntegrate";
static const char __pyx_k_SVIntegrate[] = "    SVIntegrate";
static const char __pyx_k_SVelter_Clean[] = "SVelter Clean";
static const char __pyx_k_SVelter_Setup[] = "SVelter Setup";
static const char __pyx_k_SVelter_BPSearch[] = "SVelter  BPSearch";
static const char __pyx_k_SVelter_NullModel[] = "SVelter NullModel";
static const char __pyx_k_SVelter_SVPredict[] = "SVelter  SVPredict";
static const char __pyx_k_svelter_sv_readme[] = "svelter_sv.readme";
static const char __pyx_k_Optional_Parameters[] = "Optional Parameters:";
static const char __pyx_k_Required_Parameters[] = "Required Parameters:";
static const char __pyx_k_SVelter_BPIntegrate[] = "SVelter  BPIntegrate";
static const char __pyx_k_SVelter_SVIntegrate[] = "SVelter SVIntegrate";
static const char __pyx_k_SVelter_PredefinedBP[] = "SVelter PredefinedBP";
static const char __pyx_k_print_default_parameters[] = "print_default_parameters";
static const char __pyx_k_bp_path_specify_the_path_where[] = "    --bp-path, specify the path where all called breakpoints were kept under";
static const char __pyx_k_null_random_length_specify_the[] = "    --null-random-length, specify the length of random regions if --copyneutral parameter not used [5000]";
static const char __pyx_k_print_default_parameters_clean[] = "print_default_parameters_clean";
static const char __pyx_k_print_default_parameters_setup[] = "print_default_parameters_setup";
static const char __pyx_k_sample_input_alignment_file_in[] = "    --sample, input alignment file in bam format";
static const char __pyx_k_batch_specify_number_of_structu[] = "    --batch, specify number of structures in each separate file (if 0, output files will be calssified by chromosomes; default, all BP clustered will be integrated in one txt file)";
static const char __pyx_k_bp_file_input_txt_file_containi[] = "    --bp-file, input txt file containing clustered bps.";
static const char __pyx_k_chromosome_name_of_chromosome_t[] = "    --chromosome, name of chromosome to run. should match chromosome name in bam file";
static const char __pyx_k_data_talkowski_xuefang_local_sr[] = "/data/talkowski/xuefang/local/src/svelter/svelter_sv/readme.pyx";
static const char __pyx_k_ftp_hgdownload_cse_ucsc_edu_gol[] = "                     ftp://hgdownload.cse.ucsc.edu/goldenPath/currentGenomes/Homo_sapiens/encodeDCC/wgEncodeMapability/";
static const char __pyx_k_http_hgdownload_cse_ucsc_edu_ad[] = "                     http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/bigWigSummary";
static const char __pyx_k_input_bed_predefined_breakpoint[] = "    --input-bed, predefined breakpoints in bed format";
static const char __pyx_k_input_path_path_of_coverage_fil[] = "    --input-path, path of .coverage files produced by SVelter SVPredict";
static const char __pyx_k_null_copyneutral_length_minimum[] = "    --null-copyneutral-length, minimum length requirement for --copyneutral regions used to build null model [2000]";
static const char __pyx_k_null_copyneutral_perc_percentag[] = "    --null-copyneutral-perc, percentage of regions from --copyneutral to utilize [0.1]";
static const char __pyx_k_null_model_specify_which_stat_m[] = "    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used";
static const char __pyx_k_null_random_num_specify_the_num[] = "    --null-random-num, specify the number of random regions if --copyneutral parameter not used [10000]";
static const char __pyx_k_num_iteration_maximum_number_of[] = "    --num-iteration, maximum number of iterations per structure will run";
static const char __pyx_k_ploidy_limit_algorithm_to_speci[] = "    --ploidy, limit algorithm to specific zygosity (0:heterozygous only; 1:homozygous only; 2:both; default:2)";
static const char __pyx_k_prefix_output_prefix_for_vcf_an[] = "    --prefix, output prefix for vcf and svelter files [input.vcf, input.svelter]";
static const char __pyx_k_qc_align_minimum_alignment_qual[] = "    --qc-align, minimum alignment quality required for mapped reads in bam file [20]";
static const char __pyx_k_qc_map_cutoff_the_minimum_mappi[] = "    --qc-map-cutoff, the minimum mapping quality required for a breakpoint to be reported [0.0]";
static const char __pyx_k_qc_map_file_bigWig_file_used_to[] = "    --qc-map-file, .bigWig file used to decide local genomic mappability, avaliable from: ";
static const char __pyx_k_qc_map_tool_the_tool_extracts_m[] = "    --qc-map-tool, the tool extracts mappability information from a bigWig file,avaliable from: ";
static const char __pyx_k_qc_split_minimum_alighment_of_c[] = "    --qc-split, minimum alighment of clipped parts of reads considered as a soft clip [20]";
static const char __pyx_k_qc_structure_minimum_quality_sc[] = "    --qc-structure, minimum quality score of a resolved structure to be considered as PASS and included in the output vcf file";
static const char __pyx_k_ref_index_folders_containin_pre[] = "    --ref-index, folders containin pre-indexed files, if applicable. For certain versions of human genome, the indexed files are availabel from https://github.com/mills-lab/svelter.";
static const char __pyx_k_reference_absolute_path_of_refe[] = "    --reference, absolute path of reference genome. eg: .../SVelter/reference/genome.fa";
static const char __pyx_k_split_min_len_the_minumum_lengt[] = "    --split-min-len, the minumum length of clip read considered as split; [10% of read length]";
static const char __pyx_k_support_folder_containing_all_s[] = "    --support, folder containing all supportive file including: Exclude.bed,CN2.bed,Segdup.bed";
static const char __pyx_k_workdir_writable_working_direct[] = "    --workdir, writable working directory.";
static const char __pyx_k_SVelter_0_1_Contact_xuefzhao_umi[] = "SVelter-0.1        Contact: xuefzhao@umich.edu      Last Update:2015-08-20";
static const char __pyx_k_SVelter_Setup_should_be_run_firs[] = "SVelter Setup should be run first";
static const char __pyx_k_Type_in_SVelter_options_for_deta[] = "Type in SVelter [options] for detailed parameters";
static const char __pyx_k_Usage_SVelter_Options_Parameters[] = "Usage: SVelter [Options] [Parameters]";
static const char __pyx_k_print_default_parameters_bpinteg[] = "print_default_parameters_bpintegrate";
static const char __pyx_k_print_default_parameters_bpsearc[] = "print_default_parameters_bpsearch";
static const char __pyx_k_print_default_parameters_nullmod[] = "print_default_parameters_nullmodel";
static const char __pyx_k_print_default_parameters_predefi[] = "print_default_parameters_predefinedbp";
static const char __pyx_k_print_default_parameters_svinteg[] = "print_default_parameters_svintegrate";
static const char __pyx_k_print_default_parameters_svpredi[] = "print_default_parameters_svpredict";
static const char __pyx_k_prefix_output_prefix_for_vcf_an_2[] = "    --prefix, output prefix for vcf and svelter files";
static const char __pyx_k_qc_align_minimum_alignment_qual_2[] = "    --qc-align, minimum alignment quality required for mapped reads in bam file [20)";
static const char __pyx_k_qc_map_cutoff_the_minimum_mappi_2[] = "    --qc-map-cutoff, the minimum mapping quality required for a breakpoint to be reported [0.0)";
static const char __pyx_k_qc_map_file_bigWig_file_used_to_2[] = "    --qc-map-file, .bigWig file used to decide local genomic mappability, avaliable from: ftp://hgdownload.cse.ucsc.edu/goldenPath/currentGenomes/Homo_sapiens/encodeDCC/wgEncodeMapability/";
static const char __pyx_k_qc_map_tool_the_tool_extracts_m_2[] = "    --qc-map-tool, the tool extracts mappability information from a bigWig file,avaliable from: http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/bigWigSummary";
static const char __pyx_k_qc_split_minimum_alighment_of_c_2[] = "    --qc-split, minimum alighment of clipped parts of reads considered as a soft clip [20)";
static const char __pyx_k_split_min_len_the_minumum_lengt_2[] = "    --split-min-len, the minumum length of clip read considered as split; (default:10% of read length)";
static PyObject *__pyx_kp_s_BPIntegrate;
static PyObject *__pyx_kp_s_BPSearch;
static PyObject *__pyx_kp_s_Clean;
static PyObject *__pyx_kp_s_NullModel;
static PyObject *__pyx_kp_s_Optional_Parameters;
static PyObject *__pyx_kp_s_Options;
static PyObject *__pyx_kp_s_Required_Parameters;
static PyObject *__pyx_kp_s_SVIntegrate;
static PyObject *__pyx_kp_s_SVPredict;
static PyObject *__pyx_kp_s_SVelter_0_1_Contact_xuefzhao_umi;
static PyObject *__pyx_kp_s_SVelter_BPIntegrate;
static PyObject *__pyx_kp_s_SVelter_BPSearch;
static PyObject *__pyx_kp_s_SVelter_Clean;
static PyObject *__pyx_kp_s_SVelter_NullModel;
static PyObject *__pyx_kp_s_SVelter_PredefinedBP;
static PyObject *__pyx_kp_s_SVelter_SVIntegrate;
static PyObject *__pyx_kp_s_SVelter_SVPredict;
static PyObject *__pyx_kp_s_SVelter_Setup;
static PyObject *__pyx_kp_s_SVelter_Setup_should_be_run_firs;
static PyObject *__pyx_kp_s_Setup;
static PyObject *__pyx_kp_s_Type_in_SVelter_options_for_deta;
static PyObject *__pyx_kp_s_Usage_SVelter_Options_Parameters;
static PyObject *__pyx_kp_s__2;
static PyObject *__pyx_kp_s__5;
static PyObject *__pyx_kp_s_batch_specify_number_of_structu;
static PyObject *__pyx_kp_s_bp_file_input_txt_file_containi;
static PyObject *__pyx_kp_s_bp_path_specify_the_path_where;
static PyObject *__pyx_kp_s_chromosome_name_of_chromosome_t;
static PyObject *__pyx_kp_s_data_talkowski_xuefang_local_sr;
static PyObject *__pyx_kp_s_ftp_hgdownload_cse_ucsc_edu_gol;
static PyObject *__pyx_kp_s_http_hgdownload_cse_ucsc_edu_ad;
static PyObject *__pyx_kp_s_input_bed_predefined_breakpoint;
static PyObject *__pyx_kp_s_input_path_path_of_coverage_fil;
static PyObject *__pyx_n_s_main;
static PyObject *__pyx_kp_s_null_copyneutral_length_minimum;
static PyObject *__pyx_kp_s_null_copyneutral_perc_percentag;
static PyObject *__pyx_kp_s_null_model_specify_which_stat_m;
static PyObject *__pyx_kp_s_null_random_length_specify_the;
static PyObject *__pyx_kp_s_null_random_num_specify_the_num;
static PyObject *__pyx_kp_s_num_iteration_maximum_number_of;
static PyObject *__pyx_kp_s_ploidy_limit_algorithm_to_speci;
static PyObject *__pyx_kp_s_prefix_output_prefix_for_vcf_an;
static PyObject *__pyx_kp_s_prefix_output_prefix_for_vcf_an_2;
static PyObject *__pyx_n_s_print;
static PyObject *__pyx_n_s_print_default_parameters;
static PyObject *__pyx_n_s_print_default_parameters_bpinteg;
static PyObject *__pyx_n_s_print_default_parameters_bpsearc;
static PyObject *__pyx_n_s_print_default_parameters_clean;
static PyObject *__pyx_n_s_print_default_parameters_nullmod;
static PyObject *__pyx_n_s_print_default_parameters_predefi;
static PyObject *__pyx_n_s_print_default_parameters_setup;
static PyObject *__pyx_n_s_print_default_parameters_svinteg;
static PyObject *__pyx_n_s_print_default_parameters_svpredi;
static PyObject *__pyx_kp_s_qc_align_minimum_alignment_qual;
static PyObject *__pyx_kp_s_qc_align_minimum_alignment_qual_2;
static PyObject *__pyx_kp_s_qc_map_cutoff_the_minimum_mappi;
static PyObject *__pyx_kp_s_qc_map_cutoff_the_minimum_mappi_2;
static PyObject *__pyx_kp_s_qc_map_file_bigWig_file_used_to;
static PyObject *__pyx_kp_s_qc_map_file_bigWig_file_used_to_2;
static PyObject *__pyx_kp_s_qc_map_tool_the_tool_extracts_m;
static PyObject *__pyx_kp_s_qc_map_tool_the_tool_extracts_m_2;
static PyObject *__pyx_kp_s_qc_split_minimum_alighment_of_c;
static PyObject *__pyx_kp_s_qc_split_minimum_alighment_of_c_2;
static PyObject *__pyx_kp_s_qc_structure_minimum_quality_sc;
static PyObject *__pyx_kp_s_ref_index_folders_containin_pre;
static PyObject *__pyx_kp_s_reference_absolute_path_of_refe;
static PyObject *__pyx_kp_s_sample_input_alignment_file_in;
static PyObject *__pyx_kp_s_split_min_len_the_minumum_lengt;
static PyObject *__pyx_kp_s_split_min_len_the_minumum_lengt_2;
static PyObject *__pyx_kp_s_support_folder_containing_all_s;
static PyObject *__pyx_n_s_svelter_sv_readme;
static PyObject *__pyx_n_s_test;
static PyObject *__pyx_kp_s_workdir_writable_working_direct;
static PyObject *__pyx_pf_10svelter_sv_6readme_print_default_parameters(CYTHON_UNUSED PyObject *__pyx_self); /* proto */
static PyObject *__pyx_pf_10svelter_sv_6readme_2print_default_parameters_predefinedbp(CYTHON_UNUSED PyObject *__pyx_self); /* proto */
static PyObject *__pyx_pf_10svelter_sv_6readme_4print_default_parameters_clean(CYTHON_UNUSED PyObject *__pyx_self); /* proto */
static PyObject *__pyx_pf_10svelter_sv_6readme_6print_default_parameters_setup(CYTHON_UNUSED PyObject *__pyx_self); /* proto */
static PyObject *__pyx_pf_10svelter_sv_6readme_8print_default_parameters_nullmodel(CYTHON_UNUSED PyObject *__pyx_self); /* proto */
static PyObject *__pyx_pf_10svelter_sv_6readme_10print_default_parameters_bpsearch(CYTHON_UNUSED PyObject *__pyx_self); /* proto */
static PyObject *__pyx_pf_10svelter_sv_6readme_12print_default_parameters_bpintegrate(CYTHON_UNUSED PyObject *__pyx_self); /* proto */
static PyObject *__pyx_pf_10svelter_sv_6readme_14print_default_parameters_svpredict(CYTHON_UNUSED PyObject *__pyx_self); /* proto */
static PyObject *__pyx_pf_10svelter_sv_6readme_16print_default_parameters_svintegrate(CYTHON_UNUSED PyObject *__pyx_self); /* proto */
static PyObject *__pyx_tuple_;
static PyObject *__pyx_tuple__3;
static PyObject *__pyx_tuple__4;
static PyObject *__pyx_tuple__6;
static PyObject *__pyx_tuple__7;
static PyObject *__pyx_tuple__8;
static PyObject *__pyx_tuple__9;
static PyObject *__pyx_tuple__10;
static PyObject *__pyx_tuple__11;
static PyObject *__pyx_tuple__12;
static PyObject *__pyx_tuple__13;
static PyObject *__pyx_tuple__14;
static PyObject *__pyx_tuple__15;
static PyObject *__pyx_tuple__16;
static PyObject *__pyx_tuple__17;
static PyObject *__pyx_tuple__18;
static PyObject *__pyx_tuple__19;
static PyObject *__pyx_tuple__20;
static PyObject *__pyx_tuple__21;
static PyObject *__pyx_tuple__22;
static PyObject *__pyx_tuple__23;
static PyObject *__pyx_tuple__24;
static PyObject *__pyx_tuple__25;
static PyObject *__pyx_tuple__26;
static PyObject *__pyx_tuple__27;
static PyObject *__pyx_tuple__28;
static PyObject *__pyx_tuple__29;
static PyObject *__pyx_tuple__30;
static PyObject *__pyx_tuple__31;
static PyObject *__pyx_tuple__32;
static PyObject *__pyx_tuple__33;
static PyObject *__pyx_tuple__34;
static PyObject *__pyx_tuple__35;
static PyObject *__pyx_tuple__36;
static PyObject *__pyx_tuple__37;
static PyObject *__pyx_tuple__38;
static PyObject *__pyx_tuple__39;
static PyObject *__pyx_tuple__40;
static PyObject *__pyx_tuple__41;
static PyObject *__pyx_tuple__42;
static PyObject *__pyx_tuple__43;
static PyObject *__pyx_tuple__44;
static PyObject *__pyx_tuple__45;
static PyObject *__pyx_tuple__46;
static PyObject *__pyx_tuple__47;
static PyObject *__pyx_tuple__48;
static PyObject *__pyx_tuple__49;
static PyObject *__pyx_tuple__50;
static PyObject *__pyx_tuple__51;
static PyObject *__pyx_tuple__52;
static PyObject *__pyx_tuple__53;
static PyObject *__pyx_tuple__54;
static PyObject *__pyx_tuple__55;
static PyObject *__pyx_tuple__56;
static PyObject *__pyx_tuple__57;
static PyObject *__pyx_tuple__58;
static PyObject *__pyx_tuple__59;
static PyObject *__pyx_tuple__60;
static PyObject *__pyx_tuple__61;
static PyObject *__pyx_tuple__62;
static PyObject *__pyx_tuple__63;
static PyObject *__pyx_tuple__64;
static PyObject *__pyx_tuple__65;
static PyObject *__pyx_tuple__66;
static PyObject *__pyx_tuple__67;
static PyObject *__pyx_tuple__68;
static PyObject *__pyx_tuple__69;
static PyObject *__pyx_tuple__70;
static PyObject *__pyx_tuple__71;
static PyObject *__pyx_tuple__72;
static PyObject *__pyx_tuple__73;
static PyObject *__pyx_tuple__74;
static PyObject *__pyx_tuple__75;
static PyObject *__pyx_tuple__76;
static PyObject *__pyx_tuple__77;
static PyObject *__pyx_tuple__78;
static PyObject *__pyx_tuple__79;
static PyObject *__pyx_tuple__80;
static PyObject *__pyx_tuple__81;
static PyObject *__pyx_tuple__82;
static PyObject *__pyx_tuple__83;
static PyObject *__pyx_tuple__84;
static PyObject *__pyx_tuple__85;
static PyObject *__pyx_tuple__86;
static PyObject *__pyx_tuple__87;
static PyObject *__pyx_tuple__88;
static PyObject *__pyx_tuple__89;
static PyObject *__pyx_tuple__90;
static PyObject *__pyx_tuple__91;
static PyObject *__pyx_tuple__92;
static PyObject *__pyx_tuple__93;
static PyObject *__pyx_tuple__94;
static PyObject *__pyx_tuple__95;
static PyObject *__pyx_tuple__96;
static PyObject *__pyx_tuple__97;
static PyObject *__pyx_tuple__98;
static PyObject *__pyx_tuple__99;
static PyObject *__pyx_tuple__100;
static PyObject *__pyx_tuple__101;
static PyObject *__pyx_tuple__102;
static PyObject *__pyx_tuple__103;
static PyObject *__pyx_tuple__104;
static PyObject *__pyx_tuple__105;
static PyObject *__pyx_tuple__106;
static PyObject *__pyx_tuple__107;
static PyObject *__pyx_tuple__108;
static PyObject *__pyx_tuple__109;
static PyObject *__pyx_tuple__110;
static PyObject *__pyx_tuple__111;
static PyObject *__pyx_tuple__112;
static PyObject *__pyx_tuple__113;
static PyObject *__pyx_tuple__114;
static PyObject *__pyx_tuple__115;
static PyObject *__pyx_tuple__116;
static PyObject *__pyx_tuple__117;
static PyObject *__pyx_tuple__118;
static PyObject *__pyx_tuple__119;
static PyObject *__pyx_codeobj__120;
static PyObject *__pyx_codeobj__121;
static PyObject *__pyx_codeobj__122;
static PyObject *__pyx_codeobj__123;
static PyObject *__pyx_codeobj__124;
static PyObject *__pyx_codeobj__125;
static PyObject *__pyx_codeobj__126;
static PyObject *__pyx_codeobj__127;
static PyObject *__pyx_codeobj__128;

/* "svelter_sv/readme.pyx":2
 * from __future__ import print_function
 * def print_default_parameters():             # <<<<<<<<<<<<<<
 *     print('SVelter-0.1        Contact: xuefzhao@umich.edu      Last Update:2015-08-20')
 *     print('')
 */

/* Python wrapper */
static PyObject *__pyx_pw_10svelter_sv_6readme_1print_default_parameters(PyObject *__pyx_self, CYTHON_UNUSED PyObject *unused); /*proto*/
static PyMethodDef __pyx_mdef_10svelter_sv_6readme_1print_default_parameters = {"print_default_parameters", (PyCFunction)__pyx_pw_10svelter_sv_6readme_1print_default_parameters, METH_NOARGS, 0};
static PyObject *__pyx_pw_10svelter_sv_6readme_1print_default_parameters(PyObject *__pyx_self, CYTHON_UNUSED PyObject *unused) {
  PyObject *__pyx_r = 0;
  __Pyx_RefNannyDeclarations
  __Pyx_RefNannySetupContext("print_default_parameters (wrapper)", 0);
  __pyx_r = __pyx_pf_10svelter_sv_6readme_print_default_parameters(__pyx_self);

  /* function exit code */
  __Pyx_RefNannyFinishContext();
  return __pyx_r;
}

static PyObject *__pyx_pf_10svelter_sv_6readme_print_default_parameters(CYTHON_UNUSED PyObject *__pyx_self) {
  PyObject *__pyx_r = NULL;
  __Pyx_RefNannyDeclarations
  PyObject *__pyx_t_1 = NULL;
  __Pyx_RefNannySetupContext("print_default_parameters", 0);

  /* "svelter_sv/readme.pyx":3
 * from __future__ import print_function
 * def print_default_parameters():
 *     print('SVelter-0.1        Contact: xuefzhao@umich.edu      Last Update:2015-08-20')             # <<<<<<<<<<<<<<
 *     print('')
 *     print('Usage: SVelter [Options] [Parameters]')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple_, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 3, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":4
 * def print_default_parameters():
 *     print('SVelter-0.1        Contact: xuefzhao@umich.edu      Last Update:2015-08-20')
 *     print('')             # <<<<<<<<<<<<<<
 *     print('Usage: SVelter [Options] [Parameters]')
 *     print(' ')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__3, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 4, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":5
 *     print('SVelter-0.1        Contact: xuefzhao@umich.edu      Last Update:2015-08-20')
 *     print('')
 *     print('Usage: SVelter [Options] [Parameters]')             # <<<<<<<<<<<<<<
 *     print(' ')
 *     print('Options:')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__4, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 5, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":6
 *     print('')
 *     print('Usage: SVelter [Options] [Parameters]')
 *     print(' ')             # <<<<<<<<<<<<<<
 *     print('Options:')
 *     print('    Setup')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__6, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 6, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":7
 *     print('Usage: SVelter [Options] [Parameters]')
 *     print(' ')
 *     print('Options:')             # <<<<<<<<<<<<<<
 *     print('    Setup')
 *     print('    Clean')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__7, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 7, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":8
 *     print(' ')
 *     print('Options:')
 *     print('    Setup')             # <<<<<<<<<<<<<<
 *     print('    Clean')
 *     print('    NullModel')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__8, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 8, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":9
 *     print('Options:')
 *     print('    Setup')
 *     print('    Clean')             # <<<<<<<<<<<<<<
 *     print('    NullModel')
 *     print('    BPSearch')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__9, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 9, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":10
 *     print('    Setup')
 *     print('    Clean')
 *     print('    NullModel')             # <<<<<<<<<<<<<<
 *     print('    BPSearch')
 *     print('    BPIntegrate')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__10, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 10, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":11
 *     print('    Clean')
 *     print('    NullModel')
 *     print('    BPSearch')             # <<<<<<<<<<<<<<
 *     print('    BPIntegrate')
 *     print('    SVPredict')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__11, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 11, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":12
 *     print('    NullModel')
 *     print('    BPSearch')
 *     print('    BPIntegrate')             # <<<<<<<<<<<<<<
 *     print('    SVPredict')
 *     print('    SVIntegrate')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__12, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 12, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":13
 *     print('    BPSearch')
 *     print('    BPIntegrate')
 *     print('    SVPredict')             # <<<<<<<<<<<<<<
 *     print('    SVIntegrate')
 *     print('')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__13, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 13, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":14
 *     print('    BPIntegrate')
 *     print('    SVPredict')
 *     print('    SVIntegrate')             # <<<<<<<<<<<<<<
 *     print('')
 *     print('SVelter Setup should be run first')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__14, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 14, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":15
 *     print('    SVPredict')
 *     print('    SVIntegrate')
 *     print('')             # <<<<<<<<<<<<<<
 *     print('SVelter Setup should be run first')
 *     print(' ')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__15, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 15, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":16
 *     print('    SVIntegrate')
 *     print('')
 *     print('SVelter Setup should be run first')             # <<<<<<<<<<<<<<
 *     print(' ')
 *     print('Type in SVelter [options] for detailed parameters')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__16, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 16, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":17
 *     print('')
 *     print('SVelter Setup should be run first')
 *     print(' ')             # <<<<<<<<<<<<<<
 *     print('Type in SVelter [options] for detailed parameters')
 * def print_default_parameters_predefinedbp():
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__17, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 17, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":18
 *     print('SVelter Setup should be run first')
 *     print(' ')
 *     print('Type in SVelter [options] for detailed parameters')             # <<<<<<<<<<<<<<
 * def print_default_parameters_predefinedbp():
 *     print('SVelter PredefinedBP')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__18, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 18, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":2
 * from __future__ import print_function
 * def print_default_parameters():             # <<<<<<<<<<<<<<
 *     print('SVelter-0.1        Contact: xuefzhao@umich.edu      Last Update:2015-08-20')
 *     print('')
 */

  /* function exit code */
  __pyx_r = Py_None; __Pyx_INCREF(Py_None);
  goto __pyx_L0;
  __pyx_L1_error:;
  __Pyx_XDECREF(__pyx_t_1);
  __Pyx_AddTraceback("svelter_sv.readme.print_default_parameters", __pyx_clineno, __pyx_lineno, __pyx_filename);
  __pyx_r = NULL;
  __pyx_L0:;
  __Pyx_XGIVEREF(__pyx_r);
  __Pyx_RefNannyFinishContext();
  return __pyx_r;
}

/* "svelter_sv/readme.pyx":19
 *     print(' ')
 *     print('Type in SVelter [options] for detailed parameters')
 * def print_default_parameters_predefinedbp():             # <<<<<<<<<<<<<<
 *     print('SVelter PredefinedBP')
 *     print(' ')
 */

/* Python wrapper */
static PyObject *__pyx_pw_10svelter_sv_6readme_3print_default_parameters_predefinedbp(PyObject *__pyx_self, CYTHON_UNUSED PyObject *unused); /*proto*/
static PyMethodDef __pyx_mdef_10svelter_sv_6readme_3print_default_parameters_predefinedbp = {"print_default_parameters_predefinedbp", (PyCFunction)__pyx_pw_10svelter_sv_6readme_3print_default_parameters_predefinedbp, METH_NOARGS, 0};
static PyObject *__pyx_pw_10svelter_sv_6readme_3print_default_parameters_predefinedbp(PyObject *__pyx_self, CYTHON_UNUSED PyObject *unused) {
  PyObject *__pyx_r = 0;
  __Pyx_RefNannyDeclarations
  __Pyx_RefNannySetupContext("print_default_parameters_predefinedbp (wrapper)", 0);
  __pyx_r = __pyx_pf_10svelter_sv_6readme_2print_default_parameters_predefinedbp(__pyx_self);

  /* function exit code */
  __Pyx_RefNannyFinishContext();
  return __pyx_r;
}

static PyObject *__pyx_pf_10svelter_sv_6readme_2print_default_parameters_predefinedbp(CYTHON_UNUSED PyObject *__pyx_self) {
  PyObject *__pyx_r = NULL;
  __Pyx_RefNannyDeclarations
  PyObject *__pyx_t_1 = NULL;
  __Pyx_RefNannySetupContext("print_default_parameters_predefinedbp", 0);

  /* "svelter_sv/readme.pyx":20
 *     print('Type in SVelter [options] for detailed parameters')
 * def print_default_parameters_predefinedbp():
 *     print('SVelter PredefinedBP')             # <<<<<<<<<<<<<<
 *     print(' ')
 *     print('Required Parameters:')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__19, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 20, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":21
 * def print_default_parameters_predefinedbp():
 *     print('SVelter PredefinedBP')
 *     print(' ')             # <<<<<<<<<<<<<<
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__20, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 21, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":22
 *     print('SVelter PredefinedBP')
 *     print(' ')
 *     print('Required Parameters:')             # <<<<<<<<<<<<<<
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__21, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 22, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":23
 *     print(' ')
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')             # <<<<<<<<<<<<<<
 *     print('    --workdir, writable working directory.')
 *     print('    --input-bed, predefined breakpoints in bed format')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__22, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 23, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":24
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')             # <<<<<<<<<<<<<<
 *     print('    --input-bed, predefined breakpoints in bed format')
 *     print('')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__23, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 24, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":25
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')
 *     print('    --input-bed, predefined breakpoints in bed format')             # <<<<<<<<<<<<<<
 *     print('')
 *     print('Optional Parameters:')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__24, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 25, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":26
 *     print('    --workdir, writable working directory.')
 *     print('    --input-bed, predefined breakpoints in bed format')
 *     print('')             # <<<<<<<<<<<<<<
 *     print('Optional Parameters:')
 *     print('    --prefix, output prefix for vcf and svelter files [input.vcf, input.svelter]')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__25, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 26, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":27
 *     print('    --input-bed, predefined breakpoints in bed format')
 *     print('')
 *     print('Optional Parameters:')             # <<<<<<<<<<<<<<
 *     print('    --prefix, output prefix for vcf and svelter files [input.vcf, input.svelter]')
 *     print('    --num-iteration, maximum number of iterations per structure will run')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__26, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 27, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":28
 *     print('')
 *     print('Optional Parameters:')
 *     print('    --prefix, output prefix for vcf and svelter files [input.vcf, input.svelter]')             # <<<<<<<<<<<<<<
 *     print('    --num-iteration, maximum number of iterations per structure will run')
 *     print('    --ploidy, limit algorithm to specific zygosity (0:heterozygous only; 1:homozygous only; 2:both; default:2)')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__27, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 28, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":29
 *     print('Optional Parameters:')
 *     print('    --prefix, output prefix for vcf and svelter files [input.vcf, input.svelter]')
 *     print('    --num-iteration, maximum number of iterations per structure will run')             # <<<<<<<<<<<<<<
 *     print('    --ploidy, limit algorithm to specific zygosity (0:heterozygous only; 1:homozygous only; 2:both; default:2)')
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__28, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 29, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":30
 *     print('    --prefix, output prefix for vcf and svelter files [input.vcf, input.svelter]')
 *     print('    --num-iteration, maximum number of iterations per structure will run')
 *     print('    --ploidy, limit algorithm to specific zygosity (0:heterozygous only; 1:homozygous only; 2:both; default:2)')             # <<<<<<<<<<<<<<
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')
 *     print('    --null-copyneutral-length, minimum length requirement for --copyneutral regions used to build null model [2000]')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__29, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 30, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":31
 *     print('    --num-iteration, maximum number of iterations per structure will run')
 *     print('    --ploidy, limit algorithm to specific zygosity (0:heterozygous only; 1:homozygous only; 2:both; default:2)')
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')             # <<<<<<<<<<<<<<
 *     print('    --null-copyneutral-length, minimum length requirement for --copyneutral regions used to build null model [2000]')
 *     print('    --null-copyneutral-perc, percentage of regions from --copyneutral to utilize [0.1]')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__30, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 31, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":32
 *     print('    --ploidy, limit algorithm to specific zygosity (0:heterozygous only; 1:homozygous only; 2:both; default:2)')
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')
 *     print('    --null-copyneutral-length, minimum length requirement for --copyneutral regions used to build null model [2000]')             # <<<<<<<<<<<<<<
 *     print('    --null-copyneutral-perc, percentage of regions from --copyneutral to utilize [0.1]')
 *     print('    --null-random-length, specify the length of random regions if --copyneutral parameter not used [5000]')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__31, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 32, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":33
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')
 *     print('    --null-copyneutral-length, minimum length requirement for --copyneutral regions used to build null model [2000]')
 *     print('    --null-copyneutral-perc, percentage of regions from --copyneutral to utilize [0.1]')             # <<<<<<<<<<<<<<
 *     print('    --null-random-length, specify the length of random regions if --copyneutral parameter not used [5000]')
 *     print('    --null-random-num, specify the number of random regions if --copyneutral parameter not used [10000]')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__32, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 33, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":34
 *     print('    --null-copyneutral-length, minimum length requirement for --copyneutral regions used to build null model [2000]')
 *     print('    --null-copyneutral-perc, percentage of regions from --copyneutral to utilize [0.1]')
 *     print('    --null-random-length, specify the length of random regions if --copyneutral parameter not used [5000]')             # <<<<<<<<<<<<<<
 *     print('    --null-random-num, specify the number of random regions if --copyneutral parameter not used [10000]')
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20]')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__33, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 34, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":35
 *     print('    --null-copyneutral-perc, percentage of regions from --copyneutral to utilize [0.1]')
 *     print('    --null-random-length, specify the length of random regions if --copyneutral parameter not used [5000]')
 *     print('    --null-random-num, specify the number of random regions if --copyneutral parameter not used [10000]')             # <<<<<<<<<<<<<<
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20]')
 *     print('    --qc-split, minimum alighment of clipped parts of reads considered as a soft clip [20]')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__34, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 35, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":36
 *     print('    --null-random-length, specify the length of random regions if --copyneutral parameter not used [5000]')
 *     print('    --null-random-num, specify the number of random regions if --copyneutral parameter not used [10000]')
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20]')             # <<<<<<<<<<<<<<
 *     print('    --qc-split, minimum alighment of clipped parts of reads considered as a soft clip [20]')
 *     print('    --split-min-len, the minumum length of clip read considered as split; [10% of read length]')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__35, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 36, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":37
 *     print('    --null-random-num, specify the number of random regions if --copyneutral parameter not used [10000]')
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20]')
 *     print('    --qc-split, minimum alighment of clipped parts of reads considered as a soft clip [20]')             # <<<<<<<<<<<<<<
 *     print('    --split-min-len, the minumum length of clip read considered as split; [10% of read length]')
 *     print('    --qc-structure, minimum quality score of a resolved structure to be considered as PASS and included in the output vcf file')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__36, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 37, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":38
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20]')
 *     print('    --qc-split, minimum alighment of clipped parts of reads considered as a soft clip [20]')
 *     print('    --split-min-len, the minumum length of clip read considered as split; [10% of read length]')             # <<<<<<<<<<<<<<
 *     print('    --qc-structure, minimum quality score of a resolved structure to be considered as PASS and included in the output vcf file')
 *     print('    --qc-map-tool, the tool extracts mappability information from a bigWig file,avaliable from: ')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__37, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 38, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":39
 *     print('    --qc-split, minimum alighment of clipped parts of reads considered as a soft clip [20]')
 *     print('    --split-min-len, the minumum length of clip read considered as split; [10% of read length]')
 *     print('    --qc-structure, minimum quality score of a resolved structure to be considered as PASS and included in the output vcf file')             # <<<<<<<<<<<<<<
 *     print('    --qc-map-tool, the tool extracts mappability information from a bigWig file,avaliable from: ')
 *     print('                     http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/bigWigSummary')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__38, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 39, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":40
 *     print('    --split-min-len, the minumum length of clip read considered as split; [10% of read length]')
 *     print('    --qc-structure, minimum quality score of a resolved structure to be considered as PASS and included in the output vcf file')
 *     print('    --qc-map-tool, the tool extracts mappability information from a bigWig file,avaliable from: ')             # <<<<<<<<<<<<<<
 *     print('                     http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/bigWigSummary')
 *     print('    --qc-map-file, .bigWig file used to decide local genomic mappability, avaliable from: ')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__39, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 40, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":41
 *     print('    --qc-structure, minimum quality score of a resolved structure to be considered as PASS and included in the output vcf file')
 *     print('    --qc-map-tool, the tool extracts mappability information from a bigWig file,avaliable from: ')
 *     print('                     http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/bigWigSummary')             # <<<<<<<<<<<<<<
 *     print('    --qc-map-file, .bigWig file used to decide local genomic mappability, avaliable from: ')
 *     print('                     ftp://hgdownload.cse.ucsc.edu/goldenPath/currentGenomes/Homo_sapiens/encodeDCC/wgEncodeMapability/')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__40, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 41, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":42
 *     print('    --qc-map-tool, the tool extracts mappability information from a bigWig file,avaliable from: ')
 *     print('                     http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/bigWigSummary')
 *     print('    --qc-map-file, .bigWig file used to decide local genomic mappability, avaliable from: ')             # <<<<<<<<<<<<<<
 *     print('                     ftp://hgdownload.cse.ucsc.edu/goldenPath/currentGenomes/Homo_sapiens/encodeDCC/wgEncodeMapability/')
 *     print('    --qc-map-cutoff, the minimum mapping quality required for a breakpoint to be reported [0.0]')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__41, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 42, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":43
 *     print('                     http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/bigWigSummary')
 *     print('    --qc-map-file, .bigWig file used to decide local genomic mappability, avaliable from: ')
 *     print('                     ftp://hgdownload.cse.ucsc.edu/goldenPath/currentGenomes/Homo_sapiens/encodeDCC/wgEncodeMapability/')             # <<<<<<<<<<<<<<
 *     print('    --qc-map-cutoff, the minimum mapping quality required for a breakpoint to be reported [0.0]')
 * def print_default_parameters_clean():
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__42, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 43, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":44
 *     print('    --qc-map-file, .bigWig file used to decide local genomic mappability, avaliable from: ')
 *     print('                     ftp://hgdownload.cse.ucsc.edu/goldenPath/currentGenomes/Homo_sapiens/encodeDCC/wgEncodeMapability/')
 *     print('    --qc-map-cutoff, the minimum mapping quality required for a breakpoint to be reported [0.0]')             # <<<<<<<<<<<<<<
 * def print_default_parameters_clean():
 *     print('SVelter Clean')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__43, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 44, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":19
 *     print(' ')
 *     print('Type in SVelter [options] for detailed parameters')
 * def print_default_parameters_predefinedbp():             # <<<<<<<<<<<<<<
 *     print('SVelter PredefinedBP')
 *     print(' ')
 */

  /* function exit code */
  __pyx_r = Py_None; __Pyx_INCREF(Py_None);
  goto __pyx_L0;
  __pyx_L1_error:;
  __Pyx_XDECREF(__pyx_t_1);
  __Pyx_AddTraceback("svelter_sv.readme.print_default_parameters_predefinedbp", __pyx_clineno, __pyx_lineno, __pyx_filename);
  __pyx_r = NULL;
  __pyx_L0:;
  __Pyx_XGIVEREF(__pyx_r);
  __Pyx_RefNannyFinishContext();
  return __pyx_r;
}

/* "svelter_sv/readme.pyx":45
 *     print('                     ftp://hgdownload.cse.ucsc.edu/goldenPath/currentGenomes/Homo_sapiens/encodeDCC/wgEncodeMapability/')
 *     print('    --qc-map-cutoff, the minimum mapping quality required for a breakpoint to be reported [0.0]')
 * def print_default_parameters_clean():             # <<<<<<<<<<<<<<
 *     print('SVelter Clean')
 *     print('')
 */

/* Python wrapper */
static PyObject *__pyx_pw_10svelter_sv_6readme_5print_default_parameters_clean(PyObject *__pyx_self, CYTHON_UNUSED PyObject *unused); /*proto*/
static PyMethodDef __pyx_mdef_10svelter_sv_6readme_5print_default_parameters_clean = {"print_default_parameters_clean", (PyCFunction)__pyx_pw_10svelter_sv_6readme_5print_default_parameters_clean, METH_NOARGS, 0};
static PyObject *__pyx_pw_10svelter_sv_6readme_5print_default_parameters_clean(PyObject *__pyx_self, CYTHON_UNUSED PyObject *unused) {
  PyObject *__pyx_r = 0;
  __Pyx_RefNannyDeclarations
  __Pyx_RefNannySetupContext("print_default_parameters_clean (wrapper)", 0);
  __pyx_r = __pyx_pf_10svelter_sv_6readme_4print_default_parameters_clean(__pyx_self);

  /* function exit code */
  __Pyx_RefNannyFinishContext();
  return __pyx_r;
}

static PyObject *__pyx_pf_10svelter_sv_6readme_4print_default_parameters_clean(CYTHON_UNUSED PyObject *__pyx_self) {
  PyObject *__pyx_r = NULL;
  __Pyx_RefNannyDeclarations
  PyObject *__pyx_t_1 = NULL;
  __Pyx_RefNannySetupContext("print_default_parameters_clean", 0);

  /* "svelter_sv/readme.pyx":46
 *     print('    --qc-map-cutoff, the minimum mapping quality required for a breakpoint to be reported [0.0]')
 * def print_default_parameters_clean():
 *     print('SVelter Clean')             # <<<<<<<<<<<<<<
 *     print('')
 *     print('Required Parameters:')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__44, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 46, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":47
 * def print_default_parameters_clean():
 *     print('SVelter Clean')
 *     print('')             # <<<<<<<<<<<<<<
 *     print('Required Parameters:')
 *     print('    --workdir, writable working directory.')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__45, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 47, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":48
 *     print('SVelter Clean')
 *     print('')
 *     print('Required Parameters:')             # <<<<<<<<<<<<<<
 *     print('    --workdir, writable working directory.')
 * def print_default_parameters_setup():
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__46, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 48, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":49
 *     print('')
 *     print('Required Parameters:')
 *     print('    --workdir, writable working directory.')             # <<<<<<<<<<<<<<
 * def print_default_parameters_setup():
 *     print('SVelter Setup')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__47, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 49, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":45
 *     print('                     ftp://hgdownload.cse.ucsc.edu/goldenPath/currentGenomes/Homo_sapiens/encodeDCC/wgEncodeMapability/')
 *     print('    --qc-map-cutoff, the minimum mapping quality required for a breakpoint to be reported [0.0]')
 * def print_default_parameters_clean():             # <<<<<<<<<<<<<<
 *     print('SVelter Clean')
 *     print('')
 */

  /* function exit code */
  __pyx_r = Py_None; __Pyx_INCREF(Py_None);
  goto __pyx_L0;
  __pyx_L1_error:;
  __Pyx_XDECREF(__pyx_t_1);
  __Pyx_AddTraceback("svelter_sv.readme.print_default_parameters_clean", __pyx_clineno, __pyx_lineno, __pyx_filename);
  __pyx_r = NULL;
  __pyx_L0:;
  __Pyx_XGIVEREF(__pyx_r);
  __Pyx_RefNannyFinishContext();
  return __pyx_r;
}

/* "svelter_sv/readme.pyx":50
 *     print('Required Parameters:')
 *     print('    --workdir, writable working directory.')
 * def print_default_parameters_setup():             # <<<<<<<<<<<<<<
 *     print('SVelter Setup')
 *     print('')
 */

/* Python wrapper */
static PyObject *__pyx_pw_10svelter_sv_6readme_7print_default_parameters_setup(PyObject *__pyx_self, CYTHON_UNUSED PyObject *unused); /*proto*/
static PyMethodDef __pyx_mdef_10svelter_sv_6readme_7print_default_parameters_setup = {"print_default_parameters_setup", (PyCFunction)__pyx_pw_10svelter_sv_6readme_7print_default_parameters_setup, METH_NOARGS, 0};
static PyObject *__pyx_pw_10svelter_sv_6readme_7print_default_parameters_setup(PyObject *__pyx_self, CYTHON_UNUSED PyObject *unused) {
  PyObject *__pyx_r = 0;
  __Pyx_RefNannyDeclarations
  __Pyx_RefNannySetupContext("print_default_parameters_setup (wrapper)", 0);
  __pyx_r = __pyx_pf_10svelter_sv_6readme_6print_default_parameters_setup(__pyx_self);

  /* function exit code */
  __Pyx_RefNannyFinishContext();
  return __pyx_r;
}

static PyObject *__pyx_pf_10svelter_sv_6readme_6print_default_parameters_setup(CYTHON_UNUSED PyObject *__pyx_self) {
  PyObject *__pyx_r = NULL;
  __Pyx_RefNannyDeclarations
  PyObject *__pyx_t_1 = NULL;
  __Pyx_RefNannySetupContext("print_default_parameters_setup", 0);

  /* "svelter_sv/readme.pyx":51
 *     print('    --workdir, writable working directory.')
 * def print_default_parameters_setup():
 *     print('SVelter Setup')             # <<<<<<<<<<<<<<
 *     print('')
 *     print('Required Parameters:')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__48, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 51, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":52
 * def print_default_parameters_setup():
 *     print('SVelter Setup')
 *     print('')             # <<<<<<<<<<<<<<
 *     print('Required Parameters:')
 *     print('    --workdir, writable working directory.')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__49, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 52, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":53
 *     print('SVelter Setup')
 *     print('')
 *     print('Required Parameters:')             # <<<<<<<<<<<<<<
 *     print('    --workdir, writable working directory.')
 *     print('    --reference, absolute path of reference genome. eg: .../SVelter/reference/genome.fa')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__50, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 53, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":54
 *     print('')
 *     print('Required Parameters:')
 *     print('    --workdir, writable working directory.')             # <<<<<<<<<<<<<<
 *     print('    --reference, absolute path of reference genome. eg: .../SVelter/reference/genome.fa')
 *     print('    --support, folder containing all supportive file including: Exclude.bed,CN2.bed,Segdup.bed')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__51, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 54, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":55
 *     print('Required Parameters:')
 *     print('    --workdir, writable working directory.')
 *     print('    --reference, absolute path of reference genome. eg: .../SVelter/reference/genome.fa')             # <<<<<<<<<<<<<<
 *     print('    --support, folder containing all supportive file including: Exclude.bed,CN2.bed,Segdup.bed')
 *     print('')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__52, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 55, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":56
 *     print('    --workdir, writable working directory.')
 *     print('    --reference, absolute path of reference genome. eg: .../SVelter/reference/genome.fa')
 *     print('    --support, folder containing all supportive file including: Exclude.bed,CN2.bed,Segdup.bed')             # <<<<<<<<<<<<<<
 *     print('')
 *     print('Optional Parameters:')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__53, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 56, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":57
 *     print('    --reference, absolute path of reference genome. eg: .../SVelter/reference/genome.fa')
 *     print('    --support, folder containing all supportive file including: Exclude.bed,CN2.bed,Segdup.bed')
 *     print('')             # <<<<<<<<<<<<<<
 *     print('Optional Parameters:')
 *     #print '    --support, folder containing all supportive file including: Exclude.bed,CN2.bed,Segdup.bed'
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__54, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 57, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":58
 *     print('    --support, folder containing all supportive file including: Exclude.bed,CN2.bed,Segdup.bed')
 *     print('')
 *     print('Optional Parameters:')             # <<<<<<<<<<<<<<
 *     #print '    --support, folder containing all supportive file including: Exclude.bed,CN2.bed,Segdup.bed'
 *     #print '    --svelter-path, folder which contains all SVelter scripts.'
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__55, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 58, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":64
 *     #print '    --exclude, absolute path of bed file indicating regions to be excluded from analysis. If not provided, no mappable regions will be excluded.'
 *     #print '    --segdup, absolute path of bed file indicating segmental duplications in genome, that would be excluded from analysis'
 *     print('    --ref-index, folders containin pre-indexed files, if applicable. For certain versions of human genome, the indexed files are availabel from https://github.com/mills-lab/svelter.')             # <<<<<<<<<<<<<<
 * def print_default_parameters_nullmodel():
 *     print('SVelter NullModel')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__56, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 64, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":50
 *     print('Required Parameters:')
 *     print('    --workdir, writable working directory.')
 * def print_default_parameters_setup():             # <<<<<<<<<<<<<<
 *     print('SVelter Setup')
 *     print('')
 */

  /* function exit code */
  __pyx_r = Py_None; __Pyx_INCREF(Py_None);
  goto __pyx_L0;
  __pyx_L1_error:;
  __Pyx_XDECREF(__pyx_t_1);
  __Pyx_AddTraceback("svelter_sv.readme.print_default_parameters_setup", __pyx_clineno, __pyx_lineno, __pyx_filename);
  __pyx_r = NULL;
  __pyx_L0:;
  __Pyx_XGIVEREF(__pyx_r);
  __Pyx_RefNannyFinishContext();
  return __pyx_r;
}

/* "svelter_sv/readme.pyx":65
 *     #print '    --segdup, absolute path of bed file indicating segmental duplications in genome, that would be excluded from analysis'
 *     print('    --ref-index, folders containin pre-indexed files, if applicable. For certain versions of human genome, the indexed files are availabel from https://github.com/mills-lab/svelter.')
 * def print_default_parameters_nullmodel():             # <<<<<<<<<<<<<<
 *     print('SVelter NullModel')
 *     print('')
 */

/* Python wrapper */
static PyObject *__pyx_pw_10svelter_sv_6readme_9print_default_parameters_nullmodel(PyObject *__pyx_self, CYTHON_UNUSED PyObject *unused); /*proto*/
static PyMethodDef __pyx_mdef_10svelter_sv_6readme_9print_default_parameters_nullmodel = {"print_default_parameters_nullmodel", (PyCFunction)__pyx_pw_10svelter_sv_6readme_9print_default_parameters_nullmodel, METH_NOARGS, 0};
static PyObject *__pyx_pw_10svelter_sv_6readme_9print_default_parameters_nullmodel(PyObject *__pyx_self, CYTHON_UNUSED PyObject *unused) {
  PyObject *__pyx_r = 0;
  __Pyx_RefNannyDeclarations
  __Pyx_RefNannySetupContext("print_default_parameters_nullmodel (wrapper)", 0);
  __pyx_r = __pyx_pf_10svelter_sv_6readme_8print_default_parameters_nullmodel(__pyx_self);

  /* function exit code */
  __Pyx_RefNannyFinishContext();
  return __pyx_r;
}

static PyObject *__pyx_pf_10svelter_sv_6readme_8print_default_parameters_nullmodel(CYTHON_UNUSED PyObject *__pyx_self) {
  PyObject *__pyx_r = NULL;
  __Pyx_RefNannyDeclarations
  PyObject *__pyx_t_1 = NULL;
  __Pyx_RefNannySetupContext("print_default_parameters_nullmodel", 0);

  /* "svelter_sv/readme.pyx":66
 *     print('    --ref-index, folders containin pre-indexed files, if applicable. For certain versions of human genome, the indexed files are availabel from https://github.com/mills-lab/svelter.')
 * def print_default_parameters_nullmodel():
 *     print('SVelter NullModel')             # <<<<<<<<<<<<<<
 *     print('')
 *     print('Required Parameters:')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__57, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 66, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":67
 * def print_default_parameters_nullmodel():
 *     print('SVelter NullModel')
 *     print('')             # <<<<<<<<<<<<<<
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__58, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 67, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":68
 *     print('SVelter NullModel')
 *     print('')
 *     print('Required Parameters:')             # <<<<<<<<<<<<<<
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__59, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 68, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":69
 *     print('')
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')             # <<<<<<<<<<<<<<
 *     print('    --workdir, writable working directory.')
 *     print('')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__60, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 69, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":70
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')             # <<<<<<<<<<<<<<
 *     print('')
 *     print('Optional Parameters:')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__61, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 70, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":71
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')
 *     print('')             # <<<<<<<<<<<<<<
 *     print('Optional Parameters:')
 *     print('    --chromosome, name of chromosome to run. should match chromosome name in bam file')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__62, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 71, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":72
 *     print('    --workdir, writable working directory.')
 *     print('')
 *     print('Optional Parameters:')             # <<<<<<<<<<<<<<
 *     print('    --chromosome, name of chromosome to run. should match chromosome name in bam file')
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__63, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 72, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":73
 *     print('')
 *     print('Optional Parameters:')
 *     print('    --chromosome, name of chromosome to run. should match chromosome name in bam file')             # <<<<<<<<<<<<<<
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')
 *     print('    --null-copyneutral-length, minimum length requirement for --copyneutral regions used to build null model [2000]')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__64, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 73, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":74
 *     print('Optional Parameters:')
 *     print('    --chromosome, name of chromosome to run. should match chromosome name in bam file')
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')             # <<<<<<<<<<<<<<
 *     print('    --null-copyneutral-length, minimum length requirement for --copyneutral regions used to build null model [2000]')
 *     print('    --null-copyneutral-perc, percentage of regions from --copyneutral to utilize [0.1]')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__65, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 74, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":75
 *     print('    --chromosome, name of chromosome to run. should match chromosome name in bam file')
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')
 *     print('    --null-copyneutral-length, minimum length requirement for --copyneutral regions used to build null model [2000]')             # <<<<<<<<<<<<<<
 *     print('    --null-copyneutral-perc, percentage of regions from --copyneutral to utilize [0.1]')
 *     print('    --null-random-length, specify the length of random regions if --copyneutral parameter not used [5000]')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__66, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 75, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":76
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')
 *     print('    --null-copyneutral-length, minimum length requirement for --copyneutral regions used to build null model [2000]')
 *     print('    --null-copyneutral-perc, percentage of regions from --copyneutral to utilize [0.1]')             # <<<<<<<<<<<<<<
 *     print('    --null-random-length, specify the length of random regions if --copyneutral parameter not used [5000]')
 *     print('    --null-random-num, specify the number of random regions if --copyneutral parameter not used [10000]')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__67, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 76, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":77
 *     print('    --null-copyneutral-length, minimum length requirement for --copyneutral regions used to build null model [2000]')
 *     print('    --null-copyneutral-perc, percentage of regions from --copyneutral to utilize [0.1]')
 *     print('    --null-random-length, specify the length of random regions if --copyneutral parameter not used [5000]')             # <<<<<<<<<<<<<<
 *     print('    --null-random-num, specify the number of random regions if --copyneutral parameter not used [10000]')
 *     print('    --split-min-len, the minumum length of clip read considered as split; [10% of read length]')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__68, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 77, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":78
 *     print('    --null-copyneutral-perc, percentage of regions from --copyneutral to utilize [0.1]')
 *     print('    --null-random-length, specify the length of random regions if --copyneutral parameter not used [5000]')
 *     print('    --null-random-num, specify the number of random regions if --copyneutral parameter not used [10000]')             # <<<<<<<<<<<<<<
 *     print('    --split-min-len, the minumum length of clip read considered as split; [10% of read length]')
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20]')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__69, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 78, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":79
 *     print('    --null-random-length, specify the length of random regions if --copyneutral parameter not used [5000]')
 *     print('    --null-random-num, specify the number of random regions if --copyneutral parameter not used [10000]')
 *     print('    --split-min-len, the minumum length of clip read considered as split; [10% of read length]')             # <<<<<<<<<<<<<<
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20]')
 *     print('    --qc-split, minimum alighment of clipped parts of reads considered as a soft clip [20]')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__70, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 79, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":80
 *     print('    --null-random-num, specify the number of random regions if --copyneutral parameter not used [10000]')
 *     print('    --split-min-len, the minumum length of clip read considered as split; [10% of read length]')
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20]')             # <<<<<<<<<<<<<<
 *     print('    --qc-split, minimum alighment of clipped parts of reads considered as a soft clip [20]')
 * def print_default_parameters_bpsearch():
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__71, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 80, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":81
 *     print('    --split-min-len, the minumum length of clip read considered as split; [10% of read length]')
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20]')
 *     print('    --qc-split, minimum alighment of clipped parts of reads considered as a soft clip [20]')             # <<<<<<<<<<<<<<
 * def print_default_parameters_bpsearch():
 *     print('SVelter  BPSearch')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__72, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 81, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":65
 *     #print '    --segdup, absolute path of bed file indicating segmental duplications in genome, that would be excluded from analysis'
 *     print('    --ref-index, folders containin pre-indexed files, if applicable. For certain versions of human genome, the indexed files are availabel from https://github.com/mills-lab/svelter.')
 * def print_default_parameters_nullmodel():             # <<<<<<<<<<<<<<
 *     print('SVelter NullModel')
 *     print('')
 */

  /* function exit code */
  __pyx_r = Py_None; __Pyx_INCREF(Py_None);
  goto __pyx_L0;
  __pyx_L1_error:;
  __Pyx_XDECREF(__pyx_t_1);
  __Pyx_AddTraceback("svelter_sv.readme.print_default_parameters_nullmodel", __pyx_clineno, __pyx_lineno, __pyx_filename);
  __pyx_r = NULL;
  __pyx_L0:;
  __Pyx_XGIVEREF(__pyx_r);
  __Pyx_RefNannyFinishContext();
  return __pyx_r;
}

/* "svelter_sv/readme.pyx":82
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20]')
 *     print('    --qc-split, minimum alighment of clipped parts of reads considered as a soft clip [20]')
 * def print_default_parameters_bpsearch():             # <<<<<<<<<<<<<<
 *     print('SVelter  BPSearch')
 *     print('')
 */

/* Python wrapper */
static PyObject *__pyx_pw_10svelter_sv_6readme_11print_default_parameters_bpsearch(PyObject *__pyx_self, CYTHON_UNUSED PyObject *unused); /*proto*/
static PyMethodDef __pyx_mdef_10svelter_sv_6readme_11print_default_parameters_bpsearch = {"print_default_parameters_bpsearch", (PyCFunction)__pyx_pw_10svelter_sv_6readme_11print_default_parameters_bpsearch, METH_NOARGS, 0};
static PyObject *__pyx_pw_10svelter_sv_6readme_11print_default_parameters_bpsearch(PyObject *__pyx_self, CYTHON_UNUSED PyObject *unused) {
  PyObject *__pyx_r = 0;
  __Pyx_RefNannyDeclarations
  __Pyx_RefNannySetupContext("print_default_parameters_bpsearch (wrapper)", 0);
  __pyx_r = __pyx_pf_10svelter_sv_6readme_10print_default_parameters_bpsearch(__pyx_self);

  /* function exit code */
  __Pyx_RefNannyFinishContext();
  return __pyx_r;
}

static PyObject *__pyx_pf_10svelter_sv_6readme_10print_default_parameters_bpsearch(CYTHON_UNUSED PyObject *__pyx_self) {
  PyObject *__pyx_r = NULL;
  __Pyx_RefNannyDeclarations
  PyObject *__pyx_t_1 = NULL;
  __Pyx_RefNannySetupContext("print_default_parameters_bpsearch", 0);

  /* "svelter_sv/readme.pyx":83
 *     print('    --qc-split, minimum alighment of clipped parts of reads considered as a soft clip [20]')
 * def print_default_parameters_bpsearch():
 *     print('SVelter  BPSearch')             # <<<<<<<<<<<<<<
 *     print('')
 *     print('Required Parameters:')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__73, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 83, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":84
 * def print_default_parameters_bpsearch():
 *     print('SVelter  BPSearch')
 *     print('')             # <<<<<<<<<<<<<<
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__74, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 84, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":85
 *     print('SVelter  BPSearch')
 *     print('')
 *     print('Required Parameters:')             # <<<<<<<<<<<<<<
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__75, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 85, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":86
 *     print('')
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')             # <<<<<<<<<<<<<<
 *     print('    --workdir, writable working directory.')
 *     print('')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__76, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 86, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":87
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')             # <<<<<<<<<<<<<<
 *     print('')
 *     print('Optional Parameters:')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__77, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 87, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":88
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')
 *     print('')             # <<<<<<<<<<<<<<
 *     print('Optional Parameters:')
 *     print('    --chromosome, name of chromosome to run. should match chromosome name in bam file')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__78, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 88, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":89
 *     print('    --workdir, writable working directory.')
 *     print('')
 *     print('Optional Parameters:')             # <<<<<<<<<<<<<<
 *     print('    --chromosome, name of chromosome to run. should match chromosome name in bam file')
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__79, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 89, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":90
 *     print('')
 *     print('Optional Parameters:')
 *     print('    --chromosome, name of chromosome to run. should match chromosome name in bam file')             # <<<<<<<<<<<<<<
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20)')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__80, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 90, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":91
 *     print('Optional Parameters:')
 *     print('    --chromosome, name of chromosome to run. should match chromosome name in bam file')
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')             # <<<<<<<<<<<<<<
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20)')
 *     print('    --qc-split, minimum alighment of clipped parts of reads considered as a soft clip [20)')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__81, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 91, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":92
 *     print('    --chromosome, name of chromosome to run. should match chromosome name in bam file')
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20)')             # <<<<<<<<<<<<<<
 *     print('    --qc-split, minimum alighment of clipped parts of reads considered as a soft clip [20)')
 *     print('    --split-min-len, the minumum length of clip read considered as split; (default:10% of read length)')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__82, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 92, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":93
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20)')
 *     print('    --qc-split, minimum alighment of clipped parts of reads considered as a soft clip [20)')             # <<<<<<<<<<<<<<
 *     print('    --split-min-len, the minumum length of clip read considered as split; (default:10% of read length)')
 *     print('    --qc-map-tool, the tool extracts mappability information from a bigWig file,avaliable from: http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/bigWigSummary')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__83, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 93, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":94
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20)')
 *     print('    --qc-split, minimum alighment of clipped parts of reads considered as a soft clip [20)')
 *     print('    --split-min-len, the minumum length of clip read considered as split; (default:10% of read length)')             # <<<<<<<<<<<<<<
 *     print('    --qc-map-tool, the tool extracts mappability information from a bigWig file,avaliable from: http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/bigWigSummary')
 *     print('    --qc-map-file, .bigWig file used to decide local genomic mappability, avaliable from: ftp://hgdownload.cse.ucsc.edu/goldenPath/currentGenomes/Homo_sapiens/encodeDCC/wgEncodeMapability/')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__84, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 94, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":95
 *     print('    --qc-split, minimum alighment of clipped parts of reads considered as a soft clip [20)')
 *     print('    --split-min-len, the minumum length of clip read considered as split; (default:10% of read length)')
 *     print('    --qc-map-tool, the tool extracts mappability information from a bigWig file,avaliable from: http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/bigWigSummary')             # <<<<<<<<<<<<<<
 *     print('    --qc-map-file, .bigWig file used to decide local genomic mappability, avaliable from: ftp://hgdownload.cse.ucsc.edu/goldenPath/currentGenomes/Homo_sapiens/encodeDCC/wgEncodeMapability/')
 *     print('    --qc-map-cutoff, the minimum mapping quality required for a breakpoint to be reported [0.0)')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__85, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 95, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":96
 *     print('    --split-min-len, the minumum length of clip read considered as split; (default:10% of read length)')
 *     print('    --qc-map-tool, the tool extracts mappability information from a bigWig file,avaliable from: http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/bigWigSummary')
 *     print('    --qc-map-file, .bigWig file used to decide local genomic mappability, avaliable from: ftp://hgdownload.cse.ucsc.edu/goldenPath/currentGenomes/Homo_sapiens/encodeDCC/wgEncodeMapability/')             # <<<<<<<<<<<<<<
 *     print('    --qc-map-cutoff, the minimum mapping quality required for a breakpoint to be reported [0.0)')
 * def print_default_parameters_bpintegrate():
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__86, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 96, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":97
 *     print('    --qc-map-tool, the tool extracts mappability information from a bigWig file,avaliable from: http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/bigWigSummary')
 *     print('    --qc-map-file, .bigWig file used to decide local genomic mappability, avaliable from: ftp://hgdownload.cse.ucsc.edu/goldenPath/currentGenomes/Homo_sapiens/encodeDCC/wgEncodeMapability/')
 *     print('    --qc-map-cutoff, the minimum mapping quality required for a breakpoint to be reported [0.0)')             # <<<<<<<<<<<<<<
 * def print_default_parameters_bpintegrate():
 *     print('SVelter  BPIntegrate')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__87, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 97, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":82
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20]')
 *     print('    --qc-split, minimum alighment of clipped parts of reads considered as a soft clip [20]')
 * def print_default_parameters_bpsearch():             # <<<<<<<<<<<<<<
 *     print('SVelter  BPSearch')
 *     print('')
 */

  /* function exit code */
  __pyx_r = Py_None; __Pyx_INCREF(Py_None);
  goto __pyx_L0;
  __pyx_L1_error:;
  __Pyx_XDECREF(__pyx_t_1);
  __Pyx_AddTraceback("svelter_sv.readme.print_default_parameters_bpsearch", __pyx_clineno, __pyx_lineno, __pyx_filename);
  __pyx_r = NULL;
  __pyx_L0:;
  __Pyx_XGIVEREF(__pyx_r);
  __Pyx_RefNannyFinishContext();
  return __pyx_r;
}

/* "svelter_sv/readme.pyx":98
 *     print('    --qc-map-file, .bigWig file used to decide local genomic mappability, avaliable from: ftp://hgdownload.cse.ucsc.edu/goldenPath/currentGenomes/Homo_sapiens/encodeDCC/wgEncodeMapability/')
 *     print('    --qc-map-cutoff, the minimum mapping quality required for a breakpoint to be reported [0.0)')
 * def print_default_parameters_bpintegrate():             # <<<<<<<<<<<<<<
 *     print('SVelter  BPIntegrate')
 *     print('')
 */

/* Python wrapper */
static PyObject *__pyx_pw_10svelter_sv_6readme_13print_default_parameters_bpintegrate(PyObject *__pyx_self, CYTHON_UNUSED PyObject *unused); /*proto*/
static PyMethodDef __pyx_mdef_10svelter_sv_6readme_13print_default_parameters_bpintegrate = {"print_default_parameters_bpintegrate", (PyCFunction)__pyx_pw_10svelter_sv_6readme_13print_default_parameters_bpintegrate, METH_NOARGS, 0};
static PyObject *__pyx_pw_10svelter_sv_6readme_13print_default_parameters_bpintegrate(PyObject *__pyx_self, CYTHON_UNUSED PyObject *unused) {
  PyObject *__pyx_r = 0;
  __Pyx_RefNannyDeclarations
  __Pyx_RefNannySetupContext("print_default_parameters_bpintegrate (wrapper)", 0);
  __pyx_r = __pyx_pf_10svelter_sv_6readme_12print_default_parameters_bpintegrate(__pyx_self);

  /* function exit code */
  __Pyx_RefNannyFinishContext();
  return __pyx_r;
}

static PyObject *__pyx_pf_10svelter_sv_6readme_12print_default_parameters_bpintegrate(CYTHON_UNUSED PyObject *__pyx_self) {
  PyObject *__pyx_r = NULL;
  __Pyx_RefNannyDeclarations
  PyObject *__pyx_t_1 = NULL;
  __Pyx_RefNannySetupContext("print_default_parameters_bpintegrate", 0);

  /* "svelter_sv/readme.pyx":99
 *     print('    --qc-map-cutoff, the minimum mapping quality required for a breakpoint to be reported [0.0)')
 * def print_default_parameters_bpintegrate():
 *     print('SVelter  BPIntegrate')             # <<<<<<<<<<<<<<
 *     print('')
 *     print('Required Parameters:')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__88, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 99, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":100
 * def print_default_parameters_bpintegrate():
 *     print('SVelter  BPIntegrate')
 *     print('')             # <<<<<<<<<<<<<<
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__89, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 100, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":101
 *     print('SVelter  BPIntegrate')
 *     print('')
 *     print('Required Parameters:')             # <<<<<<<<<<<<<<
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__90, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 101, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":102
 *     print('')
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')             # <<<<<<<<<<<<<<
 *     print('    --workdir, writable working directory.')
 *     print('')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__91, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 102, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":103
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')             # <<<<<<<<<<<<<<
 *     print('')
 *     print('Optional Parameters:')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__92, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 103, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":104
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')
 *     print('')             # <<<<<<<<<<<<<<
 *     print('Optional Parameters:')
 *     print('    --batch, specify number of structures in each separate file (if 0, output files will be calssified by chromosomes; default, all BP clustered will be integrated in one txt file)')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__93, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 104, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":105
 *     print('    --workdir, writable working directory.')
 *     print('')
 *     print('Optional Parameters:')             # <<<<<<<<<<<<<<
 *     print('    --batch, specify number of structures in each separate file (if 0, output files will be calssified by chromosomes; default, all BP clustered will be integrated in one txt file)')
 *     print('    --bp-path, specify the path where all called breakpoints were kept under')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__94, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 105, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":106
 *     print('')
 *     print('Optional Parameters:')
 *     print('    --batch, specify number of structures in each separate file (if 0, output files will be calssified by chromosomes; default, all BP clustered will be integrated in one txt file)')             # <<<<<<<<<<<<<<
 *     print('    --bp-path, specify the path where all called breakpoints were kept under')
 *     print('    --chromosome, name of chromosome to run. should match chromosome name in bam file')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__95, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 106, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":107
 *     print('Optional Parameters:')
 *     print('    --batch, specify number of structures in each separate file (if 0, output files will be calssified by chromosomes; default, all BP clustered will be integrated in one txt file)')
 *     print('    --bp-path, specify the path where all called breakpoints were kept under')             # <<<<<<<<<<<<<<
 *     print('    --chromosome, name of chromosome to run. should match chromosome name in bam file')
 * def print_default_parameters_svpredict():
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__96, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 107, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":108
 *     print('    --batch, specify number of structures in each separate file (if 0, output files will be calssified by chromosomes; default, all BP clustered will be integrated in one txt file)')
 *     print('    --bp-path, specify the path where all called breakpoints were kept under')
 *     print('    --chromosome, name of chromosome to run. should match chromosome name in bam file')             # <<<<<<<<<<<<<<
 * def print_default_parameters_svpredict():
 *     print('SVelter  SVPredict')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__97, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 108, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":98
 *     print('    --qc-map-file, .bigWig file used to decide local genomic mappability, avaliable from: ftp://hgdownload.cse.ucsc.edu/goldenPath/currentGenomes/Homo_sapiens/encodeDCC/wgEncodeMapability/')
 *     print('    --qc-map-cutoff, the minimum mapping quality required for a breakpoint to be reported [0.0)')
 * def print_default_parameters_bpintegrate():             # <<<<<<<<<<<<<<
 *     print('SVelter  BPIntegrate')
 *     print('')
 */

  /* function exit code */
  __pyx_r = Py_None; __Pyx_INCREF(Py_None);
  goto __pyx_L0;
  __pyx_L1_error:;
  __Pyx_XDECREF(__pyx_t_1);
  __Pyx_AddTraceback("svelter_sv.readme.print_default_parameters_bpintegrate", __pyx_clineno, __pyx_lineno, __pyx_filename);
  __pyx_r = NULL;
  __pyx_L0:;
  __Pyx_XGIVEREF(__pyx_r);
  __Pyx_RefNannyFinishContext();
  return __pyx_r;
}

/* "svelter_sv/readme.pyx":109
 *     print('    --bp-path, specify the path where all called breakpoints were kept under')
 *     print('    --chromosome, name of chromosome to run. should match chromosome name in bam file')
 * def print_default_parameters_svpredict():             # <<<<<<<<<<<<<<
 *     print('SVelter  SVPredict')
 *     print('Required Parameters:')
 */

/* Python wrapper */
static PyObject *__pyx_pw_10svelter_sv_6readme_15print_default_parameters_svpredict(PyObject *__pyx_self, CYTHON_UNUSED PyObject *unused); /*proto*/
static PyMethodDef __pyx_mdef_10svelter_sv_6readme_15print_default_parameters_svpredict = {"print_default_parameters_svpredict", (PyCFunction)__pyx_pw_10svelter_sv_6readme_15print_default_parameters_svpredict, METH_NOARGS, 0};
static PyObject *__pyx_pw_10svelter_sv_6readme_15print_default_parameters_svpredict(PyObject *__pyx_self, CYTHON_UNUSED PyObject *unused) {
  PyObject *__pyx_r = 0;
  __Pyx_RefNannyDeclarations
  __Pyx_RefNannySetupContext("print_default_parameters_svpredict (wrapper)", 0);
  __pyx_r = __pyx_pf_10svelter_sv_6readme_14print_default_parameters_svpredict(__pyx_self);

  /* function exit code */
  __Pyx_RefNannyFinishContext();
  return __pyx_r;
}

static PyObject *__pyx_pf_10svelter_sv_6readme_14print_default_parameters_svpredict(CYTHON_UNUSED PyObject *__pyx_self) {
  PyObject *__pyx_r = NULL;
  __Pyx_RefNannyDeclarations
  PyObject *__pyx_t_1 = NULL;
  __Pyx_RefNannySetupContext("print_default_parameters_svpredict", 0);

  /* "svelter_sv/readme.pyx":110
 *     print('    --chromosome, name of chromosome to run. should match chromosome name in bam file')
 * def print_default_parameters_svpredict():
 *     print('SVelter  SVPredict')             # <<<<<<<<<<<<<<
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__98, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 110, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":111
 * def print_default_parameters_svpredict():
 *     print('SVelter  SVPredict')
 *     print('Required Parameters:')             # <<<<<<<<<<<<<<
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__99, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 111, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":112
 *     print('SVelter  SVPredict')
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')             # <<<<<<<<<<<<<<
 *     print('    --workdir, writable working directory.')
 *     print('    --bp-file, input txt file containing clustered bps.')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__100, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 112, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":113
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')             # <<<<<<<<<<<<<<
 *     print('    --bp-file, input txt file containing clustered bps.')
 *     print(' ')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__101, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 113, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":114
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')
 *     print('    --bp-file, input txt file containing clustered bps.')             # <<<<<<<<<<<<<<
 *     print(' ')
 *     print('Optional Parameters:')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__102, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 114, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":115
 *     print('    --workdir, writable working directory.')
 *     print('    --bp-file, input txt file containing clustered bps.')
 *     print(' ')             # <<<<<<<<<<<<<<
 *     print('Optional Parameters:')
 *     print('    --num-iteration, maximum number of iterations per structure will run')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__103, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 115, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":116
 *     print('    --bp-file, input txt file containing clustered bps.')
 *     print(' ')
 *     print('Optional Parameters:')             # <<<<<<<<<<<<<<
 *     print('    --num-iteration, maximum number of iterations per structure will run')
 *     print('    --ploidy, limit algorithm to specific zygosity (0:heterozygous only; 1:homozygous only; 2:both; default:2)')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__104, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 116, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":117
 *     print(' ')
 *     print('Optional Parameters:')
 *     print('    --num-iteration, maximum number of iterations per structure will run')             # <<<<<<<<<<<<<<
 *     print('    --ploidy, limit algorithm to specific zygosity (0:heterozygous only; 1:homozygous only; 2:both; default:2)')
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__105, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 117, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":118
 *     print('Optional Parameters:')
 *     print('    --num-iteration, maximum number of iterations per structure will run')
 *     print('    --ploidy, limit algorithm to specific zygosity (0:heterozygous only; 1:homozygous only; 2:both; default:2)')             # <<<<<<<<<<<<<<
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20)')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__106, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 118, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":119
 *     print('    --num-iteration, maximum number of iterations per structure will run')
 *     print('    --ploidy, limit algorithm to specific zygosity (0:heterozygous only; 1:homozygous only; 2:both; default:2)')
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')             # <<<<<<<<<<<<<<
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20)')
 * def print_default_parameters_svintegrate():
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__107, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 119, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":120
 *     print('    --ploidy, limit algorithm to specific zygosity (0:heterozygous only; 1:homozygous only; 2:both; default:2)')
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20)')             # <<<<<<<<<<<<<<
 * def print_default_parameters_svintegrate():
 *     print('SVelter SVIntegrate')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__108, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 120, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":109
 *     print('    --bp-path, specify the path where all called breakpoints were kept under')
 *     print('    --chromosome, name of chromosome to run. should match chromosome name in bam file')
 * def print_default_parameters_svpredict():             # <<<<<<<<<<<<<<
 *     print('SVelter  SVPredict')
 *     print('Required Parameters:')
 */

  /* function exit code */
  __pyx_r = Py_None; __Pyx_INCREF(Py_None);
  goto __pyx_L0;
  __pyx_L1_error:;
  __Pyx_XDECREF(__pyx_t_1);
  __Pyx_AddTraceback("svelter_sv.readme.print_default_parameters_svpredict", __pyx_clineno, __pyx_lineno, __pyx_filename);
  __pyx_r = NULL;
  __pyx_L0:;
  __Pyx_XGIVEREF(__pyx_r);
  __Pyx_RefNannyFinishContext();
  return __pyx_r;
}

/* "svelter_sv/readme.pyx":121
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20)')
 * def print_default_parameters_svintegrate():             # <<<<<<<<<<<<<<
 *     print('SVelter SVIntegrate')
 *     print(' ')
 */

/* Python wrapper */
static PyObject *__pyx_pw_10svelter_sv_6readme_17print_default_parameters_svintegrate(PyObject *__pyx_self, CYTHON_UNUSED PyObject *unused); /*proto*/
static PyMethodDef __pyx_mdef_10svelter_sv_6readme_17print_default_parameters_svintegrate = {"print_default_parameters_svintegrate", (PyCFunction)__pyx_pw_10svelter_sv_6readme_17print_default_parameters_svintegrate, METH_NOARGS, 0};
static PyObject *__pyx_pw_10svelter_sv_6readme_17print_default_parameters_svintegrate(PyObject *__pyx_self, CYTHON_UNUSED PyObject *unused) {
  PyObject *__pyx_r = 0;
  __Pyx_RefNannyDeclarations
  __Pyx_RefNannySetupContext("print_default_parameters_svintegrate (wrapper)", 0);
  __pyx_r = __pyx_pf_10svelter_sv_6readme_16print_default_parameters_svintegrate(__pyx_self);

  /* function exit code */
  __Pyx_RefNannyFinishContext();
  return __pyx_r;
}

static PyObject *__pyx_pf_10svelter_sv_6readme_16print_default_parameters_svintegrate(CYTHON_UNUSED PyObject *__pyx_self) {
  PyObject *__pyx_r = NULL;
  __Pyx_RefNannyDeclarations
  PyObject *__pyx_t_1 = NULL;
  __Pyx_RefNannySetupContext("print_default_parameters_svintegrate", 0);

  /* "svelter_sv/readme.pyx":122
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20)')
 * def print_default_parameters_svintegrate():
 *     print('SVelter SVIntegrate')             # <<<<<<<<<<<<<<
 *     print(' ')
 *     print('Required Parameters:')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__109, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 122, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":123
 * def print_default_parameters_svintegrate():
 *     print('SVelter SVIntegrate')
 *     print(' ')             # <<<<<<<<<<<<<<
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__110, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 123, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":124
 *     print('SVelter SVIntegrate')
 *     print(' ')
 *     print('Required Parameters:')             # <<<<<<<<<<<<<<
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__111, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 124, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":125
 *     print(' ')
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')             # <<<<<<<<<<<<<<
 *     print('    --workdir, writable working directory.')
 *     print('    --input-path, path of .coverage files produced by SVelter SVPredict')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__112, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 125, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":126
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')             # <<<<<<<<<<<<<<
 *     print('    --input-path, path of .coverage files produced by SVelter SVPredict')
 *     print(' ')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__113, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 126, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":127
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')
 *     print('    --input-path, path of .coverage files produced by SVelter SVPredict')             # <<<<<<<<<<<<<<
 *     print(' ')
 *     print('Optional Parameters:')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__114, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 127, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":128
 *     print('    --workdir, writable working directory.')
 *     print('    --input-path, path of .coverage files produced by SVelter SVPredict')
 *     print(' ')             # <<<<<<<<<<<<<<
 *     print('Optional Parameters:')
 *     print('    --prefix, output prefix for vcf and svelter files')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__115, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 128, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":129
 *     print('    --input-path, path of .coverage files produced by SVelter SVPredict')
 *     print(' ')
 *     print('Optional Parameters:')             # <<<<<<<<<<<<<<
 *     print('    --prefix, output prefix for vcf and svelter files')
 *     print('    --input-path, path of .coverage files produced by SVelter SVPredict')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__116, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 129, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":130
 *     print(' ')
 *     print('Optional Parameters:')
 *     print('    --prefix, output prefix for vcf and svelter files')             # <<<<<<<<<<<<<<
 *     print('    --input-path, path of .coverage files produced by SVelter SVPredict')
 *     print('    --qc-structure, minimum quality score of a resolved structure to be considered as PASS and included in the output vcf file')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__117, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 130, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":131
 *     print('Optional Parameters:')
 *     print('    --prefix, output prefix for vcf and svelter files')
 *     print('    --input-path, path of .coverage files produced by SVelter SVPredict')             # <<<<<<<<<<<<<<
 *     print('    --qc-structure, minimum quality score of a resolved structure to be considered as PASS and included in the output vcf file')
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__118, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 131, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":132
 *     print('    --prefix, output prefix for vcf and svelter files')
 *     print('    --input-path, path of .coverage files produced by SVelter SVPredict')
 *     print('    --qc-structure, minimum quality score of a resolved structure to be considered as PASS and included in the output vcf file')             # <<<<<<<<<<<<<<
 */
  __pyx_t_1 = __Pyx_PyObject_Call(__pyx_builtin_print, __pyx_tuple__119, NULL); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 132, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":121
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20)')
 * def print_default_parameters_svintegrate():             # <<<<<<<<<<<<<<
 *     print('SVelter SVIntegrate')
 *     print(' ')
 */

  /* function exit code */
  __pyx_r = Py_None; __Pyx_INCREF(Py_None);
  goto __pyx_L0;
  __pyx_L1_error:;
  __Pyx_XDECREF(__pyx_t_1);
  __Pyx_AddTraceback("svelter_sv.readme.print_default_parameters_svintegrate", __pyx_clineno, __pyx_lineno, __pyx_filename);
  __pyx_r = NULL;
  __pyx_L0:;
  __Pyx_XGIVEREF(__pyx_r);
  __Pyx_RefNannyFinishContext();
  return __pyx_r;
}

static PyMethodDef __pyx_methods[] = {
  {0, 0, 0, 0}
};

#if PY_MAJOR_VERSION >= 3
static struct PyModuleDef __pyx_moduledef = {
  #if PY_VERSION_HEX < 0x03020000
    { PyObject_HEAD_INIT(NULL) NULL, 0, NULL },
  #else
    PyModuleDef_HEAD_INIT,
  #endif
    "readme",
    0, /* m_doc */
    -1, /* m_size */
    __pyx_methods /* m_methods */,
    NULL, /* m_reload */
    NULL, /* m_traverse */
    NULL, /* m_clear */
    NULL /* m_free */
};
#endif

static __Pyx_StringTabEntry __pyx_string_tab[] = {
  {&__pyx_kp_s_BPIntegrate, __pyx_k_BPIntegrate, sizeof(__pyx_k_BPIntegrate), 0, 0, 1, 0},
  {&__pyx_kp_s_BPSearch, __pyx_k_BPSearch, sizeof(__pyx_k_BPSearch), 0, 0, 1, 0},
  {&__pyx_kp_s_Clean, __pyx_k_Clean, sizeof(__pyx_k_Clean), 0, 0, 1, 0},
  {&__pyx_kp_s_NullModel, __pyx_k_NullModel, sizeof(__pyx_k_NullModel), 0, 0, 1, 0},
  {&__pyx_kp_s_Optional_Parameters, __pyx_k_Optional_Parameters, sizeof(__pyx_k_Optional_Parameters), 0, 0, 1, 0},
  {&__pyx_kp_s_Options, __pyx_k_Options, sizeof(__pyx_k_Options), 0, 0, 1, 0},
  {&__pyx_kp_s_Required_Parameters, __pyx_k_Required_Parameters, sizeof(__pyx_k_Required_Parameters), 0, 0, 1, 0},
  {&__pyx_kp_s_SVIntegrate, __pyx_k_SVIntegrate, sizeof(__pyx_k_SVIntegrate), 0, 0, 1, 0},
  {&__pyx_kp_s_SVPredict, __pyx_k_SVPredict, sizeof(__pyx_k_SVPredict), 0, 0, 1, 0},
  {&__pyx_kp_s_SVelter_0_1_Contact_xuefzhao_umi, __pyx_k_SVelter_0_1_Contact_xuefzhao_umi, sizeof(__pyx_k_SVelter_0_1_Contact_xuefzhao_umi), 0, 0, 1, 0},
  {&__pyx_kp_s_SVelter_BPIntegrate, __pyx_k_SVelter_BPIntegrate, sizeof(__pyx_k_SVelter_BPIntegrate), 0, 0, 1, 0},
  {&__pyx_kp_s_SVelter_BPSearch, __pyx_k_SVelter_BPSearch, sizeof(__pyx_k_SVelter_BPSearch), 0, 0, 1, 0},
  {&__pyx_kp_s_SVelter_Clean, __pyx_k_SVelter_Clean, sizeof(__pyx_k_SVelter_Clean), 0, 0, 1, 0},
  {&__pyx_kp_s_SVelter_NullModel, __pyx_k_SVelter_NullModel, sizeof(__pyx_k_SVelter_NullModel), 0, 0, 1, 0},
  {&__pyx_kp_s_SVelter_PredefinedBP, __pyx_k_SVelter_PredefinedBP, sizeof(__pyx_k_SVelter_PredefinedBP), 0, 0, 1, 0},
  {&__pyx_kp_s_SVelter_SVIntegrate, __pyx_k_SVelter_SVIntegrate, sizeof(__pyx_k_SVelter_SVIntegrate), 0, 0, 1, 0},
  {&__pyx_kp_s_SVelter_SVPredict, __pyx_k_SVelter_SVPredict, sizeof(__pyx_k_SVelter_SVPredict), 0, 0, 1, 0},
  {&__pyx_kp_s_SVelter_Setup, __pyx_k_SVelter_Setup, sizeof(__pyx_k_SVelter_Setup), 0, 0, 1, 0},
  {&__pyx_kp_s_SVelter_Setup_should_be_run_firs, __pyx_k_SVelter_Setup_should_be_run_firs, sizeof(__pyx_k_SVelter_Setup_should_be_run_firs), 0, 0, 1, 0},
  {&__pyx_kp_s_Setup, __pyx_k_Setup, sizeof(__pyx_k_Setup), 0, 0, 1, 0},
  {&__pyx_kp_s_Type_in_SVelter_options_for_deta, __pyx_k_Type_in_SVelter_options_for_deta, sizeof(__pyx_k_Type_in_SVelter_options_for_deta), 0, 0, 1, 0},
  {&__pyx_kp_s_Usage_SVelter_Options_Parameters, __pyx_k_Usage_SVelter_Options_Parameters, sizeof(__pyx_k_Usage_SVelter_Options_Parameters), 0, 0, 1, 0},
  {&__pyx_kp_s__2, __pyx_k__2, sizeof(__pyx_k__2), 0, 0, 1, 0},
  {&__pyx_kp_s__5, __pyx_k__5, sizeof(__pyx_k__5), 0, 0, 1, 0},
  {&__pyx_kp_s_batch_specify_number_of_structu, __pyx_k_batch_specify_number_of_structu, sizeof(__pyx_k_batch_specify_number_of_structu), 0, 0, 1, 0},
  {&__pyx_kp_s_bp_file_input_txt_file_containi, __pyx_k_bp_file_input_txt_file_containi, sizeof(__pyx_k_bp_file_input_txt_file_containi), 0, 0, 1, 0},
  {&__pyx_kp_s_bp_path_specify_the_path_where, __pyx_k_bp_path_specify_the_path_where, sizeof(__pyx_k_bp_path_specify_the_path_where), 0, 0, 1, 0},
  {&__pyx_kp_s_chromosome_name_of_chromosome_t, __pyx_k_chromosome_name_of_chromosome_t, sizeof(__pyx_k_chromosome_name_of_chromosome_t), 0, 0, 1, 0},
  {&__pyx_kp_s_data_talkowski_xuefang_local_sr, __pyx_k_data_talkowski_xuefang_local_sr, sizeof(__pyx_k_data_talkowski_xuefang_local_sr), 0, 0, 1, 0},
  {&__pyx_kp_s_ftp_hgdownload_cse_ucsc_edu_gol, __pyx_k_ftp_hgdownload_cse_ucsc_edu_gol, sizeof(__pyx_k_ftp_hgdownload_cse_ucsc_edu_gol), 0, 0, 1, 0},
  {&__pyx_kp_s_http_hgdownload_cse_ucsc_edu_ad, __pyx_k_http_hgdownload_cse_ucsc_edu_ad, sizeof(__pyx_k_http_hgdownload_cse_ucsc_edu_ad), 0, 0, 1, 0},
  {&__pyx_kp_s_input_bed_predefined_breakpoint, __pyx_k_input_bed_predefined_breakpoint, sizeof(__pyx_k_input_bed_predefined_breakpoint), 0, 0, 1, 0},
  {&__pyx_kp_s_input_path_path_of_coverage_fil, __pyx_k_input_path_path_of_coverage_fil, sizeof(__pyx_k_input_path_path_of_coverage_fil), 0, 0, 1, 0},
  {&__pyx_n_s_main, __pyx_k_main, sizeof(__pyx_k_main), 0, 0, 1, 1},
  {&__pyx_kp_s_null_copyneutral_length_minimum, __pyx_k_null_copyneutral_length_minimum, sizeof(__pyx_k_null_copyneutral_length_minimum), 0, 0, 1, 0},
  {&__pyx_kp_s_null_copyneutral_perc_percentag, __pyx_k_null_copyneutral_perc_percentag, sizeof(__pyx_k_null_copyneutral_perc_percentag), 0, 0, 1, 0},
  {&__pyx_kp_s_null_model_specify_which_stat_m, __pyx_k_null_model_specify_which_stat_m, sizeof(__pyx_k_null_model_specify_which_stat_m), 0, 0, 1, 0},
  {&__pyx_kp_s_null_random_length_specify_the, __pyx_k_null_random_length_specify_the, sizeof(__pyx_k_null_random_length_specify_the), 0, 0, 1, 0},
  {&__pyx_kp_s_null_random_num_specify_the_num, __pyx_k_null_random_num_specify_the_num, sizeof(__pyx_k_null_random_num_specify_the_num), 0, 0, 1, 0},
  {&__pyx_kp_s_num_iteration_maximum_number_of, __pyx_k_num_iteration_maximum_number_of, sizeof(__pyx_k_num_iteration_maximum_number_of), 0, 0, 1, 0},
  {&__pyx_kp_s_ploidy_limit_algorithm_to_speci, __pyx_k_ploidy_limit_algorithm_to_speci, sizeof(__pyx_k_ploidy_limit_algorithm_to_speci), 0, 0, 1, 0},
  {&__pyx_kp_s_prefix_output_prefix_for_vcf_an, __pyx_k_prefix_output_prefix_for_vcf_an, sizeof(__pyx_k_prefix_output_prefix_for_vcf_an), 0, 0, 1, 0},
  {&__pyx_kp_s_prefix_output_prefix_for_vcf_an_2, __pyx_k_prefix_output_prefix_for_vcf_an_2, sizeof(__pyx_k_prefix_output_prefix_for_vcf_an_2), 0, 0, 1, 0},
  {&__pyx_n_s_print, __pyx_k_print, sizeof(__pyx_k_print), 0, 0, 1, 1},
  {&__pyx_n_s_print_default_parameters, __pyx_k_print_default_parameters, sizeof(__pyx_k_print_default_parameters), 0, 0, 1, 1},
  {&__pyx_n_s_print_default_parameters_bpinteg, __pyx_k_print_default_parameters_bpinteg, sizeof(__pyx_k_print_default_parameters_bpinteg), 0, 0, 1, 1},
  {&__pyx_n_s_print_default_parameters_bpsearc, __pyx_k_print_default_parameters_bpsearc, sizeof(__pyx_k_print_default_parameters_bpsearc), 0, 0, 1, 1},
  {&__pyx_n_s_print_default_parameters_clean, __pyx_k_print_default_parameters_clean, sizeof(__pyx_k_print_default_parameters_clean), 0, 0, 1, 1},
  {&__pyx_n_s_print_default_parameters_nullmod, __pyx_k_print_default_parameters_nullmod, sizeof(__pyx_k_print_default_parameters_nullmod), 0, 0, 1, 1},
  {&__pyx_n_s_print_default_parameters_predefi, __pyx_k_print_default_parameters_predefi, sizeof(__pyx_k_print_default_parameters_predefi), 0, 0, 1, 1},
  {&__pyx_n_s_print_default_parameters_setup, __pyx_k_print_default_parameters_setup, sizeof(__pyx_k_print_default_parameters_setup), 0, 0, 1, 1},
  {&__pyx_n_s_print_default_parameters_svinteg, __pyx_k_print_default_parameters_svinteg, sizeof(__pyx_k_print_default_parameters_svinteg), 0, 0, 1, 1},
  {&__pyx_n_s_print_default_parameters_svpredi, __pyx_k_print_default_parameters_svpredi, sizeof(__pyx_k_print_default_parameters_svpredi), 0, 0, 1, 1},
  {&__pyx_kp_s_qc_align_minimum_alignment_qual, __pyx_k_qc_align_minimum_alignment_qual, sizeof(__pyx_k_qc_align_minimum_alignment_qual), 0, 0, 1, 0},
  {&__pyx_kp_s_qc_align_minimum_alignment_qual_2, __pyx_k_qc_align_minimum_alignment_qual_2, sizeof(__pyx_k_qc_align_minimum_alignment_qual_2), 0, 0, 1, 0},
  {&__pyx_kp_s_qc_map_cutoff_the_minimum_mappi, __pyx_k_qc_map_cutoff_the_minimum_mappi, sizeof(__pyx_k_qc_map_cutoff_the_minimum_mappi), 0, 0, 1, 0},
  {&__pyx_kp_s_qc_map_cutoff_the_minimum_mappi_2, __pyx_k_qc_map_cutoff_the_minimum_mappi_2, sizeof(__pyx_k_qc_map_cutoff_the_minimum_mappi_2), 0, 0, 1, 0},
  {&__pyx_kp_s_qc_map_file_bigWig_file_used_to, __pyx_k_qc_map_file_bigWig_file_used_to, sizeof(__pyx_k_qc_map_file_bigWig_file_used_to), 0, 0, 1, 0},
  {&__pyx_kp_s_qc_map_file_bigWig_file_used_to_2, __pyx_k_qc_map_file_bigWig_file_used_to_2, sizeof(__pyx_k_qc_map_file_bigWig_file_used_to_2), 0, 0, 1, 0},
  {&__pyx_kp_s_qc_map_tool_the_tool_extracts_m, __pyx_k_qc_map_tool_the_tool_extracts_m, sizeof(__pyx_k_qc_map_tool_the_tool_extracts_m), 0, 0, 1, 0},
  {&__pyx_kp_s_qc_map_tool_the_tool_extracts_m_2, __pyx_k_qc_map_tool_the_tool_extracts_m_2, sizeof(__pyx_k_qc_map_tool_the_tool_extracts_m_2), 0, 0, 1, 0},
  {&__pyx_kp_s_qc_split_minimum_alighment_of_c, __pyx_k_qc_split_minimum_alighment_of_c, sizeof(__pyx_k_qc_split_minimum_alighment_of_c), 0, 0, 1, 0},
  {&__pyx_kp_s_qc_split_minimum_alighment_of_c_2, __pyx_k_qc_split_minimum_alighment_of_c_2, sizeof(__pyx_k_qc_split_minimum_alighment_of_c_2), 0, 0, 1, 0},
  {&__pyx_kp_s_qc_structure_minimum_quality_sc, __pyx_k_qc_structure_minimum_quality_sc, sizeof(__pyx_k_qc_structure_minimum_quality_sc), 0, 0, 1, 0},
  {&__pyx_kp_s_ref_index_folders_containin_pre, __pyx_k_ref_index_folders_containin_pre, sizeof(__pyx_k_ref_index_folders_containin_pre), 0, 0, 1, 0},
  {&__pyx_kp_s_reference_absolute_path_of_refe, __pyx_k_reference_absolute_path_of_refe, sizeof(__pyx_k_reference_absolute_path_of_refe), 0, 0, 1, 0},
  {&__pyx_kp_s_sample_input_alignment_file_in, __pyx_k_sample_input_alignment_file_in, sizeof(__pyx_k_sample_input_alignment_file_in), 0, 0, 1, 0},
  {&__pyx_kp_s_split_min_len_the_minumum_lengt, __pyx_k_split_min_len_the_minumum_lengt, sizeof(__pyx_k_split_min_len_the_minumum_lengt), 0, 0, 1, 0},
  {&__pyx_kp_s_split_min_len_the_minumum_lengt_2, __pyx_k_split_min_len_the_minumum_lengt_2, sizeof(__pyx_k_split_min_len_the_minumum_lengt_2), 0, 0, 1, 0},
  {&__pyx_kp_s_support_folder_containing_all_s, __pyx_k_support_folder_containing_all_s, sizeof(__pyx_k_support_folder_containing_all_s), 0, 0, 1, 0},
  {&__pyx_n_s_svelter_sv_readme, __pyx_k_svelter_sv_readme, sizeof(__pyx_k_svelter_sv_readme), 0, 0, 1, 1},
  {&__pyx_n_s_test, __pyx_k_test, sizeof(__pyx_k_test), 0, 0, 1, 1},
  {&__pyx_kp_s_workdir_writable_working_direct, __pyx_k_workdir_writable_working_direct, sizeof(__pyx_k_workdir_writable_working_direct), 0, 0, 1, 0},
  {0, 0, 0, 0, 0, 0, 0}
};
static int __Pyx_InitCachedBuiltins(void) {
  __pyx_builtin_print = __Pyx_GetBuiltinName(__pyx_n_s_print); if (!__pyx_builtin_print) __PYX_ERR(0, 3, __pyx_L1_error)
  return 0;
  __pyx_L1_error:;
  return -1;
}

static int __Pyx_InitCachedConstants(void) {
  __Pyx_RefNannyDeclarations
  __Pyx_RefNannySetupContext("__Pyx_InitCachedConstants", 0);

  /* "svelter_sv/readme.pyx":3
 * from __future__ import print_function
 * def print_default_parameters():
 *     print('SVelter-0.1        Contact: xuefzhao@umich.edu      Last Update:2015-08-20')             # <<<<<<<<<<<<<<
 *     print('')
 *     print('Usage: SVelter [Options] [Parameters]')
 */
  __pyx_tuple_ = PyTuple_Pack(1, __pyx_kp_s_SVelter_0_1_Contact_xuefzhao_umi); if (unlikely(!__pyx_tuple_)) __PYX_ERR(0, 3, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple_);
  __Pyx_GIVEREF(__pyx_tuple_);

  /* "svelter_sv/readme.pyx":4
 * def print_default_parameters():
 *     print('SVelter-0.1        Contact: xuefzhao@umich.edu      Last Update:2015-08-20')
 *     print('')             # <<<<<<<<<<<<<<
 *     print('Usage: SVelter [Options] [Parameters]')
 *     print(' ')
 */
  __pyx_tuple__3 = PyTuple_Pack(1, __pyx_kp_s__2); if (unlikely(!__pyx_tuple__3)) __PYX_ERR(0, 4, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__3);
  __Pyx_GIVEREF(__pyx_tuple__3);

  /* "svelter_sv/readme.pyx":5
 *     print('SVelter-0.1        Contact: xuefzhao@umich.edu      Last Update:2015-08-20')
 *     print('')
 *     print('Usage: SVelter [Options] [Parameters]')             # <<<<<<<<<<<<<<
 *     print(' ')
 *     print('Options:')
 */
  __pyx_tuple__4 = PyTuple_Pack(1, __pyx_kp_s_Usage_SVelter_Options_Parameters); if (unlikely(!__pyx_tuple__4)) __PYX_ERR(0, 5, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__4);
  __Pyx_GIVEREF(__pyx_tuple__4);

  /* "svelter_sv/readme.pyx":6
 *     print('')
 *     print('Usage: SVelter [Options] [Parameters]')
 *     print(' ')             # <<<<<<<<<<<<<<
 *     print('Options:')
 *     print('    Setup')
 */
  __pyx_tuple__6 = PyTuple_Pack(1, __pyx_kp_s__5); if (unlikely(!__pyx_tuple__6)) __PYX_ERR(0, 6, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__6);
  __Pyx_GIVEREF(__pyx_tuple__6);

  /* "svelter_sv/readme.pyx":7
 *     print('Usage: SVelter [Options] [Parameters]')
 *     print(' ')
 *     print('Options:')             # <<<<<<<<<<<<<<
 *     print('    Setup')
 *     print('    Clean')
 */
  __pyx_tuple__7 = PyTuple_Pack(1, __pyx_kp_s_Options); if (unlikely(!__pyx_tuple__7)) __PYX_ERR(0, 7, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__7);
  __Pyx_GIVEREF(__pyx_tuple__7);

  /* "svelter_sv/readme.pyx":8
 *     print(' ')
 *     print('Options:')
 *     print('    Setup')             # <<<<<<<<<<<<<<
 *     print('    Clean')
 *     print('    NullModel')
 */
  __pyx_tuple__8 = PyTuple_Pack(1, __pyx_kp_s_Setup); if (unlikely(!__pyx_tuple__8)) __PYX_ERR(0, 8, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__8);
  __Pyx_GIVEREF(__pyx_tuple__8);

  /* "svelter_sv/readme.pyx":9
 *     print('Options:')
 *     print('    Setup')
 *     print('    Clean')             # <<<<<<<<<<<<<<
 *     print('    NullModel')
 *     print('    BPSearch')
 */
  __pyx_tuple__9 = PyTuple_Pack(1, __pyx_kp_s_Clean); if (unlikely(!__pyx_tuple__9)) __PYX_ERR(0, 9, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__9);
  __Pyx_GIVEREF(__pyx_tuple__9);

  /* "svelter_sv/readme.pyx":10
 *     print('    Setup')
 *     print('    Clean')
 *     print('    NullModel')             # <<<<<<<<<<<<<<
 *     print('    BPSearch')
 *     print('    BPIntegrate')
 */
  __pyx_tuple__10 = PyTuple_Pack(1, __pyx_kp_s_NullModel); if (unlikely(!__pyx_tuple__10)) __PYX_ERR(0, 10, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__10);
  __Pyx_GIVEREF(__pyx_tuple__10);

  /* "svelter_sv/readme.pyx":11
 *     print('    Clean')
 *     print('    NullModel')
 *     print('    BPSearch')             # <<<<<<<<<<<<<<
 *     print('    BPIntegrate')
 *     print('    SVPredict')
 */
  __pyx_tuple__11 = PyTuple_Pack(1, __pyx_kp_s_BPSearch); if (unlikely(!__pyx_tuple__11)) __PYX_ERR(0, 11, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__11);
  __Pyx_GIVEREF(__pyx_tuple__11);

  /* "svelter_sv/readme.pyx":12
 *     print('    NullModel')
 *     print('    BPSearch')
 *     print('    BPIntegrate')             # <<<<<<<<<<<<<<
 *     print('    SVPredict')
 *     print('    SVIntegrate')
 */
  __pyx_tuple__12 = PyTuple_Pack(1, __pyx_kp_s_BPIntegrate); if (unlikely(!__pyx_tuple__12)) __PYX_ERR(0, 12, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__12);
  __Pyx_GIVEREF(__pyx_tuple__12);

  /* "svelter_sv/readme.pyx":13
 *     print('    BPSearch')
 *     print('    BPIntegrate')
 *     print('    SVPredict')             # <<<<<<<<<<<<<<
 *     print('    SVIntegrate')
 *     print('')
 */
  __pyx_tuple__13 = PyTuple_Pack(1, __pyx_kp_s_SVPredict); if (unlikely(!__pyx_tuple__13)) __PYX_ERR(0, 13, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__13);
  __Pyx_GIVEREF(__pyx_tuple__13);

  /* "svelter_sv/readme.pyx":14
 *     print('    BPIntegrate')
 *     print('    SVPredict')
 *     print('    SVIntegrate')             # <<<<<<<<<<<<<<
 *     print('')
 *     print('SVelter Setup should be run first')
 */
  __pyx_tuple__14 = PyTuple_Pack(1, __pyx_kp_s_SVIntegrate); if (unlikely(!__pyx_tuple__14)) __PYX_ERR(0, 14, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__14);
  __Pyx_GIVEREF(__pyx_tuple__14);

  /* "svelter_sv/readme.pyx":15
 *     print('    SVPredict')
 *     print('    SVIntegrate')
 *     print('')             # <<<<<<<<<<<<<<
 *     print('SVelter Setup should be run first')
 *     print(' ')
 */
  __pyx_tuple__15 = PyTuple_Pack(1, __pyx_kp_s__2); if (unlikely(!__pyx_tuple__15)) __PYX_ERR(0, 15, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__15);
  __Pyx_GIVEREF(__pyx_tuple__15);

  /* "svelter_sv/readme.pyx":16
 *     print('    SVIntegrate')
 *     print('')
 *     print('SVelter Setup should be run first')             # <<<<<<<<<<<<<<
 *     print(' ')
 *     print('Type in SVelter [options] for detailed parameters')
 */
  __pyx_tuple__16 = PyTuple_Pack(1, __pyx_kp_s_SVelter_Setup_should_be_run_firs); if (unlikely(!__pyx_tuple__16)) __PYX_ERR(0, 16, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__16);
  __Pyx_GIVEREF(__pyx_tuple__16);

  /* "svelter_sv/readme.pyx":17
 *     print('')
 *     print('SVelter Setup should be run first')
 *     print(' ')             # <<<<<<<<<<<<<<
 *     print('Type in SVelter [options] for detailed parameters')
 * def print_default_parameters_predefinedbp():
 */
  __pyx_tuple__17 = PyTuple_Pack(1, __pyx_kp_s__5); if (unlikely(!__pyx_tuple__17)) __PYX_ERR(0, 17, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__17);
  __Pyx_GIVEREF(__pyx_tuple__17);

  /* "svelter_sv/readme.pyx":18
 *     print('SVelter Setup should be run first')
 *     print(' ')
 *     print('Type in SVelter [options] for detailed parameters')             # <<<<<<<<<<<<<<
 * def print_default_parameters_predefinedbp():
 *     print('SVelter PredefinedBP')
 */
  __pyx_tuple__18 = PyTuple_Pack(1, __pyx_kp_s_Type_in_SVelter_options_for_deta); if (unlikely(!__pyx_tuple__18)) __PYX_ERR(0, 18, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__18);
  __Pyx_GIVEREF(__pyx_tuple__18);

  /* "svelter_sv/readme.pyx":20
 *     print('Type in SVelter [options] for detailed parameters')
 * def print_default_parameters_predefinedbp():
 *     print('SVelter PredefinedBP')             # <<<<<<<<<<<<<<
 *     print(' ')
 *     print('Required Parameters:')
 */
  __pyx_tuple__19 = PyTuple_Pack(1, __pyx_kp_s_SVelter_PredefinedBP); if (unlikely(!__pyx_tuple__19)) __PYX_ERR(0, 20, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__19);
  __Pyx_GIVEREF(__pyx_tuple__19);

  /* "svelter_sv/readme.pyx":21
 * def print_default_parameters_predefinedbp():
 *     print('SVelter PredefinedBP')
 *     print(' ')             # <<<<<<<<<<<<<<
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')
 */
  __pyx_tuple__20 = PyTuple_Pack(1, __pyx_kp_s__5); if (unlikely(!__pyx_tuple__20)) __PYX_ERR(0, 21, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__20);
  __Pyx_GIVEREF(__pyx_tuple__20);

  /* "svelter_sv/readme.pyx":22
 *     print('SVelter PredefinedBP')
 *     print(' ')
 *     print('Required Parameters:')             # <<<<<<<<<<<<<<
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')
 */
  __pyx_tuple__21 = PyTuple_Pack(1, __pyx_kp_s_Required_Parameters); if (unlikely(!__pyx_tuple__21)) __PYX_ERR(0, 22, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__21);
  __Pyx_GIVEREF(__pyx_tuple__21);

  /* "svelter_sv/readme.pyx":23
 *     print(' ')
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')             # <<<<<<<<<<<<<<
 *     print('    --workdir, writable working directory.')
 *     print('    --input-bed, predefined breakpoints in bed format')
 */
  __pyx_tuple__22 = PyTuple_Pack(1, __pyx_kp_s_sample_input_alignment_file_in); if (unlikely(!__pyx_tuple__22)) __PYX_ERR(0, 23, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__22);
  __Pyx_GIVEREF(__pyx_tuple__22);

  /* "svelter_sv/readme.pyx":24
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')             # <<<<<<<<<<<<<<
 *     print('    --input-bed, predefined breakpoints in bed format')
 *     print('')
 */
  __pyx_tuple__23 = PyTuple_Pack(1, __pyx_kp_s_workdir_writable_working_direct); if (unlikely(!__pyx_tuple__23)) __PYX_ERR(0, 24, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__23);
  __Pyx_GIVEREF(__pyx_tuple__23);

  /* "svelter_sv/readme.pyx":25
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')
 *     print('    --input-bed, predefined breakpoints in bed format')             # <<<<<<<<<<<<<<
 *     print('')
 *     print('Optional Parameters:')
 */
  __pyx_tuple__24 = PyTuple_Pack(1, __pyx_kp_s_input_bed_predefined_breakpoint); if (unlikely(!__pyx_tuple__24)) __PYX_ERR(0, 25, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__24);
  __Pyx_GIVEREF(__pyx_tuple__24);

  /* "svelter_sv/readme.pyx":26
 *     print('    --workdir, writable working directory.')
 *     print('    --input-bed, predefined breakpoints in bed format')
 *     print('')             # <<<<<<<<<<<<<<
 *     print('Optional Parameters:')
 *     print('    --prefix, output prefix for vcf and svelter files [input.vcf, input.svelter]')
 */
  __pyx_tuple__25 = PyTuple_Pack(1, __pyx_kp_s__2); if (unlikely(!__pyx_tuple__25)) __PYX_ERR(0, 26, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__25);
  __Pyx_GIVEREF(__pyx_tuple__25);

  /* "svelter_sv/readme.pyx":27
 *     print('    --input-bed, predefined breakpoints in bed format')
 *     print('')
 *     print('Optional Parameters:')             # <<<<<<<<<<<<<<
 *     print('    --prefix, output prefix for vcf and svelter files [input.vcf, input.svelter]')
 *     print('    --num-iteration, maximum number of iterations per structure will run')
 */
  __pyx_tuple__26 = PyTuple_Pack(1, __pyx_kp_s_Optional_Parameters); if (unlikely(!__pyx_tuple__26)) __PYX_ERR(0, 27, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__26);
  __Pyx_GIVEREF(__pyx_tuple__26);

  /* "svelter_sv/readme.pyx":28
 *     print('')
 *     print('Optional Parameters:')
 *     print('    --prefix, output prefix for vcf and svelter files [input.vcf, input.svelter]')             # <<<<<<<<<<<<<<
 *     print('    --num-iteration, maximum number of iterations per structure will run')
 *     print('    --ploidy, limit algorithm to specific zygosity (0:heterozygous only; 1:homozygous only; 2:both; default:2)')
 */
  __pyx_tuple__27 = PyTuple_Pack(1, __pyx_kp_s_prefix_output_prefix_for_vcf_an); if (unlikely(!__pyx_tuple__27)) __PYX_ERR(0, 28, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__27);
  __Pyx_GIVEREF(__pyx_tuple__27);

  /* "svelter_sv/readme.pyx":29
 *     print('Optional Parameters:')
 *     print('    --prefix, output prefix for vcf and svelter files [input.vcf, input.svelter]')
 *     print('    --num-iteration, maximum number of iterations per structure will run')             # <<<<<<<<<<<<<<
 *     print('    --ploidy, limit algorithm to specific zygosity (0:heterozygous only; 1:homozygous only; 2:both; default:2)')
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')
 */
  __pyx_tuple__28 = PyTuple_Pack(1, __pyx_kp_s_num_iteration_maximum_number_of); if (unlikely(!__pyx_tuple__28)) __PYX_ERR(0, 29, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__28);
  __Pyx_GIVEREF(__pyx_tuple__28);

  /* "svelter_sv/readme.pyx":30
 *     print('    --prefix, output prefix for vcf and svelter files [input.vcf, input.svelter]')
 *     print('    --num-iteration, maximum number of iterations per structure will run')
 *     print('    --ploidy, limit algorithm to specific zygosity (0:heterozygous only; 1:homozygous only; 2:both; default:2)')             # <<<<<<<<<<<<<<
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')
 *     print('    --null-copyneutral-length, minimum length requirement for --copyneutral regions used to build null model [2000]')
 */
  __pyx_tuple__29 = PyTuple_Pack(1, __pyx_kp_s_ploidy_limit_algorithm_to_speci); if (unlikely(!__pyx_tuple__29)) __PYX_ERR(0, 30, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__29);
  __Pyx_GIVEREF(__pyx_tuple__29);

  /* "svelter_sv/readme.pyx":31
 *     print('    --num-iteration, maximum number of iterations per structure will run')
 *     print('    --ploidy, limit algorithm to specific zygosity (0:heterozygous only; 1:homozygous only; 2:both; default:2)')
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')             # <<<<<<<<<<<<<<
 *     print('    --null-copyneutral-length, minimum length requirement for --copyneutral regions used to build null model [2000]')
 *     print('    --null-copyneutral-perc, percentage of regions from --copyneutral to utilize [0.1]')
 */
  __pyx_tuple__30 = PyTuple_Pack(1, __pyx_kp_s_null_model_specify_which_stat_m); if (unlikely(!__pyx_tuple__30)) __PYX_ERR(0, 31, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__30);
  __Pyx_GIVEREF(__pyx_tuple__30);

  /* "svelter_sv/readme.pyx":32
 *     print('    --ploidy, limit algorithm to specific zygosity (0:heterozygous only; 1:homozygous only; 2:both; default:2)')
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')
 *     print('    --null-copyneutral-length, minimum length requirement for --copyneutral regions used to build null model [2000]')             # <<<<<<<<<<<<<<
 *     print('    --null-copyneutral-perc, percentage of regions from --copyneutral to utilize [0.1]')
 *     print('    --null-random-length, specify the length of random regions if --copyneutral parameter not used [5000]')
 */
  __pyx_tuple__31 = PyTuple_Pack(1, __pyx_kp_s_null_copyneutral_length_minimum); if (unlikely(!__pyx_tuple__31)) __PYX_ERR(0, 32, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__31);
  __Pyx_GIVEREF(__pyx_tuple__31);

  /* "svelter_sv/readme.pyx":33
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')
 *     print('    --null-copyneutral-length, minimum length requirement for --copyneutral regions used to build null model [2000]')
 *     print('    --null-copyneutral-perc, percentage of regions from --copyneutral to utilize [0.1]')             # <<<<<<<<<<<<<<
 *     print('    --null-random-length, specify the length of random regions if --copyneutral parameter not used [5000]')
 *     print('    --null-random-num, specify the number of random regions if --copyneutral parameter not used [10000]')
 */
  __pyx_tuple__32 = PyTuple_Pack(1, __pyx_kp_s_null_copyneutral_perc_percentag); if (unlikely(!__pyx_tuple__32)) __PYX_ERR(0, 33, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__32);
  __Pyx_GIVEREF(__pyx_tuple__32);

  /* "svelter_sv/readme.pyx":34
 *     print('    --null-copyneutral-length, minimum length requirement for --copyneutral regions used to build null model [2000]')
 *     print('    --null-copyneutral-perc, percentage of regions from --copyneutral to utilize [0.1]')
 *     print('    --null-random-length, specify the length of random regions if --copyneutral parameter not used [5000]')             # <<<<<<<<<<<<<<
 *     print('    --null-random-num, specify the number of random regions if --copyneutral parameter not used [10000]')
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20]')
 */
  __pyx_tuple__33 = PyTuple_Pack(1, __pyx_kp_s_null_random_length_specify_the); if (unlikely(!__pyx_tuple__33)) __PYX_ERR(0, 34, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__33);
  __Pyx_GIVEREF(__pyx_tuple__33);

  /* "svelter_sv/readme.pyx":35
 *     print('    --null-copyneutral-perc, percentage of regions from --copyneutral to utilize [0.1]')
 *     print('    --null-random-length, specify the length of random regions if --copyneutral parameter not used [5000]')
 *     print('    --null-random-num, specify the number of random regions if --copyneutral parameter not used [10000]')             # <<<<<<<<<<<<<<
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20]')
 *     print('    --qc-split, minimum alighment of clipped parts of reads considered as a soft clip [20]')
 */
  __pyx_tuple__34 = PyTuple_Pack(1, __pyx_kp_s_null_random_num_specify_the_num); if (unlikely(!__pyx_tuple__34)) __PYX_ERR(0, 35, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__34);
  __Pyx_GIVEREF(__pyx_tuple__34);

  /* "svelter_sv/readme.pyx":36
 *     print('    --null-random-length, specify the length of random regions if --copyneutral parameter not used [5000]')
 *     print('    --null-random-num, specify the number of random regions if --copyneutral parameter not used [10000]')
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20]')             # <<<<<<<<<<<<<<
 *     print('    --qc-split, minimum alighment of clipped parts of reads considered as a soft clip [20]')
 *     print('    --split-min-len, the minumum length of clip read considered as split; [10% of read length]')
 */
  __pyx_tuple__35 = PyTuple_Pack(1, __pyx_kp_s_qc_align_minimum_alignment_qual); if (unlikely(!__pyx_tuple__35)) __PYX_ERR(0, 36, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__35);
  __Pyx_GIVEREF(__pyx_tuple__35);

  /* "svelter_sv/readme.pyx":37
 *     print('    --null-random-num, specify the number of random regions if --copyneutral parameter not used [10000]')
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20]')
 *     print('    --qc-split, minimum alighment of clipped parts of reads considered as a soft clip [20]')             # <<<<<<<<<<<<<<
 *     print('    --split-min-len, the minumum length of clip read considered as split; [10% of read length]')
 *     print('    --qc-structure, minimum quality score of a resolved structure to be considered as PASS and included in the output vcf file')
 */
  __pyx_tuple__36 = PyTuple_Pack(1, __pyx_kp_s_qc_split_minimum_alighment_of_c); if (unlikely(!__pyx_tuple__36)) __PYX_ERR(0, 37, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__36);
  __Pyx_GIVEREF(__pyx_tuple__36);

  /* "svelter_sv/readme.pyx":38
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20]')
 *     print('    --qc-split, minimum alighment of clipped parts of reads considered as a soft clip [20]')
 *     print('    --split-min-len, the minumum length of clip read considered as split; [10% of read length]')             # <<<<<<<<<<<<<<
 *     print('    --qc-structure, minimum quality score of a resolved structure to be considered as PASS and included in the output vcf file')
 *     print('    --qc-map-tool, the tool extracts mappability information from a bigWig file,avaliable from: ')
 */
  __pyx_tuple__37 = PyTuple_Pack(1, __pyx_kp_s_split_min_len_the_minumum_lengt); if (unlikely(!__pyx_tuple__37)) __PYX_ERR(0, 38, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__37);
  __Pyx_GIVEREF(__pyx_tuple__37);

  /* "svelter_sv/readme.pyx":39
 *     print('    --qc-split, minimum alighment of clipped parts of reads considered as a soft clip [20]')
 *     print('    --split-min-len, the minumum length of clip read considered as split; [10% of read length]')
 *     print('    --qc-structure, minimum quality score of a resolved structure to be considered as PASS and included in the output vcf file')             # <<<<<<<<<<<<<<
 *     print('    --qc-map-tool, the tool extracts mappability information from a bigWig file,avaliable from: ')
 *     print('                     http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/bigWigSummary')
 */
  __pyx_tuple__38 = PyTuple_Pack(1, __pyx_kp_s_qc_structure_minimum_quality_sc); if (unlikely(!__pyx_tuple__38)) __PYX_ERR(0, 39, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__38);
  __Pyx_GIVEREF(__pyx_tuple__38);

  /* "svelter_sv/readme.pyx":40
 *     print('    --split-min-len, the minumum length of clip read considered as split; [10% of read length]')
 *     print('    --qc-structure, minimum quality score of a resolved structure to be considered as PASS and included in the output vcf file')
 *     print('    --qc-map-tool, the tool extracts mappability information from a bigWig file,avaliable from: ')             # <<<<<<<<<<<<<<
 *     print('                     http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/bigWigSummary')
 *     print('    --qc-map-file, .bigWig file used to decide local genomic mappability, avaliable from: ')
 */
  __pyx_tuple__39 = PyTuple_Pack(1, __pyx_kp_s_qc_map_tool_the_tool_extracts_m); if (unlikely(!__pyx_tuple__39)) __PYX_ERR(0, 40, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__39);
  __Pyx_GIVEREF(__pyx_tuple__39);

  /* "svelter_sv/readme.pyx":41
 *     print('    --qc-structure, minimum quality score of a resolved structure to be considered as PASS and included in the output vcf file')
 *     print('    --qc-map-tool, the tool extracts mappability information from a bigWig file,avaliable from: ')
 *     print('                     http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/bigWigSummary')             # <<<<<<<<<<<<<<
 *     print('    --qc-map-file, .bigWig file used to decide local genomic mappability, avaliable from: ')
 *     print('                     ftp://hgdownload.cse.ucsc.edu/goldenPath/currentGenomes/Homo_sapiens/encodeDCC/wgEncodeMapability/')
 */
  __pyx_tuple__40 = PyTuple_Pack(1, __pyx_kp_s_http_hgdownload_cse_ucsc_edu_ad); if (unlikely(!__pyx_tuple__40)) __PYX_ERR(0, 41, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__40);
  __Pyx_GIVEREF(__pyx_tuple__40);

  /* "svelter_sv/readme.pyx":42
 *     print('    --qc-map-tool, the tool extracts mappability information from a bigWig file,avaliable from: ')
 *     print('                     http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/bigWigSummary')
 *     print('    --qc-map-file, .bigWig file used to decide local genomic mappability, avaliable from: ')             # <<<<<<<<<<<<<<
 *     print('                     ftp://hgdownload.cse.ucsc.edu/goldenPath/currentGenomes/Homo_sapiens/encodeDCC/wgEncodeMapability/')
 *     print('    --qc-map-cutoff, the minimum mapping quality required for a breakpoint to be reported [0.0]')
 */
  __pyx_tuple__41 = PyTuple_Pack(1, __pyx_kp_s_qc_map_file_bigWig_file_used_to); if (unlikely(!__pyx_tuple__41)) __PYX_ERR(0, 42, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__41);
  __Pyx_GIVEREF(__pyx_tuple__41);

  /* "svelter_sv/readme.pyx":43
 *     print('                     http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/bigWigSummary')
 *     print('    --qc-map-file, .bigWig file used to decide local genomic mappability, avaliable from: ')
 *     print('                     ftp://hgdownload.cse.ucsc.edu/goldenPath/currentGenomes/Homo_sapiens/encodeDCC/wgEncodeMapability/')             # <<<<<<<<<<<<<<
 *     print('    --qc-map-cutoff, the minimum mapping quality required for a breakpoint to be reported [0.0]')
 * def print_default_parameters_clean():
 */
  __pyx_tuple__42 = PyTuple_Pack(1, __pyx_kp_s_ftp_hgdownload_cse_ucsc_edu_gol); if (unlikely(!__pyx_tuple__42)) __PYX_ERR(0, 43, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__42);
  __Pyx_GIVEREF(__pyx_tuple__42);

  /* "svelter_sv/readme.pyx":44
 *     print('    --qc-map-file, .bigWig file used to decide local genomic mappability, avaliable from: ')
 *     print('                     ftp://hgdownload.cse.ucsc.edu/goldenPath/currentGenomes/Homo_sapiens/encodeDCC/wgEncodeMapability/')
 *     print('    --qc-map-cutoff, the minimum mapping quality required for a breakpoint to be reported [0.0]')             # <<<<<<<<<<<<<<
 * def print_default_parameters_clean():
 *     print('SVelter Clean')
 */
  __pyx_tuple__43 = PyTuple_Pack(1, __pyx_kp_s_qc_map_cutoff_the_minimum_mappi); if (unlikely(!__pyx_tuple__43)) __PYX_ERR(0, 44, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__43);
  __Pyx_GIVEREF(__pyx_tuple__43);

  /* "svelter_sv/readme.pyx":46
 *     print('    --qc-map-cutoff, the minimum mapping quality required for a breakpoint to be reported [0.0]')
 * def print_default_parameters_clean():
 *     print('SVelter Clean')             # <<<<<<<<<<<<<<
 *     print('')
 *     print('Required Parameters:')
 */
  __pyx_tuple__44 = PyTuple_Pack(1, __pyx_kp_s_SVelter_Clean); if (unlikely(!__pyx_tuple__44)) __PYX_ERR(0, 46, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__44);
  __Pyx_GIVEREF(__pyx_tuple__44);

  /* "svelter_sv/readme.pyx":47
 * def print_default_parameters_clean():
 *     print('SVelter Clean')
 *     print('')             # <<<<<<<<<<<<<<
 *     print('Required Parameters:')
 *     print('    --workdir, writable working directory.')
 */
  __pyx_tuple__45 = PyTuple_Pack(1, __pyx_kp_s__2); if (unlikely(!__pyx_tuple__45)) __PYX_ERR(0, 47, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__45);
  __Pyx_GIVEREF(__pyx_tuple__45);

  /* "svelter_sv/readme.pyx":48
 *     print('SVelter Clean')
 *     print('')
 *     print('Required Parameters:')             # <<<<<<<<<<<<<<
 *     print('    --workdir, writable working directory.')
 * def print_default_parameters_setup():
 */
  __pyx_tuple__46 = PyTuple_Pack(1, __pyx_kp_s_Required_Parameters); if (unlikely(!__pyx_tuple__46)) __PYX_ERR(0, 48, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__46);
  __Pyx_GIVEREF(__pyx_tuple__46);

  /* "svelter_sv/readme.pyx":49
 *     print('')
 *     print('Required Parameters:')
 *     print('    --workdir, writable working directory.')             # <<<<<<<<<<<<<<
 * def print_default_parameters_setup():
 *     print('SVelter Setup')
 */
  __pyx_tuple__47 = PyTuple_Pack(1, __pyx_kp_s_workdir_writable_working_direct); if (unlikely(!__pyx_tuple__47)) __PYX_ERR(0, 49, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__47);
  __Pyx_GIVEREF(__pyx_tuple__47);

  /* "svelter_sv/readme.pyx":51
 *     print('    --workdir, writable working directory.')
 * def print_default_parameters_setup():
 *     print('SVelter Setup')             # <<<<<<<<<<<<<<
 *     print('')
 *     print('Required Parameters:')
 */
  __pyx_tuple__48 = PyTuple_Pack(1, __pyx_kp_s_SVelter_Setup); if (unlikely(!__pyx_tuple__48)) __PYX_ERR(0, 51, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__48);
  __Pyx_GIVEREF(__pyx_tuple__48);

  /* "svelter_sv/readme.pyx":52
 * def print_default_parameters_setup():
 *     print('SVelter Setup')
 *     print('')             # <<<<<<<<<<<<<<
 *     print('Required Parameters:')
 *     print('    --workdir, writable working directory.')
 */
  __pyx_tuple__49 = PyTuple_Pack(1, __pyx_kp_s__2); if (unlikely(!__pyx_tuple__49)) __PYX_ERR(0, 52, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__49);
  __Pyx_GIVEREF(__pyx_tuple__49);

  /* "svelter_sv/readme.pyx":53
 *     print('SVelter Setup')
 *     print('')
 *     print('Required Parameters:')             # <<<<<<<<<<<<<<
 *     print('    --workdir, writable working directory.')
 *     print('    --reference, absolute path of reference genome. eg: .../SVelter/reference/genome.fa')
 */
  __pyx_tuple__50 = PyTuple_Pack(1, __pyx_kp_s_Required_Parameters); if (unlikely(!__pyx_tuple__50)) __PYX_ERR(0, 53, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__50);
  __Pyx_GIVEREF(__pyx_tuple__50);

  /* "svelter_sv/readme.pyx":54
 *     print('')
 *     print('Required Parameters:')
 *     print('    --workdir, writable working directory.')             # <<<<<<<<<<<<<<
 *     print('    --reference, absolute path of reference genome. eg: .../SVelter/reference/genome.fa')
 *     print('    --support, folder containing all supportive file including: Exclude.bed,CN2.bed,Segdup.bed')
 */
  __pyx_tuple__51 = PyTuple_Pack(1, __pyx_kp_s_workdir_writable_working_direct); if (unlikely(!__pyx_tuple__51)) __PYX_ERR(0, 54, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__51);
  __Pyx_GIVEREF(__pyx_tuple__51);

  /* "svelter_sv/readme.pyx":55
 *     print('Required Parameters:')
 *     print('    --workdir, writable working directory.')
 *     print('    --reference, absolute path of reference genome. eg: .../SVelter/reference/genome.fa')             # <<<<<<<<<<<<<<
 *     print('    --support, folder containing all supportive file including: Exclude.bed,CN2.bed,Segdup.bed')
 *     print('')
 */
  __pyx_tuple__52 = PyTuple_Pack(1, __pyx_kp_s_reference_absolute_path_of_refe); if (unlikely(!__pyx_tuple__52)) __PYX_ERR(0, 55, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__52);
  __Pyx_GIVEREF(__pyx_tuple__52);

  /* "svelter_sv/readme.pyx":56
 *     print('    --workdir, writable working directory.')
 *     print('    --reference, absolute path of reference genome. eg: .../SVelter/reference/genome.fa')
 *     print('    --support, folder containing all supportive file including: Exclude.bed,CN2.bed,Segdup.bed')             # <<<<<<<<<<<<<<
 *     print('')
 *     print('Optional Parameters:')
 */
  __pyx_tuple__53 = PyTuple_Pack(1, __pyx_kp_s_support_folder_containing_all_s); if (unlikely(!__pyx_tuple__53)) __PYX_ERR(0, 56, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__53);
  __Pyx_GIVEREF(__pyx_tuple__53);

  /* "svelter_sv/readme.pyx":57
 *     print('    --reference, absolute path of reference genome. eg: .../SVelter/reference/genome.fa')
 *     print('    --support, folder containing all supportive file including: Exclude.bed,CN2.bed,Segdup.bed')
 *     print('')             # <<<<<<<<<<<<<<
 *     print('Optional Parameters:')
 *     #print '    --support, folder containing all supportive file including: Exclude.bed,CN2.bed,Segdup.bed'
 */
  __pyx_tuple__54 = PyTuple_Pack(1, __pyx_kp_s__2); if (unlikely(!__pyx_tuple__54)) __PYX_ERR(0, 57, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__54);
  __Pyx_GIVEREF(__pyx_tuple__54);

  /* "svelter_sv/readme.pyx":58
 *     print('    --support, folder containing all supportive file including: Exclude.bed,CN2.bed,Segdup.bed')
 *     print('')
 *     print('Optional Parameters:')             # <<<<<<<<<<<<<<
 *     #print '    --support, folder containing all supportive file including: Exclude.bed,CN2.bed,Segdup.bed'
 *     #print '    --svelter-path, folder which contains all SVelter scripts.'
 */
  __pyx_tuple__55 = PyTuple_Pack(1, __pyx_kp_s_Optional_Parameters); if (unlikely(!__pyx_tuple__55)) __PYX_ERR(0, 58, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__55);
  __Pyx_GIVEREF(__pyx_tuple__55);

  /* "svelter_sv/readme.pyx":64
 *     #print '    --exclude, absolute path of bed file indicating regions to be excluded from analysis. If not provided, no mappable regions will be excluded.'
 *     #print '    --segdup, absolute path of bed file indicating segmental duplications in genome, that would be excluded from analysis'
 *     print('    --ref-index, folders containin pre-indexed files, if applicable. For certain versions of human genome, the indexed files are availabel from https://github.com/mills-lab/svelter.')             # <<<<<<<<<<<<<<
 * def print_default_parameters_nullmodel():
 *     print('SVelter NullModel')
 */
  __pyx_tuple__56 = PyTuple_Pack(1, __pyx_kp_s_ref_index_folders_containin_pre); if (unlikely(!__pyx_tuple__56)) __PYX_ERR(0, 64, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__56);
  __Pyx_GIVEREF(__pyx_tuple__56);

  /* "svelter_sv/readme.pyx":66
 *     print('    --ref-index, folders containin pre-indexed files, if applicable. For certain versions of human genome, the indexed files are availabel from https://github.com/mills-lab/svelter.')
 * def print_default_parameters_nullmodel():
 *     print('SVelter NullModel')             # <<<<<<<<<<<<<<
 *     print('')
 *     print('Required Parameters:')
 */
  __pyx_tuple__57 = PyTuple_Pack(1, __pyx_kp_s_SVelter_NullModel); if (unlikely(!__pyx_tuple__57)) __PYX_ERR(0, 66, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__57);
  __Pyx_GIVEREF(__pyx_tuple__57);

  /* "svelter_sv/readme.pyx":67
 * def print_default_parameters_nullmodel():
 *     print('SVelter NullModel')
 *     print('')             # <<<<<<<<<<<<<<
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')
 */
  __pyx_tuple__58 = PyTuple_Pack(1, __pyx_kp_s__2); if (unlikely(!__pyx_tuple__58)) __PYX_ERR(0, 67, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__58);
  __Pyx_GIVEREF(__pyx_tuple__58);

  /* "svelter_sv/readme.pyx":68
 *     print('SVelter NullModel')
 *     print('')
 *     print('Required Parameters:')             # <<<<<<<<<<<<<<
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')
 */
  __pyx_tuple__59 = PyTuple_Pack(1, __pyx_kp_s_Required_Parameters); if (unlikely(!__pyx_tuple__59)) __PYX_ERR(0, 68, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__59);
  __Pyx_GIVEREF(__pyx_tuple__59);

  /* "svelter_sv/readme.pyx":69
 *     print('')
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')             # <<<<<<<<<<<<<<
 *     print('    --workdir, writable working directory.')
 *     print('')
 */
  __pyx_tuple__60 = PyTuple_Pack(1, __pyx_kp_s_sample_input_alignment_file_in); if (unlikely(!__pyx_tuple__60)) __PYX_ERR(0, 69, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__60);
  __Pyx_GIVEREF(__pyx_tuple__60);

  /* "svelter_sv/readme.pyx":70
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')             # <<<<<<<<<<<<<<
 *     print('')
 *     print('Optional Parameters:')
 */
  __pyx_tuple__61 = PyTuple_Pack(1, __pyx_kp_s_workdir_writable_working_direct); if (unlikely(!__pyx_tuple__61)) __PYX_ERR(0, 70, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__61);
  __Pyx_GIVEREF(__pyx_tuple__61);

  /* "svelter_sv/readme.pyx":71
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')
 *     print('')             # <<<<<<<<<<<<<<
 *     print('Optional Parameters:')
 *     print('    --chromosome, name of chromosome to run. should match chromosome name in bam file')
 */
  __pyx_tuple__62 = PyTuple_Pack(1, __pyx_kp_s__2); if (unlikely(!__pyx_tuple__62)) __PYX_ERR(0, 71, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__62);
  __Pyx_GIVEREF(__pyx_tuple__62);

  /* "svelter_sv/readme.pyx":72
 *     print('    --workdir, writable working directory.')
 *     print('')
 *     print('Optional Parameters:')             # <<<<<<<<<<<<<<
 *     print('    --chromosome, name of chromosome to run. should match chromosome name in bam file')
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')
 */
  __pyx_tuple__63 = PyTuple_Pack(1, __pyx_kp_s_Optional_Parameters); if (unlikely(!__pyx_tuple__63)) __PYX_ERR(0, 72, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__63);
  __Pyx_GIVEREF(__pyx_tuple__63);

  /* "svelter_sv/readme.pyx":73
 *     print('')
 *     print('Optional Parameters:')
 *     print('    --chromosome, name of chromosome to run. should match chromosome name in bam file')             # <<<<<<<<<<<<<<
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')
 *     print('    --null-copyneutral-length, minimum length requirement for --copyneutral regions used to build null model [2000]')
 */
  __pyx_tuple__64 = PyTuple_Pack(1, __pyx_kp_s_chromosome_name_of_chromosome_t); if (unlikely(!__pyx_tuple__64)) __PYX_ERR(0, 73, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__64);
  __Pyx_GIVEREF(__pyx_tuple__64);

  /* "svelter_sv/readme.pyx":74
 *     print('Optional Parameters:')
 *     print('    --chromosome, name of chromosome to run. should match chromosome name in bam file')
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')             # <<<<<<<<<<<<<<
 *     print('    --null-copyneutral-length, minimum length requirement for --copyneutral regions used to build null model [2000]')
 *     print('    --null-copyneutral-perc, percentage of regions from --copyneutral to utilize [0.1]')
 */
  __pyx_tuple__65 = PyTuple_Pack(1, __pyx_kp_s_null_model_specify_which_stat_m); if (unlikely(!__pyx_tuple__65)) __PYX_ERR(0, 74, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__65);
  __Pyx_GIVEREF(__pyx_tuple__65);

  /* "svelter_sv/readme.pyx":75
 *     print('    --chromosome, name of chromosome to run. should match chromosome name in bam file')
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')
 *     print('    --null-copyneutral-length, minimum length requirement for --copyneutral regions used to build null model [2000]')             # <<<<<<<<<<<<<<
 *     print('    --null-copyneutral-perc, percentage of regions from --copyneutral to utilize [0.1]')
 *     print('    --null-random-length, specify the length of random regions if --copyneutral parameter not used [5000]')
 */
  __pyx_tuple__66 = PyTuple_Pack(1, __pyx_kp_s_null_copyneutral_length_minimum); if (unlikely(!__pyx_tuple__66)) __PYX_ERR(0, 75, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__66);
  __Pyx_GIVEREF(__pyx_tuple__66);

  /* "svelter_sv/readme.pyx":76
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')
 *     print('    --null-copyneutral-length, minimum length requirement for --copyneutral regions used to build null model [2000]')
 *     print('    --null-copyneutral-perc, percentage of regions from --copyneutral to utilize [0.1]')             # <<<<<<<<<<<<<<
 *     print('    --null-random-length, specify the length of random regions if --copyneutral parameter not used [5000]')
 *     print('    --null-random-num, specify the number of random regions if --copyneutral parameter not used [10000]')
 */
  __pyx_tuple__67 = PyTuple_Pack(1, __pyx_kp_s_null_copyneutral_perc_percentag); if (unlikely(!__pyx_tuple__67)) __PYX_ERR(0, 76, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__67);
  __Pyx_GIVEREF(__pyx_tuple__67);

  /* "svelter_sv/readme.pyx":77
 *     print('    --null-copyneutral-length, minimum length requirement for --copyneutral regions used to build null model [2000]')
 *     print('    --null-copyneutral-perc, percentage of regions from --copyneutral to utilize [0.1]')
 *     print('    --null-random-length, specify the length of random regions if --copyneutral parameter not used [5000]')             # <<<<<<<<<<<<<<
 *     print('    --null-random-num, specify the number of random regions if --copyneutral parameter not used [10000]')
 *     print('    --split-min-len, the minumum length of clip read considered as split; [10% of read length]')
 */
  __pyx_tuple__68 = PyTuple_Pack(1, __pyx_kp_s_null_random_length_specify_the); if (unlikely(!__pyx_tuple__68)) __PYX_ERR(0, 77, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__68);
  __Pyx_GIVEREF(__pyx_tuple__68);

  /* "svelter_sv/readme.pyx":78
 *     print('    --null-copyneutral-perc, percentage of regions from --copyneutral to utilize [0.1]')
 *     print('    --null-random-length, specify the length of random regions if --copyneutral parameter not used [5000]')
 *     print('    --null-random-num, specify the number of random regions if --copyneutral parameter not used [10000]')             # <<<<<<<<<<<<<<
 *     print('    --split-min-len, the minumum length of clip read considered as split; [10% of read length]')
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20]')
 */
  __pyx_tuple__69 = PyTuple_Pack(1, __pyx_kp_s_null_random_num_specify_the_num); if (unlikely(!__pyx_tuple__69)) __PYX_ERR(0, 78, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__69);
  __Pyx_GIVEREF(__pyx_tuple__69);

  /* "svelter_sv/readme.pyx":79
 *     print('    --null-random-length, specify the length of random regions if --copyneutral parameter not used [5000]')
 *     print('    --null-random-num, specify the number of random regions if --copyneutral parameter not used [10000]')
 *     print('    --split-min-len, the minumum length of clip read considered as split; [10% of read length]')             # <<<<<<<<<<<<<<
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20]')
 *     print('    --qc-split, minimum alighment of clipped parts of reads considered as a soft clip [20]')
 */
  __pyx_tuple__70 = PyTuple_Pack(1, __pyx_kp_s_split_min_len_the_minumum_lengt); if (unlikely(!__pyx_tuple__70)) __PYX_ERR(0, 79, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__70);
  __Pyx_GIVEREF(__pyx_tuple__70);

  /* "svelter_sv/readme.pyx":80
 *     print('    --null-random-num, specify the number of random regions if --copyneutral parameter not used [10000]')
 *     print('    --split-min-len, the minumum length of clip read considered as split; [10% of read length]')
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20]')             # <<<<<<<<<<<<<<
 *     print('    --qc-split, minimum alighment of clipped parts of reads considered as a soft clip [20]')
 * def print_default_parameters_bpsearch():
 */
  __pyx_tuple__71 = PyTuple_Pack(1, __pyx_kp_s_qc_align_minimum_alignment_qual); if (unlikely(!__pyx_tuple__71)) __PYX_ERR(0, 80, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__71);
  __Pyx_GIVEREF(__pyx_tuple__71);

  /* "svelter_sv/readme.pyx":81
 *     print('    --split-min-len, the minumum length of clip read considered as split; [10% of read length]')
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20]')
 *     print('    --qc-split, minimum alighment of clipped parts of reads considered as a soft clip [20]')             # <<<<<<<<<<<<<<
 * def print_default_parameters_bpsearch():
 *     print('SVelter  BPSearch')
 */
  __pyx_tuple__72 = PyTuple_Pack(1, __pyx_kp_s_qc_split_minimum_alighment_of_c); if (unlikely(!__pyx_tuple__72)) __PYX_ERR(0, 81, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__72);
  __Pyx_GIVEREF(__pyx_tuple__72);

  /* "svelter_sv/readme.pyx":83
 *     print('    --qc-split, minimum alighment of clipped parts of reads considered as a soft clip [20]')
 * def print_default_parameters_bpsearch():
 *     print('SVelter  BPSearch')             # <<<<<<<<<<<<<<
 *     print('')
 *     print('Required Parameters:')
 */
  __pyx_tuple__73 = PyTuple_Pack(1, __pyx_kp_s_SVelter_BPSearch); if (unlikely(!__pyx_tuple__73)) __PYX_ERR(0, 83, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__73);
  __Pyx_GIVEREF(__pyx_tuple__73);

  /* "svelter_sv/readme.pyx":84
 * def print_default_parameters_bpsearch():
 *     print('SVelter  BPSearch')
 *     print('')             # <<<<<<<<<<<<<<
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')
 */
  __pyx_tuple__74 = PyTuple_Pack(1, __pyx_kp_s__2); if (unlikely(!__pyx_tuple__74)) __PYX_ERR(0, 84, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__74);
  __Pyx_GIVEREF(__pyx_tuple__74);

  /* "svelter_sv/readme.pyx":85
 *     print('SVelter  BPSearch')
 *     print('')
 *     print('Required Parameters:')             # <<<<<<<<<<<<<<
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')
 */
  __pyx_tuple__75 = PyTuple_Pack(1, __pyx_kp_s_Required_Parameters); if (unlikely(!__pyx_tuple__75)) __PYX_ERR(0, 85, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__75);
  __Pyx_GIVEREF(__pyx_tuple__75);

  /* "svelter_sv/readme.pyx":86
 *     print('')
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')             # <<<<<<<<<<<<<<
 *     print('    --workdir, writable working directory.')
 *     print('')
 */
  __pyx_tuple__76 = PyTuple_Pack(1, __pyx_kp_s_sample_input_alignment_file_in); if (unlikely(!__pyx_tuple__76)) __PYX_ERR(0, 86, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__76);
  __Pyx_GIVEREF(__pyx_tuple__76);

  /* "svelter_sv/readme.pyx":87
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')             # <<<<<<<<<<<<<<
 *     print('')
 *     print('Optional Parameters:')
 */
  __pyx_tuple__77 = PyTuple_Pack(1, __pyx_kp_s_workdir_writable_working_direct); if (unlikely(!__pyx_tuple__77)) __PYX_ERR(0, 87, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__77);
  __Pyx_GIVEREF(__pyx_tuple__77);

  /* "svelter_sv/readme.pyx":88
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')
 *     print('')             # <<<<<<<<<<<<<<
 *     print('Optional Parameters:')
 *     print('    --chromosome, name of chromosome to run. should match chromosome name in bam file')
 */
  __pyx_tuple__78 = PyTuple_Pack(1, __pyx_kp_s__2); if (unlikely(!__pyx_tuple__78)) __PYX_ERR(0, 88, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__78);
  __Pyx_GIVEREF(__pyx_tuple__78);

  /* "svelter_sv/readme.pyx":89
 *     print('    --workdir, writable working directory.')
 *     print('')
 *     print('Optional Parameters:')             # <<<<<<<<<<<<<<
 *     print('    --chromosome, name of chromosome to run. should match chromosome name in bam file')
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')
 */
  __pyx_tuple__79 = PyTuple_Pack(1, __pyx_kp_s_Optional_Parameters); if (unlikely(!__pyx_tuple__79)) __PYX_ERR(0, 89, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__79);
  __Pyx_GIVEREF(__pyx_tuple__79);

  /* "svelter_sv/readme.pyx":90
 *     print('')
 *     print('Optional Parameters:')
 *     print('    --chromosome, name of chromosome to run. should match chromosome name in bam file')             # <<<<<<<<<<<<<<
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20)')
 */
  __pyx_tuple__80 = PyTuple_Pack(1, __pyx_kp_s_chromosome_name_of_chromosome_t); if (unlikely(!__pyx_tuple__80)) __PYX_ERR(0, 90, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__80);
  __Pyx_GIVEREF(__pyx_tuple__80);

  /* "svelter_sv/readme.pyx":91
 *     print('Optional Parameters:')
 *     print('    --chromosome, name of chromosome to run. should match chromosome name in bam file')
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')             # <<<<<<<<<<<<<<
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20)')
 *     print('    --qc-split, minimum alighment of clipped parts of reads considered as a soft clip [20)')
 */
  __pyx_tuple__81 = PyTuple_Pack(1, __pyx_kp_s_null_model_specify_which_stat_m); if (unlikely(!__pyx_tuple__81)) __PYX_ERR(0, 91, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__81);
  __Pyx_GIVEREF(__pyx_tuple__81);

  /* "svelter_sv/readme.pyx":92
 *     print('    --chromosome, name of chromosome to run. should match chromosome name in bam file')
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20)')             # <<<<<<<<<<<<<<
 *     print('    --qc-split, minimum alighment of clipped parts of reads considered as a soft clip [20)')
 *     print('    --split-min-len, the minumum length of clip read considered as split; (default:10% of read length)')
 */
  __pyx_tuple__82 = PyTuple_Pack(1, __pyx_kp_s_qc_align_minimum_alignment_qual_2); if (unlikely(!__pyx_tuple__82)) __PYX_ERR(0, 92, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__82);
  __Pyx_GIVEREF(__pyx_tuple__82);

  /* "svelter_sv/readme.pyx":93
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20)')
 *     print('    --qc-split, minimum alighment of clipped parts of reads considered as a soft clip [20)')             # <<<<<<<<<<<<<<
 *     print('    --split-min-len, the minumum length of clip read considered as split; (default:10% of read length)')
 *     print('    --qc-map-tool, the tool extracts mappability information from a bigWig file,avaliable from: http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/bigWigSummary')
 */
  __pyx_tuple__83 = PyTuple_Pack(1, __pyx_kp_s_qc_split_minimum_alighment_of_c_2); if (unlikely(!__pyx_tuple__83)) __PYX_ERR(0, 93, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__83);
  __Pyx_GIVEREF(__pyx_tuple__83);

  /* "svelter_sv/readme.pyx":94
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20)')
 *     print('    --qc-split, minimum alighment of clipped parts of reads considered as a soft clip [20)')
 *     print('    --split-min-len, the minumum length of clip read considered as split; (default:10% of read length)')             # <<<<<<<<<<<<<<
 *     print('    --qc-map-tool, the tool extracts mappability information from a bigWig file,avaliable from: http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/bigWigSummary')
 *     print('    --qc-map-file, .bigWig file used to decide local genomic mappability, avaliable from: ftp://hgdownload.cse.ucsc.edu/goldenPath/currentGenomes/Homo_sapiens/encodeDCC/wgEncodeMapability/')
 */
  __pyx_tuple__84 = PyTuple_Pack(1, __pyx_kp_s_split_min_len_the_minumum_lengt_2); if (unlikely(!__pyx_tuple__84)) __PYX_ERR(0, 94, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__84);
  __Pyx_GIVEREF(__pyx_tuple__84);

  /* "svelter_sv/readme.pyx":95
 *     print('    --qc-split, minimum alighment of clipped parts of reads considered as a soft clip [20)')
 *     print('    --split-min-len, the minumum length of clip read considered as split; (default:10% of read length)')
 *     print('    --qc-map-tool, the tool extracts mappability information from a bigWig file,avaliable from: http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/bigWigSummary')             # <<<<<<<<<<<<<<
 *     print('    --qc-map-file, .bigWig file used to decide local genomic mappability, avaliable from: ftp://hgdownload.cse.ucsc.edu/goldenPath/currentGenomes/Homo_sapiens/encodeDCC/wgEncodeMapability/')
 *     print('    --qc-map-cutoff, the minimum mapping quality required for a breakpoint to be reported [0.0)')
 */
  __pyx_tuple__85 = PyTuple_Pack(1, __pyx_kp_s_qc_map_tool_the_tool_extracts_m_2); if (unlikely(!__pyx_tuple__85)) __PYX_ERR(0, 95, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__85);
  __Pyx_GIVEREF(__pyx_tuple__85);

  /* "svelter_sv/readme.pyx":96
 *     print('    --split-min-len, the minumum length of clip read considered as split; (default:10% of read length)')
 *     print('    --qc-map-tool, the tool extracts mappability information from a bigWig file,avaliable from: http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/bigWigSummary')
 *     print('    --qc-map-file, .bigWig file used to decide local genomic mappability, avaliable from: ftp://hgdownload.cse.ucsc.edu/goldenPath/currentGenomes/Homo_sapiens/encodeDCC/wgEncodeMapability/')             # <<<<<<<<<<<<<<
 *     print('    --qc-map-cutoff, the minimum mapping quality required for a breakpoint to be reported [0.0)')
 * def print_default_parameters_bpintegrate():
 */
  __pyx_tuple__86 = PyTuple_Pack(1, __pyx_kp_s_qc_map_file_bigWig_file_used_to_2); if (unlikely(!__pyx_tuple__86)) __PYX_ERR(0, 96, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__86);
  __Pyx_GIVEREF(__pyx_tuple__86);

  /* "svelter_sv/readme.pyx":97
 *     print('    --qc-map-tool, the tool extracts mappability information from a bigWig file,avaliable from: http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/bigWigSummary')
 *     print('    --qc-map-file, .bigWig file used to decide local genomic mappability, avaliable from: ftp://hgdownload.cse.ucsc.edu/goldenPath/currentGenomes/Homo_sapiens/encodeDCC/wgEncodeMapability/')
 *     print('    --qc-map-cutoff, the minimum mapping quality required for a breakpoint to be reported [0.0)')             # <<<<<<<<<<<<<<
 * def print_default_parameters_bpintegrate():
 *     print('SVelter  BPIntegrate')
 */
  __pyx_tuple__87 = PyTuple_Pack(1, __pyx_kp_s_qc_map_cutoff_the_minimum_mappi_2); if (unlikely(!__pyx_tuple__87)) __PYX_ERR(0, 97, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__87);
  __Pyx_GIVEREF(__pyx_tuple__87);

  /* "svelter_sv/readme.pyx":99
 *     print('    --qc-map-cutoff, the minimum mapping quality required for a breakpoint to be reported [0.0)')
 * def print_default_parameters_bpintegrate():
 *     print('SVelter  BPIntegrate')             # <<<<<<<<<<<<<<
 *     print('')
 *     print('Required Parameters:')
 */
  __pyx_tuple__88 = PyTuple_Pack(1, __pyx_kp_s_SVelter_BPIntegrate); if (unlikely(!__pyx_tuple__88)) __PYX_ERR(0, 99, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__88);
  __Pyx_GIVEREF(__pyx_tuple__88);

  /* "svelter_sv/readme.pyx":100
 * def print_default_parameters_bpintegrate():
 *     print('SVelter  BPIntegrate')
 *     print('')             # <<<<<<<<<<<<<<
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')
 */
  __pyx_tuple__89 = PyTuple_Pack(1, __pyx_kp_s__2); if (unlikely(!__pyx_tuple__89)) __PYX_ERR(0, 100, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__89);
  __Pyx_GIVEREF(__pyx_tuple__89);

  /* "svelter_sv/readme.pyx":101
 *     print('SVelter  BPIntegrate')
 *     print('')
 *     print('Required Parameters:')             # <<<<<<<<<<<<<<
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')
 */
  __pyx_tuple__90 = PyTuple_Pack(1, __pyx_kp_s_Required_Parameters); if (unlikely(!__pyx_tuple__90)) __PYX_ERR(0, 101, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__90);
  __Pyx_GIVEREF(__pyx_tuple__90);

  /* "svelter_sv/readme.pyx":102
 *     print('')
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')             # <<<<<<<<<<<<<<
 *     print('    --workdir, writable working directory.')
 *     print('')
 */
  __pyx_tuple__91 = PyTuple_Pack(1, __pyx_kp_s_sample_input_alignment_file_in); if (unlikely(!__pyx_tuple__91)) __PYX_ERR(0, 102, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__91);
  __Pyx_GIVEREF(__pyx_tuple__91);

  /* "svelter_sv/readme.pyx":103
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')             # <<<<<<<<<<<<<<
 *     print('')
 *     print('Optional Parameters:')
 */
  __pyx_tuple__92 = PyTuple_Pack(1, __pyx_kp_s_workdir_writable_working_direct); if (unlikely(!__pyx_tuple__92)) __PYX_ERR(0, 103, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__92);
  __Pyx_GIVEREF(__pyx_tuple__92);

  /* "svelter_sv/readme.pyx":104
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')
 *     print('')             # <<<<<<<<<<<<<<
 *     print('Optional Parameters:')
 *     print('    --batch, specify number of structures in each separate file (if 0, output files will be calssified by chromosomes; default, all BP clustered will be integrated in one txt file)')
 */
  __pyx_tuple__93 = PyTuple_Pack(1, __pyx_kp_s__2); if (unlikely(!__pyx_tuple__93)) __PYX_ERR(0, 104, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__93);
  __Pyx_GIVEREF(__pyx_tuple__93);

  /* "svelter_sv/readme.pyx":105
 *     print('    --workdir, writable working directory.')
 *     print('')
 *     print('Optional Parameters:')             # <<<<<<<<<<<<<<
 *     print('    --batch, specify number of structures in each separate file (if 0, output files will be calssified by chromosomes; default, all BP clustered will be integrated in one txt file)')
 *     print('    --bp-path, specify the path where all called breakpoints were kept under')
 */
  __pyx_tuple__94 = PyTuple_Pack(1, __pyx_kp_s_Optional_Parameters); if (unlikely(!__pyx_tuple__94)) __PYX_ERR(0, 105, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__94);
  __Pyx_GIVEREF(__pyx_tuple__94);

  /* "svelter_sv/readme.pyx":106
 *     print('')
 *     print('Optional Parameters:')
 *     print('    --batch, specify number of structures in each separate file (if 0, output files will be calssified by chromosomes; default, all BP clustered will be integrated in one txt file)')             # <<<<<<<<<<<<<<
 *     print('    --bp-path, specify the path where all called breakpoints were kept under')
 *     print('    --chromosome, name of chromosome to run. should match chromosome name in bam file')
 */
  __pyx_tuple__95 = PyTuple_Pack(1, __pyx_kp_s_batch_specify_number_of_structu); if (unlikely(!__pyx_tuple__95)) __PYX_ERR(0, 106, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__95);
  __Pyx_GIVEREF(__pyx_tuple__95);

  /* "svelter_sv/readme.pyx":107
 *     print('Optional Parameters:')
 *     print('    --batch, specify number of structures in each separate file (if 0, output files will be calssified by chromosomes; default, all BP clustered will be integrated in one txt file)')
 *     print('    --bp-path, specify the path where all called breakpoints were kept under')             # <<<<<<<<<<<<<<
 *     print('    --chromosome, name of chromosome to run. should match chromosome name in bam file')
 * def print_default_parameters_svpredict():
 */
  __pyx_tuple__96 = PyTuple_Pack(1, __pyx_kp_s_bp_path_specify_the_path_where); if (unlikely(!__pyx_tuple__96)) __PYX_ERR(0, 107, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__96);
  __Pyx_GIVEREF(__pyx_tuple__96);

  /* "svelter_sv/readme.pyx":108
 *     print('    --batch, specify number of structures in each separate file (if 0, output files will be calssified by chromosomes; default, all BP clustered will be integrated in one txt file)')
 *     print('    --bp-path, specify the path where all called breakpoints were kept under')
 *     print('    --chromosome, name of chromosome to run. should match chromosome name in bam file')             # <<<<<<<<<<<<<<
 * def print_default_parameters_svpredict():
 *     print('SVelter  SVPredict')
 */
  __pyx_tuple__97 = PyTuple_Pack(1, __pyx_kp_s_chromosome_name_of_chromosome_t); if (unlikely(!__pyx_tuple__97)) __PYX_ERR(0, 108, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__97);
  __Pyx_GIVEREF(__pyx_tuple__97);

  /* "svelter_sv/readme.pyx":110
 *     print('    --chromosome, name of chromosome to run. should match chromosome name in bam file')
 * def print_default_parameters_svpredict():
 *     print('SVelter  SVPredict')             # <<<<<<<<<<<<<<
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')
 */
  __pyx_tuple__98 = PyTuple_Pack(1, __pyx_kp_s_SVelter_SVPredict); if (unlikely(!__pyx_tuple__98)) __PYX_ERR(0, 110, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__98);
  __Pyx_GIVEREF(__pyx_tuple__98);

  /* "svelter_sv/readme.pyx":111
 * def print_default_parameters_svpredict():
 *     print('SVelter  SVPredict')
 *     print('Required Parameters:')             # <<<<<<<<<<<<<<
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')
 */
  __pyx_tuple__99 = PyTuple_Pack(1, __pyx_kp_s_Required_Parameters); if (unlikely(!__pyx_tuple__99)) __PYX_ERR(0, 111, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__99);
  __Pyx_GIVEREF(__pyx_tuple__99);

  /* "svelter_sv/readme.pyx":112
 *     print('SVelter  SVPredict')
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')             # <<<<<<<<<<<<<<
 *     print('    --workdir, writable working directory.')
 *     print('    --bp-file, input txt file containing clustered bps.')
 */
  __pyx_tuple__100 = PyTuple_Pack(1, __pyx_kp_s_sample_input_alignment_file_in); if (unlikely(!__pyx_tuple__100)) __PYX_ERR(0, 112, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__100);
  __Pyx_GIVEREF(__pyx_tuple__100);

  /* "svelter_sv/readme.pyx":113
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')             # <<<<<<<<<<<<<<
 *     print('    --bp-file, input txt file containing clustered bps.')
 *     print(' ')
 */
  __pyx_tuple__101 = PyTuple_Pack(1, __pyx_kp_s_workdir_writable_working_direct); if (unlikely(!__pyx_tuple__101)) __PYX_ERR(0, 113, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__101);
  __Pyx_GIVEREF(__pyx_tuple__101);

  /* "svelter_sv/readme.pyx":114
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')
 *     print('    --bp-file, input txt file containing clustered bps.')             # <<<<<<<<<<<<<<
 *     print(' ')
 *     print('Optional Parameters:')
 */
  __pyx_tuple__102 = PyTuple_Pack(1, __pyx_kp_s_bp_file_input_txt_file_containi); if (unlikely(!__pyx_tuple__102)) __PYX_ERR(0, 114, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__102);
  __Pyx_GIVEREF(__pyx_tuple__102);

  /* "svelter_sv/readme.pyx":115
 *     print('    --workdir, writable working directory.')
 *     print('    --bp-file, input txt file containing clustered bps.')
 *     print(' ')             # <<<<<<<<<<<<<<
 *     print('Optional Parameters:')
 *     print('    --num-iteration, maximum number of iterations per structure will run')
 */
  __pyx_tuple__103 = PyTuple_Pack(1, __pyx_kp_s__5); if (unlikely(!__pyx_tuple__103)) __PYX_ERR(0, 115, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__103);
  __Pyx_GIVEREF(__pyx_tuple__103);

  /* "svelter_sv/readme.pyx":116
 *     print('    --bp-file, input txt file containing clustered bps.')
 *     print(' ')
 *     print('Optional Parameters:')             # <<<<<<<<<<<<<<
 *     print('    --num-iteration, maximum number of iterations per structure will run')
 *     print('    --ploidy, limit algorithm to specific zygosity (0:heterozygous only; 1:homozygous only; 2:both; default:2)')
 */
  __pyx_tuple__104 = PyTuple_Pack(1, __pyx_kp_s_Optional_Parameters); if (unlikely(!__pyx_tuple__104)) __PYX_ERR(0, 116, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__104);
  __Pyx_GIVEREF(__pyx_tuple__104);

  /* "svelter_sv/readme.pyx":117
 *     print(' ')
 *     print('Optional Parameters:')
 *     print('    --num-iteration, maximum number of iterations per structure will run')             # <<<<<<<<<<<<<<
 *     print('    --ploidy, limit algorithm to specific zygosity (0:heterozygous only; 1:homozygous only; 2:both; default:2)')
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')
 */
  __pyx_tuple__105 = PyTuple_Pack(1, __pyx_kp_s_num_iteration_maximum_number_of); if (unlikely(!__pyx_tuple__105)) __PYX_ERR(0, 117, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__105);
  __Pyx_GIVEREF(__pyx_tuple__105);

  /* "svelter_sv/readme.pyx":118
 *     print('Optional Parameters:')
 *     print('    --num-iteration, maximum number of iterations per structure will run')
 *     print('    --ploidy, limit algorithm to specific zygosity (0:heterozygous only; 1:homozygous only; 2:both; default:2)')             # <<<<<<<<<<<<<<
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20)')
 */
  __pyx_tuple__106 = PyTuple_Pack(1, __pyx_kp_s_ploidy_limit_algorithm_to_speci); if (unlikely(!__pyx_tuple__106)) __PYX_ERR(0, 118, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__106);
  __Pyx_GIVEREF(__pyx_tuple__106);

  /* "svelter_sv/readme.pyx":119
 *     print('    --num-iteration, maximum number of iterations per structure will run')
 *     print('    --ploidy, limit algorithm to specific zygosity (0:heterozygous only; 1:homozygous only; 2:both; default:2)')
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')             # <<<<<<<<<<<<<<
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20)')
 * def print_default_parameters_svintegrate():
 */
  __pyx_tuple__107 = PyTuple_Pack(1, __pyx_kp_s_null_model_specify_which_stat_m); if (unlikely(!__pyx_tuple__107)) __PYX_ERR(0, 119, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__107);
  __Pyx_GIVEREF(__pyx_tuple__107);

  /* "svelter_sv/readme.pyx":120
 *     print('    --ploidy, limit algorithm to specific zygosity (0:heterozygous only; 1:homozygous only; 2:both; default:2)')
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20)')             # <<<<<<<<<<<<<<
 * def print_default_parameters_svintegrate():
 *     print('SVelter SVIntegrate')
 */
  __pyx_tuple__108 = PyTuple_Pack(1, __pyx_kp_s_qc_align_minimum_alignment_qual_2); if (unlikely(!__pyx_tuple__108)) __PYX_ERR(0, 120, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__108);
  __Pyx_GIVEREF(__pyx_tuple__108);

  /* "svelter_sv/readme.pyx":122
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20)')
 * def print_default_parameters_svintegrate():
 *     print('SVelter SVIntegrate')             # <<<<<<<<<<<<<<
 *     print(' ')
 *     print('Required Parameters:')
 */
  __pyx_tuple__109 = PyTuple_Pack(1, __pyx_kp_s_SVelter_SVIntegrate); if (unlikely(!__pyx_tuple__109)) __PYX_ERR(0, 122, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__109);
  __Pyx_GIVEREF(__pyx_tuple__109);

  /* "svelter_sv/readme.pyx":123
 * def print_default_parameters_svintegrate():
 *     print('SVelter SVIntegrate')
 *     print(' ')             # <<<<<<<<<<<<<<
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')
 */
  __pyx_tuple__110 = PyTuple_Pack(1, __pyx_kp_s__5); if (unlikely(!__pyx_tuple__110)) __PYX_ERR(0, 123, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__110);
  __Pyx_GIVEREF(__pyx_tuple__110);

  /* "svelter_sv/readme.pyx":124
 *     print('SVelter SVIntegrate')
 *     print(' ')
 *     print('Required Parameters:')             # <<<<<<<<<<<<<<
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')
 */
  __pyx_tuple__111 = PyTuple_Pack(1, __pyx_kp_s_Required_Parameters); if (unlikely(!__pyx_tuple__111)) __PYX_ERR(0, 124, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__111);
  __Pyx_GIVEREF(__pyx_tuple__111);

  /* "svelter_sv/readme.pyx":125
 *     print(' ')
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')             # <<<<<<<<<<<<<<
 *     print('    --workdir, writable working directory.')
 *     print('    --input-path, path of .coverage files produced by SVelter SVPredict')
 */
  __pyx_tuple__112 = PyTuple_Pack(1, __pyx_kp_s_sample_input_alignment_file_in); if (unlikely(!__pyx_tuple__112)) __PYX_ERR(0, 125, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__112);
  __Pyx_GIVEREF(__pyx_tuple__112);

  /* "svelter_sv/readme.pyx":126
 *     print('Required Parameters:')
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')             # <<<<<<<<<<<<<<
 *     print('    --input-path, path of .coverage files produced by SVelter SVPredict')
 *     print(' ')
 */
  __pyx_tuple__113 = PyTuple_Pack(1, __pyx_kp_s_workdir_writable_working_direct); if (unlikely(!__pyx_tuple__113)) __PYX_ERR(0, 126, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__113);
  __Pyx_GIVEREF(__pyx_tuple__113);

  /* "svelter_sv/readme.pyx":127
 *     print('    --sample, input alignment file in bam format')
 *     print('    --workdir, writable working directory.')
 *     print('    --input-path, path of .coverage files produced by SVelter SVPredict')             # <<<<<<<<<<<<<<
 *     print(' ')
 *     print('Optional Parameters:')
 */
  __pyx_tuple__114 = PyTuple_Pack(1, __pyx_kp_s_input_path_path_of_coverage_fil); if (unlikely(!__pyx_tuple__114)) __PYX_ERR(0, 127, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__114);
  __Pyx_GIVEREF(__pyx_tuple__114);

  /* "svelter_sv/readme.pyx":128
 *     print('    --workdir, writable working directory.')
 *     print('    --input-path, path of .coverage files produced by SVelter SVPredict')
 *     print(' ')             # <<<<<<<<<<<<<<
 *     print('Optional Parameters:')
 *     print('    --prefix, output prefix for vcf and svelter files')
 */
  __pyx_tuple__115 = PyTuple_Pack(1, __pyx_kp_s__5); if (unlikely(!__pyx_tuple__115)) __PYX_ERR(0, 128, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__115);
  __Pyx_GIVEREF(__pyx_tuple__115);

  /* "svelter_sv/readme.pyx":129
 *     print('    --input-path, path of .coverage files produced by SVelter SVPredict')
 *     print(' ')
 *     print('Optional Parameters:')             # <<<<<<<<<<<<<<
 *     print('    --prefix, output prefix for vcf and svelter files')
 *     print('    --input-path, path of .coverage files produced by SVelter SVPredict')
 */
  __pyx_tuple__116 = PyTuple_Pack(1, __pyx_kp_s_Optional_Parameters); if (unlikely(!__pyx_tuple__116)) __PYX_ERR(0, 129, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__116);
  __Pyx_GIVEREF(__pyx_tuple__116);

  /* "svelter_sv/readme.pyx":130
 *     print(' ')
 *     print('Optional Parameters:')
 *     print('    --prefix, output prefix for vcf and svelter files')             # <<<<<<<<<<<<<<
 *     print('    --input-path, path of .coverage files produced by SVelter SVPredict')
 *     print('    --qc-structure, minimum quality score of a resolved structure to be considered as PASS and included in the output vcf file')
 */
  __pyx_tuple__117 = PyTuple_Pack(1, __pyx_kp_s_prefix_output_prefix_for_vcf_an_2); if (unlikely(!__pyx_tuple__117)) __PYX_ERR(0, 130, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__117);
  __Pyx_GIVEREF(__pyx_tuple__117);

  /* "svelter_sv/readme.pyx":131
 *     print('Optional Parameters:')
 *     print('    --prefix, output prefix for vcf and svelter files')
 *     print('    --input-path, path of .coverage files produced by SVelter SVPredict')             # <<<<<<<<<<<<<<
 *     print('    --qc-structure, minimum quality score of a resolved structure to be considered as PASS and included in the output vcf file')
 */
  __pyx_tuple__118 = PyTuple_Pack(1, __pyx_kp_s_input_path_path_of_coverage_fil); if (unlikely(!__pyx_tuple__118)) __PYX_ERR(0, 131, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__118);
  __Pyx_GIVEREF(__pyx_tuple__118);

  /* "svelter_sv/readme.pyx":132
 *     print('    --prefix, output prefix for vcf and svelter files')
 *     print('    --input-path, path of .coverage files produced by SVelter SVPredict')
 *     print('    --qc-structure, minimum quality score of a resolved structure to be considered as PASS and included in the output vcf file')             # <<<<<<<<<<<<<<
 */
  __pyx_tuple__119 = PyTuple_Pack(1, __pyx_kp_s_qc_structure_minimum_quality_sc); if (unlikely(!__pyx_tuple__119)) __PYX_ERR(0, 132, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_tuple__119);
  __Pyx_GIVEREF(__pyx_tuple__119);

  /* "svelter_sv/readme.pyx":2
 * from __future__ import print_function
 * def print_default_parameters():             # <<<<<<<<<<<<<<
 *     print('SVelter-0.1        Contact: xuefzhao@umich.edu      Last Update:2015-08-20')
 *     print('')
 */
  __pyx_codeobj__120 = (PyObject*)__Pyx_PyCode_New(0, 0, 0, 0, 0, __pyx_empty_bytes, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_kp_s_data_talkowski_xuefang_local_sr, __pyx_n_s_print_default_parameters, 2, __pyx_empty_bytes); if (unlikely(!__pyx_codeobj__120)) __PYX_ERR(0, 2, __pyx_L1_error)

  /* "svelter_sv/readme.pyx":19
 *     print(' ')
 *     print('Type in SVelter [options] for detailed parameters')
 * def print_default_parameters_predefinedbp():             # <<<<<<<<<<<<<<
 *     print('SVelter PredefinedBP')
 *     print(' ')
 */
  __pyx_codeobj__121 = (PyObject*)__Pyx_PyCode_New(0, 0, 0, 0, 0, __pyx_empty_bytes, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_kp_s_data_talkowski_xuefang_local_sr, __pyx_n_s_print_default_parameters_predefi, 19, __pyx_empty_bytes); if (unlikely(!__pyx_codeobj__121)) __PYX_ERR(0, 19, __pyx_L1_error)

  /* "svelter_sv/readme.pyx":45
 *     print('                     ftp://hgdownload.cse.ucsc.edu/goldenPath/currentGenomes/Homo_sapiens/encodeDCC/wgEncodeMapability/')
 *     print('    --qc-map-cutoff, the minimum mapping quality required for a breakpoint to be reported [0.0]')
 * def print_default_parameters_clean():             # <<<<<<<<<<<<<<
 *     print('SVelter Clean')
 *     print('')
 */
  __pyx_codeobj__122 = (PyObject*)__Pyx_PyCode_New(0, 0, 0, 0, 0, __pyx_empty_bytes, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_kp_s_data_talkowski_xuefang_local_sr, __pyx_n_s_print_default_parameters_clean, 45, __pyx_empty_bytes); if (unlikely(!__pyx_codeobj__122)) __PYX_ERR(0, 45, __pyx_L1_error)

  /* "svelter_sv/readme.pyx":50
 *     print('Required Parameters:')
 *     print('    --workdir, writable working directory.')
 * def print_default_parameters_setup():             # <<<<<<<<<<<<<<
 *     print('SVelter Setup')
 *     print('')
 */
  __pyx_codeobj__123 = (PyObject*)__Pyx_PyCode_New(0, 0, 0, 0, 0, __pyx_empty_bytes, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_kp_s_data_talkowski_xuefang_local_sr, __pyx_n_s_print_default_parameters_setup, 50, __pyx_empty_bytes); if (unlikely(!__pyx_codeobj__123)) __PYX_ERR(0, 50, __pyx_L1_error)

  /* "svelter_sv/readme.pyx":65
 *     #print '    --segdup, absolute path of bed file indicating segmental duplications in genome, that would be excluded from analysis'
 *     print('    --ref-index, folders containin pre-indexed files, if applicable. For certain versions of human genome, the indexed files are availabel from https://github.com/mills-lab/svelter.')
 * def print_default_parameters_nullmodel():             # <<<<<<<<<<<<<<
 *     print('SVelter NullModel')
 *     print('')
 */
  __pyx_codeobj__124 = (PyObject*)__Pyx_PyCode_New(0, 0, 0, 0, 0, __pyx_empty_bytes, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_kp_s_data_talkowski_xuefang_local_sr, __pyx_n_s_print_default_parameters_nullmod, 65, __pyx_empty_bytes); if (unlikely(!__pyx_codeobj__124)) __PYX_ERR(0, 65, __pyx_L1_error)

  /* "svelter_sv/readme.pyx":82
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20]')
 *     print('    --qc-split, minimum alighment of clipped parts of reads considered as a soft clip [20]')
 * def print_default_parameters_bpsearch():             # <<<<<<<<<<<<<<
 *     print('SVelter  BPSearch')
 *     print('')
 */
  __pyx_codeobj__125 = (PyObject*)__Pyx_PyCode_New(0, 0, 0, 0, 0, __pyx_empty_bytes, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_kp_s_data_talkowski_xuefang_local_sr, __pyx_n_s_print_default_parameters_bpsearc, 82, __pyx_empty_bytes); if (unlikely(!__pyx_codeobj__125)) __PYX_ERR(0, 82, __pyx_L1_error)

  /* "svelter_sv/readme.pyx":98
 *     print('    --qc-map-file, .bigWig file used to decide local genomic mappability, avaliable from: ftp://hgdownload.cse.ucsc.edu/goldenPath/currentGenomes/Homo_sapiens/encodeDCC/wgEncodeMapability/')
 *     print('    --qc-map-cutoff, the minimum mapping quality required for a breakpoint to be reported [0.0)')
 * def print_default_parameters_bpintegrate():             # <<<<<<<<<<<<<<
 *     print('SVelter  BPIntegrate')
 *     print('')
 */
  __pyx_codeobj__126 = (PyObject*)__Pyx_PyCode_New(0, 0, 0, 0, 0, __pyx_empty_bytes, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_kp_s_data_talkowski_xuefang_local_sr, __pyx_n_s_print_default_parameters_bpinteg, 98, __pyx_empty_bytes); if (unlikely(!__pyx_codeobj__126)) __PYX_ERR(0, 98, __pyx_L1_error)

  /* "svelter_sv/readme.pyx":109
 *     print('    --bp-path, specify the path where all called breakpoints were kept under')
 *     print('    --chromosome, name of chromosome to run. should match chromosome name in bam file')
 * def print_default_parameters_svpredict():             # <<<<<<<<<<<<<<
 *     print('SVelter  SVPredict')
 *     print('Required Parameters:')
 */
  __pyx_codeobj__127 = (PyObject*)__Pyx_PyCode_New(0, 0, 0, 0, 0, __pyx_empty_bytes, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_kp_s_data_talkowski_xuefang_local_sr, __pyx_n_s_print_default_parameters_svpredi, 109, __pyx_empty_bytes); if (unlikely(!__pyx_codeobj__127)) __PYX_ERR(0, 109, __pyx_L1_error)

  /* "svelter_sv/readme.pyx":121
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20)')
 * def print_default_parameters_svintegrate():             # <<<<<<<<<<<<<<
 *     print('SVelter SVIntegrate')
 *     print(' ')
 */
  __pyx_codeobj__128 = (PyObject*)__Pyx_PyCode_New(0, 0, 0, 0, 0, __pyx_empty_bytes, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_kp_s_data_talkowski_xuefang_local_sr, __pyx_n_s_print_default_parameters_svinteg, 121, __pyx_empty_bytes); if (unlikely(!__pyx_codeobj__128)) __PYX_ERR(0, 121, __pyx_L1_error)
  __Pyx_RefNannyFinishContext();
  return 0;
  __pyx_L1_error:;
  __Pyx_RefNannyFinishContext();
  return -1;
}

static int __Pyx_InitGlobals(void) {
  if (__Pyx_InitStrings(__pyx_string_tab) < 0) __PYX_ERR(0, 1, __pyx_L1_error);
  return 0;
  __pyx_L1_error:;
  return -1;
}

#if PY_MAJOR_VERSION < 3
PyMODINIT_FUNC initreadme(void); /*proto*/
PyMODINIT_FUNC initreadme(void)
#else
PyMODINIT_FUNC PyInit_readme(void); /*proto*/
PyMODINIT_FUNC PyInit_readme(void)
#endif
{
  PyObject *__pyx_t_1 = NULL;
  __Pyx_RefNannyDeclarations
  #if CYTHON_REFNANNY
  __Pyx_RefNanny = __Pyx_RefNannyImportAPI("refnanny");
  if (!__Pyx_RefNanny) {
      PyErr_Clear();
      __Pyx_RefNanny = __Pyx_RefNannyImportAPI("Cython.Runtime.refnanny");
      if (!__Pyx_RefNanny)
          Py_FatalError("failed to import 'refnanny' module");
  }
  #endif
  __Pyx_RefNannySetupContext("PyMODINIT_FUNC PyInit_readme(void)", 0);
  if (__Pyx_check_binary_version() < 0) __PYX_ERR(0, 1, __pyx_L1_error)
  __pyx_empty_tuple = PyTuple_New(0); if (unlikely(!__pyx_empty_tuple)) __PYX_ERR(0, 1, __pyx_L1_error)
  __pyx_empty_bytes = PyBytes_FromStringAndSize("", 0); if (unlikely(!__pyx_empty_bytes)) __PYX_ERR(0, 1, __pyx_L1_error)
  __pyx_empty_unicode = PyUnicode_FromStringAndSize("", 0); if (unlikely(!__pyx_empty_unicode)) __PYX_ERR(0, 1, __pyx_L1_error)
  #ifdef __Pyx_CyFunction_USED
  if (__pyx_CyFunction_init() < 0) __PYX_ERR(0, 1, __pyx_L1_error)
  #endif
  #ifdef __Pyx_FusedFunction_USED
  if (__pyx_FusedFunction_init() < 0) __PYX_ERR(0, 1, __pyx_L1_error)
  #endif
  #ifdef __Pyx_Coroutine_USED
  if (__pyx_Coroutine_init() < 0) __PYX_ERR(0, 1, __pyx_L1_error)
  #endif
  #ifdef __Pyx_Generator_USED
  if (__pyx_Generator_init() < 0) __PYX_ERR(0, 1, __pyx_L1_error)
  #endif
  #ifdef __Pyx_StopAsyncIteration_USED
  if (__pyx_StopAsyncIteration_init() < 0) __PYX_ERR(0, 1, __pyx_L1_error)
  #endif
  /*--- Library function declarations ---*/
  /*--- Threads initialization code ---*/
  #if defined(__PYX_FORCE_INIT_THREADS) && __PYX_FORCE_INIT_THREADS
  #ifdef WITH_THREAD /* Python build with threading support? */
  PyEval_InitThreads();
  #endif
  #endif
  /*--- Module creation code ---*/
  #if PY_MAJOR_VERSION < 3
  __pyx_m = Py_InitModule4("readme", __pyx_methods, 0, 0, PYTHON_API_VERSION); Py_XINCREF(__pyx_m);
  #else
  __pyx_m = PyModule_Create(&__pyx_moduledef);
  #endif
  if (unlikely(!__pyx_m)) __PYX_ERR(0, 1, __pyx_L1_error)
  __pyx_d = PyModule_GetDict(__pyx_m); if (unlikely(!__pyx_d)) __PYX_ERR(0, 1, __pyx_L1_error)
  Py_INCREF(__pyx_d);
  __pyx_b = PyImport_AddModule(__Pyx_BUILTIN_MODULE_NAME); if (unlikely(!__pyx_b)) __PYX_ERR(0, 1, __pyx_L1_error)
  #if CYTHON_COMPILING_IN_PYPY
  Py_INCREF(__pyx_b);
  #endif
  if (PyObject_SetAttrString(__pyx_m, "__builtins__", __pyx_b) < 0) __PYX_ERR(0, 1, __pyx_L1_error);
  /*--- Initialize various global constants etc. ---*/
  if (__Pyx_InitGlobals() < 0) __PYX_ERR(0, 1, __pyx_L1_error)
  #if PY_MAJOR_VERSION < 3 && (__PYX_DEFAULT_STRING_ENCODING_IS_ASCII || __PYX_DEFAULT_STRING_ENCODING_IS_DEFAULT)
  if (__Pyx_init_sys_getdefaultencoding_params() < 0) __PYX_ERR(0, 1, __pyx_L1_error)
  #endif
  if (__pyx_module_is_main_svelter_sv__readme) {
    if (PyObject_SetAttrString(__pyx_m, "__name__", __pyx_n_s_main) < 0) __PYX_ERR(0, 1, __pyx_L1_error)
  }
  #if PY_MAJOR_VERSION >= 3
  {
    PyObject *modules = PyImport_GetModuleDict(); if (unlikely(!modules)) __PYX_ERR(0, 1, __pyx_L1_error)
    if (!PyDict_GetItemString(modules, "svelter_sv.readme")) {
      if (unlikely(PyDict_SetItemString(modules, "svelter_sv.readme", __pyx_m) < 0)) __PYX_ERR(0, 1, __pyx_L1_error)
    }
  }
  #endif
  /*--- Builtin init code ---*/
  if (__Pyx_InitCachedBuiltins() < 0) __PYX_ERR(0, 1, __pyx_L1_error)
  /*--- Constants init code ---*/
  if (__Pyx_InitCachedConstants() < 0) __PYX_ERR(0, 1, __pyx_L1_error)
  /*--- Global init code ---*/
  /*--- Variable export code ---*/
  /*--- Function export code ---*/
  /*--- Type init code ---*/
  /*--- Type import code ---*/
  /*--- Variable import code ---*/
  /*--- Function import code ---*/
  /*--- Execution code ---*/
  #if defined(__Pyx_Generator_USED) || defined(__Pyx_Coroutine_USED)
  if (__Pyx_patch_abc() < 0) __PYX_ERR(0, 1, __pyx_L1_error)
  #endif

  /* "svelter_sv/readme.pyx":2
 * from __future__ import print_function
 * def print_default_parameters():             # <<<<<<<<<<<<<<
 *     print('SVelter-0.1        Contact: xuefzhao@umich.edu      Last Update:2015-08-20')
 *     print('')
 */
  __pyx_t_1 = PyCFunction_NewEx(&__pyx_mdef_10svelter_sv_6readme_1print_default_parameters, NULL, __pyx_n_s_svelter_sv_readme); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 2, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  if (PyDict_SetItem(__pyx_d, __pyx_n_s_print_default_parameters, __pyx_t_1) < 0) __PYX_ERR(0, 2, __pyx_L1_error)
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":19
 *     print(' ')
 *     print('Type in SVelter [options] for detailed parameters')
 * def print_default_parameters_predefinedbp():             # <<<<<<<<<<<<<<
 *     print('SVelter PredefinedBP')
 *     print(' ')
 */
  __pyx_t_1 = PyCFunction_NewEx(&__pyx_mdef_10svelter_sv_6readme_3print_default_parameters_predefinedbp, NULL, __pyx_n_s_svelter_sv_readme); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 19, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  if (PyDict_SetItem(__pyx_d, __pyx_n_s_print_default_parameters_predefi, __pyx_t_1) < 0) __PYX_ERR(0, 19, __pyx_L1_error)
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":45
 *     print('                     ftp://hgdownload.cse.ucsc.edu/goldenPath/currentGenomes/Homo_sapiens/encodeDCC/wgEncodeMapability/')
 *     print('    --qc-map-cutoff, the minimum mapping quality required for a breakpoint to be reported [0.0]')
 * def print_default_parameters_clean():             # <<<<<<<<<<<<<<
 *     print('SVelter Clean')
 *     print('')
 */
  __pyx_t_1 = PyCFunction_NewEx(&__pyx_mdef_10svelter_sv_6readme_5print_default_parameters_clean, NULL, __pyx_n_s_svelter_sv_readme); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 45, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  if (PyDict_SetItem(__pyx_d, __pyx_n_s_print_default_parameters_clean, __pyx_t_1) < 0) __PYX_ERR(0, 45, __pyx_L1_error)
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":50
 *     print('Required Parameters:')
 *     print('    --workdir, writable working directory.')
 * def print_default_parameters_setup():             # <<<<<<<<<<<<<<
 *     print('SVelter Setup')
 *     print('')
 */
  __pyx_t_1 = PyCFunction_NewEx(&__pyx_mdef_10svelter_sv_6readme_7print_default_parameters_setup, NULL, __pyx_n_s_svelter_sv_readme); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 50, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  if (PyDict_SetItem(__pyx_d, __pyx_n_s_print_default_parameters_setup, __pyx_t_1) < 0) __PYX_ERR(0, 50, __pyx_L1_error)
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":65
 *     #print '    --segdup, absolute path of bed file indicating segmental duplications in genome, that would be excluded from analysis'
 *     print('    --ref-index, folders containin pre-indexed files, if applicable. For certain versions of human genome, the indexed files are availabel from https://github.com/mills-lab/svelter.')
 * def print_default_parameters_nullmodel():             # <<<<<<<<<<<<<<
 *     print('SVelter NullModel')
 *     print('')
 */
  __pyx_t_1 = PyCFunction_NewEx(&__pyx_mdef_10svelter_sv_6readme_9print_default_parameters_nullmodel, NULL, __pyx_n_s_svelter_sv_readme); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 65, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  if (PyDict_SetItem(__pyx_d, __pyx_n_s_print_default_parameters_nullmod, __pyx_t_1) < 0) __PYX_ERR(0, 65, __pyx_L1_error)
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":82
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20]')
 *     print('    --qc-split, minimum alighment of clipped parts of reads considered as a soft clip [20]')
 * def print_default_parameters_bpsearch():             # <<<<<<<<<<<<<<
 *     print('SVelter  BPSearch')
 *     print('')
 */
  __pyx_t_1 = PyCFunction_NewEx(&__pyx_mdef_10svelter_sv_6readme_11print_default_parameters_bpsearch, NULL, __pyx_n_s_svelter_sv_readme); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 82, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  if (PyDict_SetItem(__pyx_d, __pyx_n_s_print_default_parameters_bpsearc, __pyx_t_1) < 0) __PYX_ERR(0, 82, __pyx_L1_error)
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":98
 *     print('    --qc-map-file, .bigWig file used to decide local genomic mappability, avaliable from: ftp://hgdownload.cse.ucsc.edu/goldenPath/currentGenomes/Homo_sapiens/encodeDCC/wgEncodeMapability/')
 *     print('    --qc-map-cutoff, the minimum mapping quality required for a breakpoint to be reported [0.0)')
 * def print_default_parameters_bpintegrate():             # <<<<<<<<<<<<<<
 *     print('SVelter  BPIntegrate')
 *     print('')
 */
  __pyx_t_1 = PyCFunction_NewEx(&__pyx_mdef_10svelter_sv_6readme_13print_default_parameters_bpintegrate, NULL, __pyx_n_s_svelter_sv_readme); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 98, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  if (PyDict_SetItem(__pyx_d, __pyx_n_s_print_default_parameters_bpinteg, __pyx_t_1) < 0) __PYX_ERR(0, 98, __pyx_L1_error)
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":109
 *     print('    --bp-path, specify the path where all called breakpoints were kept under')
 *     print('    --chromosome, name of chromosome to run. should match chromosome name in bam file')
 * def print_default_parameters_svpredict():             # <<<<<<<<<<<<<<
 *     print('SVelter  SVPredict')
 *     print('Required Parameters:')
 */
  __pyx_t_1 = PyCFunction_NewEx(&__pyx_mdef_10svelter_sv_6readme_15print_default_parameters_svpredict, NULL, __pyx_n_s_svelter_sv_readme); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 109, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  if (PyDict_SetItem(__pyx_d, __pyx_n_s_print_default_parameters_svpredi, __pyx_t_1) < 0) __PYX_ERR(0, 109, __pyx_L1_error)
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":121
 *     print('    --null-model, specify which stat model to be fitted on each parameter. if --null-model==C / Complex, negative bimodal distribution will be fitted to insertlenth; else, normal will be used')
 *     print('    --qc-align, minimum alignment quality required for mapped reads in bam file [20)')
 * def print_default_parameters_svintegrate():             # <<<<<<<<<<<<<<
 *     print('SVelter SVIntegrate')
 *     print(' ')
 */
  __pyx_t_1 = PyCFunction_NewEx(&__pyx_mdef_10svelter_sv_6readme_17print_default_parameters_svintegrate, NULL, __pyx_n_s_svelter_sv_readme); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 121, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  if (PyDict_SetItem(__pyx_d, __pyx_n_s_print_default_parameters_svinteg, __pyx_t_1) < 0) __PYX_ERR(0, 121, __pyx_L1_error)
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /* "svelter_sv/readme.pyx":1
 * from __future__ import print_function             # <<<<<<<<<<<<<<
 * def print_default_parameters():
 *     print('SVelter-0.1        Contact: xuefzhao@umich.edu      Last Update:2015-08-20')
 */
  __pyx_t_1 = PyDict_New(); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 1, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  if (PyDict_SetItem(__pyx_d, __pyx_n_s_test, __pyx_t_1) < 0) __PYX_ERR(0, 1, __pyx_L1_error)
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;

  /*--- Wrapped vars code ---*/

  goto __pyx_L0;
  __pyx_L1_error:;
  __Pyx_XDECREF(__pyx_t_1);
  if (__pyx_m) {
    if (__pyx_d) {
      __Pyx_AddTraceback("init svelter_sv.readme", __pyx_clineno, __pyx_lineno, __pyx_filename);
    }
    Py_DECREF(__pyx_m); __pyx_m = 0;
  } else if (!PyErr_Occurred()) {
    PyErr_SetString(PyExc_ImportError, "init svelter_sv.readme");
  }
  __pyx_L0:;
  __Pyx_RefNannyFinishContext();
  #if PY_MAJOR_VERSION < 3
  return;
  #else
  return __pyx_m;
  #endif
}

/* --- Runtime support code --- */
/* Refnanny */
#if CYTHON_REFNANNY
static __Pyx_RefNannyAPIStruct *__Pyx_RefNannyImportAPI(const char *modname) {
    PyObject *m = NULL, *p = NULL;
    void *r = NULL;
    m = PyImport_ImportModule((char *)modname);
    if (!m) goto end;
    p = PyObject_GetAttrString(m, (char *)"RefNannyAPI");
    if (!p) goto end;
    r = PyLong_AsVoidPtr(p);
end:
    Py_XDECREF(p);
    Py_XDECREF(m);
    return (__Pyx_RefNannyAPIStruct *)r;
}
#endif

/* GetBuiltinName */
static PyObject *__Pyx_GetBuiltinName(PyObject *name) {
    PyObject* result = __Pyx_PyObject_GetAttrStr(__pyx_b, name);
    if (unlikely(!result)) {
        PyErr_Format(PyExc_NameError,
#if PY_MAJOR_VERSION >= 3
            "name '%U' is not defined", name);
#else
            "name '%.200s' is not defined", PyString_AS_STRING(name));
#endif
    }
    return result;
}

/* PyObjectCall */
#if CYTHON_COMPILING_IN_CPYTHON
static CYTHON_INLINE PyObject* __Pyx_PyObject_Call(PyObject *func, PyObject *arg, PyObject *kw) {
    PyObject *result;
    ternaryfunc call = func->ob_type->tp_call;
    if (unlikely(!call))
        return PyObject_Call(func, arg, kw);
    if (unlikely(Py_EnterRecursiveCall((char*)" while calling a Python object")))
        return NULL;
    result = (*call)(func, arg, kw);
    Py_LeaveRecursiveCall();
    if (unlikely(!result) && unlikely(!PyErr_Occurred())) {
        PyErr_SetString(
            PyExc_SystemError,
            "NULL result without error in PyObject_Call");
    }
    return result;
}
#endif

/* CodeObjectCache */
static int __pyx_bisect_code_objects(__Pyx_CodeObjectCacheEntry* entries, int count, int code_line) {
    int start = 0, mid = 0, end = count - 1;
    if (end >= 0 && code_line > entries[end].code_line) {
        return count;
    }
    while (start < end) {
        mid = start + (end - start) / 2;
        if (code_line < entries[mid].code_line) {
            end = mid;
        } else if (code_line > entries[mid].code_line) {
             start = mid + 1;
        } else {
            return mid;
        }
    }
    if (code_line <= entries[mid].code_line) {
        return mid;
    } else {
        return mid + 1;
    }
}
static PyCodeObject *__pyx_find_code_object(int code_line) {
    PyCodeObject* code_object;
    int pos;
    if (unlikely(!code_line) || unlikely(!__pyx_code_cache.entries)) {
        return NULL;
    }
    pos = __pyx_bisect_code_objects(__pyx_code_cache.entries, __pyx_code_cache.count, code_line);
    if (unlikely(pos >= __pyx_code_cache.count) || unlikely(__pyx_code_cache.entries[pos].code_line != code_line)) {
        return NULL;
    }
    code_object = __pyx_code_cache.entries[pos].code_object;
    Py_INCREF(code_object);
    return code_object;
}
static void __pyx_insert_code_object(int code_line, PyCodeObject* code_object) {
    int pos, i;
    __Pyx_CodeObjectCacheEntry* entries = __pyx_code_cache.entries;
    if (unlikely(!code_line)) {
        return;
    }
    if (unlikely(!entries)) {
        entries = (__Pyx_CodeObjectCacheEntry*)PyMem_Malloc(64*sizeof(__Pyx_CodeObjectCacheEntry));
        if (likely(entries)) {
            __pyx_code_cache.entries = entries;
            __pyx_code_cache.max_count = 64;
            __pyx_code_cache.count = 1;
            entries[0].code_line = code_line;
            entries[0].code_object = code_object;
            Py_INCREF(code_object);
        }
        return;
    }
    pos = __pyx_bisect_code_objects(__pyx_code_cache.entries, __pyx_code_cache.count, code_line);
    if ((pos < __pyx_code_cache.count) && unlikely(__pyx_code_cache.entries[pos].code_line == code_line)) {
        PyCodeObject* tmp = entries[pos].code_object;
        entries[pos].code_object = code_object;
        Py_DECREF(tmp);
        return;
    }
    if (__pyx_code_cache.count == __pyx_code_cache.max_count) {
        int new_max = __pyx_code_cache.max_count + 64;
        entries = (__Pyx_CodeObjectCacheEntry*)PyMem_Realloc(
            __pyx_code_cache.entries, (size_t)new_max*sizeof(__Pyx_CodeObjectCacheEntry));
        if (unlikely(!entries)) {
            return;
        }
        __pyx_code_cache.entries = entries;
        __pyx_code_cache.max_count = new_max;
    }
    for (i=__pyx_code_cache.count; i>pos; i--) {
        entries[i] = entries[i-1];
    }
    entries[pos].code_line = code_line;
    entries[pos].code_object = code_object;
    __pyx_code_cache.count++;
    Py_INCREF(code_object);
}

/* AddTraceback */
#include "compile.h"
#include "frameobject.h"
#include "traceback.h"
static PyCodeObject* __Pyx_CreateCodeObjectForTraceback(
            const char *funcname, int c_line,
            int py_line, const char *filename) {
    PyCodeObject *py_code = 0;
    PyObject *py_srcfile = 0;
    PyObject *py_funcname = 0;
    #if PY_MAJOR_VERSION < 3
    py_srcfile = PyString_FromString(filename);
    #else
    py_srcfile = PyUnicode_FromString(filename);
    #endif
    if (!py_srcfile) goto bad;
    if (c_line) {
        #if PY_MAJOR_VERSION < 3
        py_funcname = PyString_FromFormat( "%s (%s:%d)", funcname, __pyx_cfilenm, c_line);
        #else
        py_funcname = PyUnicode_FromFormat( "%s (%s:%d)", funcname, __pyx_cfilenm, c_line);
        #endif
    }
    else {
        #if PY_MAJOR_VERSION < 3
        py_funcname = PyString_FromString(funcname);
        #else
        py_funcname = PyUnicode_FromString(funcname);
        #endif
    }
    if (!py_funcname) goto bad;
    py_code = __Pyx_PyCode_New(
        0,
        0,
        0,
        0,
        0,
        __pyx_empty_bytes, /*PyObject *code,*/
        __pyx_empty_tuple, /*PyObject *consts,*/
        __pyx_empty_tuple, /*PyObject *names,*/
        __pyx_empty_tuple, /*PyObject *varnames,*/
        __pyx_empty_tuple, /*PyObject *freevars,*/
        __pyx_empty_tuple, /*PyObject *cellvars,*/
        py_srcfile,   /*PyObject *filename,*/
        py_funcname,  /*PyObject *name,*/
        py_line,
        __pyx_empty_bytes  /*PyObject *lnotab*/
    );
    Py_DECREF(py_srcfile);
    Py_DECREF(py_funcname);
    return py_code;
bad:
    Py_XDECREF(py_srcfile);
    Py_XDECREF(py_funcname);
    return NULL;
}
static void __Pyx_AddTraceback(const char *funcname, int c_line,
                               int py_line, const char *filename) {
    PyCodeObject *py_code = 0;
    PyFrameObject *py_frame = 0;
    py_code = __pyx_find_code_object(c_line ? c_line : py_line);
    if (!py_code) {
        py_code = __Pyx_CreateCodeObjectForTraceback(
            funcname, c_line, py_line, filename);
        if (!py_code) goto bad;
        __pyx_insert_code_object(c_line ? c_line : py_line, py_code);
    }
    py_frame = PyFrame_New(
        PyThreadState_GET(), /*PyThreadState *tstate,*/
        py_code,             /*PyCodeObject *code,*/
        __pyx_d,      /*PyObject *globals,*/
        0                    /*PyObject *locals*/
    );
    if (!py_frame) goto bad;
    __Pyx_PyFrame_SetLineNumber(py_frame, py_line);
    PyTraceBack_Here(py_frame);
bad:
    Py_XDECREF(py_code);
    Py_XDECREF(py_frame);
}

/* CIntToPy */
static CYTHON_INLINE PyObject* __Pyx_PyInt_From_long(long value) {
    const long neg_one = (long) -1, const_zero = (long) 0;
    const int is_unsigned = neg_one > const_zero;
    if (is_unsigned) {
        if (sizeof(long) < sizeof(long)) {
            return PyInt_FromLong((long) value);
        } else if (sizeof(long) <= sizeof(unsigned long)) {
            return PyLong_FromUnsignedLong((unsigned long) value);
#ifdef HAVE_LONG_LONG
        } else if (sizeof(long) <= sizeof(unsigned PY_LONG_LONG)) {
            return PyLong_FromUnsignedLongLong((unsigned PY_LONG_LONG) value);
#endif
        }
    } else {
        if (sizeof(long) <= sizeof(long)) {
            return PyInt_FromLong((long) value);
#ifdef HAVE_LONG_LONG
        } else if (sizeof(long) <= sizeof(PY_LONG_LONG)) {
            return PyLong_FromLongLong((PY_LONG_LONG) value);
#endif
        }
    }
    {
        int one = 1; int little = (int)*(unsigned char *)&one;
        unsigned char *bytes = (unsigned char *)&value;
        return _PyLong_FromByteArray(bytes, sizeof(long),
                                     little, !is_unsigned);
    }
}

/* CIntFromPyVerify */
#define __PYX_VERIFY_RETURN_INT(target_type, func_type, func_value)\
    __PYX__VERIFY_RETURN_INT(target_type, func_type, func_value, 0)
#define __PYX_VERIFY_RETURN_INT_EXC(target_type, func_type, func_value)\
    __PYX__VERIFY_RETURN_INT(target_type, func_type, func_value, 1)
#define __PYX__VERIFY_RETURN_INT(target_type, func_type, func_value, exc)\
    {\
        func_type value = func_value;\
        if (sizeof(target_type) < sizeof(func_type)) {\
            if (unlikely(value != (func_type) (target_type) value)) {\
                func_type zero = 0;\
                if (exc && unlikely(value == (func_type)-1 && PyErr_Occurred()))\
                    return (target_type) -1;\
                if (is_unsigned && unlikely(value < zero))\
                    goto raise_neg_overflow;\
                else\
                    goto raise_overflow;\
            }\
        }\
        return (target_type) value;\
    }

/* CIntFromPy */
static CYTHON_INLINE long __Pyx_PyInt_As_long(PyObject *x) {
    const long neg_one = (long) -1, const_zero = (long) 0;
    const int is_unsigned = neg_one > const_zero;
#if PY_MAJOR_VERSION < 3
    if (likely(PyInt_Check(x))) {
        if (sizeof(long) < sizeof(long)) {
            __PYX_VERIFY_RETURN_INT(long, long, PyInt_AS_LONG(x))
        } else {
            long val = PyInt_AS_LONG(x);
            if (is_unsigned && unlikely(val < 0)) {
                goto raise_neg_overflow;
            }
            return (long) val;
        }
    } else
#endif
    if (likely(PyLong_Check(x))) {
        if (is_unsigned) {
#if CYTHON_USE_PYLONG_INTERNALS
            const digit* digits = ((PyLongObject*)x)->ob_digit;
            switch (Py_SIZE(x)) {
                case  0: return (long) 0;
                case  1: __PYX_VERIFY_RETURN_INT(long, digit, digits[0])
                case 2:
                    if (8 * sizeof(long) > 1 * PyLong_SHIFT) {
                        if (8 * sizeof(unsigned long) > 2 * PyLong_SHIFT) {
                            __PYX_VERIFY_RETURN_INT(long, unsigned long, (((((unsigned long)digits[1]) << PyLong_SHIFT) | (unsigned long)digits[0])))
                        } else if (8 * sizeof(long) >= 2 * PyLong_SHIFT) {
                            return (long) (((((long)digits[1]) << PyLong_SHIFT) | (long)digits[0]));
                        }
                    }
                    break;
                case 3:
                    if (8 * sizeof(long) > 2 * PyLong_SHIFT) {
                        if (8 * sizeof(unsigned long) > 3 * PyLong_SHIFT) {
                            __PYX_VERIFY_RETURN_INT(long, unsigned long, (((((((unsigned long)digits[2]) << PyLong_SHIFT) | (unsigned long)digits[1]) << PyLong_SHIFT) | (unsigned long)digits[0])))
                        } else if (8 * sizeof(long) >= 3 * PyLong_SHIFT) {
                            return (long) (((((((long)digits[2]) << PyLong_SHIFT) | (long)digits[1]) << PyLong_SHIFT) | (long)digits[0]));
                        }
                    }
                    break;
                case 4:
                    if (8 * sizeof(long) > 3 * PyLong_SHIFT) {
                        if (8 * sizeof(unsigned long) > 4 * PyLong_SHIFT) {
                            __PYX_VERIFY_RETURN_INT(long, unsigned long, (((((((((unsigned long)digits[3]) << PyLong_SHIFT) | (unsigned long)digits[2]) << PyLong_SHIFT) | (unsigned long)digits[1]) << PyLong_SHIFT) | (unsigned long)digits[0])))
                        } else if (8 * sizeof(long) >= 4 * PyLong_SHIFT) {
                            return (long) (((((((((long)digits[3]) << PyLong_SHIFT) | (long)digits[2]) << PyLong_SHIFT) | (long)digits[1]) << PyLong_SHIFT) | (long)digits[0]));
                        }
                    }
                    break;
            }
#endif
#if CYTHON_COMPILING_IN_CPYTHON
            if (unlikely(Py_SIZE(x) < 0)) {
                goto raise_neg_overflow;
            }
#else
            {
                int result = PyObject_RichCompareBool(x, Py_False, Py_LT);
                if (unlikely(result < 0))
                    return (long) -1;
                if (unlikely(result == 1))
                    goto raise_neg_overflow;
            }
#endif
            if (sizeof(long) <= sizeof(unsigned long)) {
                __PYX_VERIFY_RETURN_INT_EXC(long, unsigned long, PyLong_AsUnsignedLong(x))
#ifdef HAVE_LONG_LONG
            } else if (sizeof(long) <= sizeof(unsigned PY_LONG_LONG)) {
                __PYX_VERIFY_RETURN_INT_EXC(long, unsigned PY_LONG_LONG, PyLong_AsUnsignedLongLong(x))
#endif
            }
        } else {
#if CYTHON_USE_PYLONG_INTERNALS
            const digit* digits = ((PyLongObject*)x)->ob_digit;
            switch (Py_SIZE(x)) {
                case  0: return (long) 0;
                case -1: __PYX_VERIFY_RETURN_INT(long, sdigit, (sdigit) (-(sdigit)digits[0]))
                case  1: __PYX_VERIFY_RETURN_INT(long,  digit, +digits[0])
                case -2:
                    if (8 * sizeof(long) - 1 > 1 * PyLong_SHIFT) {
                        if (8 * sizeof(unsigned long) > 2 * PyLong_SHIFT) {
                            __PYX_VERIFY_RETURN_INT(long, long, -(long) (((((unsigned long)digits[1]) << PyLong_SHIFT) | (unsigned long)digits[0])))
                        } else if (8 * sizeof(long) - 1 > 2 * PyLong_SHIFT) {
                            return (long) (((long)-1)*(((((long)digits[1]) << PyLong_SHIFT) | (long)digits[0])));
                        }
                    }
                    break;
                case 2:
                    if (8 * sizeof(long) > 1 * PyLong_SHIFT) {
                        if (8 * sizeof(unsigned long) > 2 * PyLong_SHIFT) {
                            __PYX_VERIFY_RETURN_INT(long, unsigned long, (((((unsigned long)digits[1]) << PyLong_SHIFT) | (unsigned long)digits[0])))
                        } else if (8 * sizeof(long) - 1 > 2 * PyLong_SHIFT) {
                            return (long) ((((((long)digits[1]) << PyLong_SHIFT) | (long)digits[0])));
                        }
                    }
                    break;
                case -3:
                    if (8 * sizeof(long) - 1 > 2 * PyLong_SHIFT) {
                        if (8 * sizeof(unsigned long) > 3 * PyLong_SHIFT) {
                            __PYX_VERIFY_RETURN_INT(long, long, -(long) (((((((unsigned long)digits[2]) << PyLong_SHIFT) | (unsigned long)digits[1]) << PyLong_SHIFT) | (unsigned long)digits[0])))
                        } else if (8 * sizeof(long) - 1 > 3 * PyLong_SHIFT) {
                            return (long) (((long)-1)*(((((((long)digits[2]) << PyLong_SHIFT) | (long)digits[1]) << PyLong_SHIFT) | (long)digits[0])));
                        }
                    }
                    break;
                case 3:
                    if (8 * sizeof(long) > 2 * PyLong_SHIFT) {
                        if (8 * sizeof(unsigned long) > 3 * PyLong_SHIFT) {
                            __PYX_VERIFY_RETURN_INT(long, unsigned long, (((((((unsigned long)digits[2]) << PyLong_SHIFT) | (unsigned long)digits[1]) << PyLong_SHIFT) | (unsigned long)digits[0])))
                        } else if (8 * sizeof(long) - 1 > 3 * PyLong_SHIFT) {
                            return (long) ((((((((long)digits[2]) << PyLong_SHIFT) | (long)digits[1]) << PyLong_SHIFT) | (long)digits[0])));
                        }
                    }
                    break;
                case -4:
                    if (8 * sizeof(long) - 1 > 3 * PyLong_SHIFT) {
                        if (8 * sizeof(unsigned long) > 4 * PyLong_SHIFT) {
                            __PYX_VERIFY_RETURN_INT(long, long, -(long) (((((((((unsigned long)digits[3]) << PyLong_SHIFT) | (unsigned long)digits[2]) << PyLong_SHIFT) | (unsigned long)digits[1]) << PyLong_SHIFT) | (unsigned long)digits[0])))
                        } else if (8 * sizeof(long) - 1 > 4 * PyLong_SHIFT) {
                            return (long) (((long)-1)*(((((((((long)digits[3]) << PyLong_SHIFT) | (long)digits[2]) << PyLong_SHIFT) | (long)digits[1]) << PyLong_SHIFT) | (long)digits[0])));
                        }
                    }
                    break;
                case 4:
                    if (8 * sizeof(long) > 3 * PyLong_SHIFT) {
                        if (8 * sizeof(unsigned long) > 4 * PyLong_SHIFT) {
                            __PYX_VERIFY_RETURN_INT(long, unsigned long, (((((((((unsigned long)digits[3]) << PyLong_SHIFT) | (unsigned long)digits[2]) << PyLong_SHIFT) | (unsigned long)digits[1]) << PyLong_SHIFT) | (unsigned long)digits[0])))
                        } else if (8 * sizeof(long) - 1 > 4 * PyLong_SHIFT) {
                            return (long) ((((((((((long)digits[3]) << PyLong_SHIFT) | (long)digits[2]) << PyLong_SHIFT) | (long)digits[1]) << PyLong_SHIFT) | (long)digits[0])));
                        }
                    }
                    break;
            }
#endif
            if (sizeof(long) <= sizeof(long)) {
                __PYX_VERIFY_RETURN_INT_EXC(long, long, PyLong_AsLong(x))
#ifdef HAVE_LONG_LONG
            } else if (sizeof(long) <= sizeof(PY_LONG_LONG)) {
                __PYX_VERIFY_RETURN_INT_EXC(long, PY_LONG_LONG, PyLong_AsLongLong(x))
#endif
            }
        }
        {
#if CYTHON_COMPILING_IN_PYPY && !defined(_PyLong_AsByteArray)
            PyErr_SetString(PyExc_RuntimeError,
                            "_PyLong_AsByteArray() not available in PyPy, cannot convert large numbers");
#else
            long val;
            PyObject *v = __Pyx_PyNumber_IntOrLong(x);
 #if PY_MAJOR_VERSION < 3
            if (likely(v) && !PyLong_Check(v)) {
                PyObject *tmp = v;
                v = PyNumber_Long(tmp);
                Py_DECREF(tmp);
            }
 #endif
            if (likely(v)) {
                int one = 1; int is_little = (int)*(unsigned char *)&one;
                unsigned char *bytes = (unsigned char *)&val;
                int ret = _PyLong_AsByteArray((PyLongObject *)v,
                                              bytes, sizeof(val),
                                              is_little, !is_unsigned);
                Py_DECREF(v);
                if (likely(!ret))
                    return val;
            }
#endif
            return (long) -1;
        }
    } else {
        long val;
        PyObject *tmp = __Pyx_PyNumber_IntOrLong(x);
        if (!tmp) return (long) -1;
        val = __Pyx_PyInt_As_long(tmp);
        Py_DECREF(tmp);
        return val;
    }
raise_overflow:
    PyErr_SetString(PyExc_OverflowError,
        "value too large to convert to long");
    return (long) -1;
raise_neg_overflow:
    PyErr_SetString(PyExc_OverflowError,
        "can't convert negative value to long");
    return (long) -1;
}

/* CIntFromPy */
static CYTHON_INLINE int __Pyx_PyInt_As_int(PyObject *x) {
    const int neg_one = (int) -1, const_zero = (int) 0;
    const int is_unsigned = neg_one > const_zero;
#if PY_MAJOR_VERSION < 3
    if (likely(PyInt_Check(x))) {
        if (sizeof(int) < sizeof(long)) {
            __PYX_VERIFY_RETURN_INT(int, long, PyInt_AS_LONG(x))
        } else {
            long val = PyInt_AS_LONG(x);
            if (is_unsigned && unlikely(val < 0)) {
                goto raise_neg_overflow;
            }
            return (int) val;
        }
    } else
#endif
    if (likely(PyLong_Check(x))) {
        if (is_unsigned) {
#if CYTHON_USE_PYLONG_INTERNALS
            const digit* digits = ((PyLongObject*)x)->ob_digit;
            switch (Py_SIZE(x)) {
                case  0: return (int) 0;
                case  1: __PYX_VERIFY_RETURN_INT(int, digit, digits[0])
                case 2:
                    if (8 * sizeof(int) > 1 * PyLong_SHIFT) {
                        if (8 * sizeof(unsigned long) > 2 * PyLong_SHIFT) {
                            __PYX_VERIFY_RETURN_INT(int, unsigned long, (((((unsigned long)digits[1]) << PyLong_SHIFT) | (unsigned long)digits[0])))
                        } else if (8 * sizeof(int) >= 2 * PyLong_SHIFT) {
                            return (int) (((((int)digits[1]) << PyLong_SHIFT) | (int)digits[0]));
                        }
                    }
                    break;
                case 3:
                    if (8 * sizeof(int) > 2 * PyLong_SHIFT) {
                        if (8 * sizeof(unsigned long) > 3 * PyLong_SHIFT) {
                            __PYX_VERIFY_RETURN_INT(int, unsigned long, (((((((unsigned long)digits[2]) << PyLong_SHIFT) | (unsigned long)digits[1]) << PyLong_SHIFT) | (unsigned long)digits[0])))
                        } else if (8 * sizeof(int) >= 3 * PyLong_SHIFT) {
                            return (int) (((((((int)digits[2]) << PyLong_SHIFT) | (int)digits[1]) << PyLong_SHIFT) | (int)digits[0]));
                        }
                    }
                    break;
                case 4:
                    if (8 * sizeof(int) > 3 * PyLong_SHIFT) {
                        if (8 * sizeof(unsigned long) > 4 * PyLong_SHIFT) {
                            __PYX_VERIFY_RETURN_INT(int, unsigned long, (((((((((unsigned long)digits[3]) << PyLong_SHIFT) | (unsigned long)digits[2]) << PyLong_SHIFT) | (unsigned long)digits[1]) << PyLong_SHIFT) | (unsigned long)digits[0])))
                        } else if (8 * sizeof(int) >= 4 * PyLong_SHIFT) {
                            return (int) (((((((((int)digits[3]) << PyLong_SHIFT) | (int)digits[2]) << PyLong_SHIFT) | (int)digits[1]) << PyLong_SHIFT) | (int)digits[0]));
                        }
                    }
                    break;
            }
#endif
#if CYTHON_COMPILING_IN_CPYTHON
            if (unlikely(Py_SIZE(x) < 0)) {
                goto raise_neg_overflow;
            }
#else
            {
                int result = PyObject_RichCompareBool(x, Py_False, Py_LT);
                if (unlikely(result < 0))
                    return (int) -1;
                if (unlikely(result == 1))
                    goto raise_neg_overflow;
            }
#endif
            if (sizeof(int) <= sizeof(unsigned long)) {
                __PYX_VERIFY_RETURN_INT_EXC(int, unsigned long, PyLong_AsUnsignedLong(x))
#ifdef HAVE_LONG_LONG
            } else if (sizeof(int) <= sizeof(unsigned PY_LONG_LONG)) {
                __PYX_VERIFY_RETURN_INT_EXC(int, unsigned PY_LONG_LONG, PyLong_AsUnsignedLongLong(x))
#endif
            }
        } else {
#if CYTHON_USE_PYLONG_INTERNALS
            const digit* digits = ((PyLongObject*)x)->ob_digit;
            switch (Py_SIZE(x)) {
                case  0: return (int) 0;
                case -1: __PYX_VERIFY_RETURN_INT(int, sdigit, (sdigit) (-(sdigit)digits[0]))
                case  1: __PYX_VERIFY_RETURN_INT(int,  digit, +digits[0])
                case -2:
                    if (8 * sizeof(int) - 1 > 1 * PyLong_SHIFT) {
                        if (8 * sizeof(unsigned long) > 2 * PyLong_SHIFT) {
                            __PYX_VERIFY_RETURN_INT(int, long, -(long) (((((unsigned long)digits[1]) << PyLong_SHIFT) | (unsigned long)digits[0])))
                        } else if (8 * sizeof(int) - 1 > 2 * PyLong_SHIFT) {
                            return (int) (((int)-1)*(((((int)digits[1]) << PyLong_SHIFT) | (int)digits[0])));
                        }
                    }
                    break;
                case 2:
                    if (8 * sizeof(int) > 1 * PyLong_SHIFT) {
                        if (8 * sizeof(unsigned long) > 2 * PyLong_SHIFT) {
                            __PYX_VERIFY_RETURN_INT(int, unsigned long, (((((unsigned long)digits[1]) << PyLong_SHIFT) | (unsigned long)digits[0])))
                        } else if (8 * sizeof(int) - 1 > 2 * PyLong_SHIFT) {
                            return (int) ((((((int)digits[1]) << PyLong_SHIFT) | (int)digits[0])));
                        }
                    }
                    break;
                case -3:
                    if (8 * sizeof(int) - 1 > 2 * PyLong_SHIFT) {
                        if (8 * sizeof(unsigned long) > 3 * PyLong_SHIFT) {
                            __PYX_VERIFY_RETURN_INT(int, long, -(long) (((((((unsigned long)digits[2]) << PyLong_SHIFT) | (unsigned long)digits[1]) << PyLong_SHIFT) | (unsigned long)digits[0])))
                        } else if (8 * sizeof(int) - 1 > 3 * PyLong_SHIFT) {
                            return (int) (((int)-1)*(((((((int)digits[2]) << PyLong_SHIFT) | (int)digits[1]) << PyLong_SHIFT) | (int)digits[0])));
                        }
                    }
                    break;
                case 3:
                    if (8 * sizeof(int) > 2 * PyLong_SHIFT) {
                        if (8 * sizeof(unsigned long) > 3 * PyLong_SHIFT) {
                            __PYX_VERIFY_RETURN_INT(int, unsigned long, (((((((unsigned long)digits[2]) << PyLong_SHIFT) | (unsigned long)digits[1]) << PyLong_SHIFT) | (unsigned long)digits[0])))
                        } else if (8 * sizeof(int) - 1 > 3 * PyLong_SHIFT) {
                            return (int) ((((((((int)digits[2]) << PyLong_SHIFT) | (int)digits[1]) << PyLong_SHIFT) | (int)digits[0])));
                        }
                    }
                    break;
                case -4:
                    if (8 * sizeof(int) - 1 > 3 * PyLong_SHIFT) {
                        if (8 * sizeof(unsigned long) > 4 * PyLong_SHIFT) {
                            __PYX_VERIFY_RETURN_INT(int, long, -(long) (((((((((unsigned long)digits[3]) << PyLong_SHIFT) | (unsigned long)digits[2]) << PyLong_SHIFT) | (unsigned long)digits[1]) << PyLong_SHIFT) | (unsigned long)digits[0])))
                        } else if (8 * sizeof(int) - 1 > 4 * PyLong_SHIFT) {
                            return (int) (((int)-1)*(((((((((int)digits[3]) << PyLong_SHIFT) | (int)digits[2]) << PyLong_SHIFT) | (int)digits[1]) << PyLong_SHIFT) | (int)digits[0])));
                        }
                    }
                    break;
                case 4:
                    if (8 * sizeof(int) > 3 * PyLong_SHIFT) {
                        if (8 * sizeof(unsigned long) > 4 * PyLong_SHIFT) {
                            __PYX_VERIFY_RETURN_INT(int, unsigned long, (((((((((unsigned long)digits[3]) << PyLong_SHIFT) | (unsigned long)digits[2]) << PyLong_SHIFT) | (unsigned long)digits[1]) << PyLong_SHIFT) | (unsigned long)digits[0])))
                        } else if (8 * sizeof(int) - 1 > 4 * PyLong_SHIFT) {
                            return (int) ((((((((((int)digits[3]) << PyLong_SHIFT) | (int)digits[2]) << PyLong_SHIFT) | (int)digits[1]) << PyLong_SHIFT) | (int)digits[0])));
                        }
                    }
                    break;
            }
#endif
            if (sizeof(int) <= sizeof(long)) {
                __PYX_VERIFY_RETURN_INT_EXC(int, long, PyLong_AsLong(x))
#ifdef HAVE_LONG_LONG
            } else if (sizeof(int) <= sizeof(PY_LONG_LONG)) {
                __PYX_VERIFY_RETURN_INT_EXC(int, PY_LONG_LONG, PyLong_AsLongLong(x))
#endif
            }
        }
        {
#if CYTHON_COMPILING_IN_PYPY && !defined(_PyLong_AsByteArray)
            PyErr_SetString(PyExc_RuntimeError,
                            "_PyLong_AsByteArray() not available in PyPy, cannot convert large numbers");
#else
            int val;
            PyObject *v = __Pyx_PyNumber_IntOrLong(x);
 #if PY_MAJOR_VERSION < 3
            if (likely(v) && !PyLong_Check(v)) {
                PyObject *tmp = v;
                v = PyNumber_Long(tmp);
                Py_DECREF(tmp);
            }
 #endif
            if (likely(v)) {
                int one = 1; int is_little = (int)*(unsigned char *)&one;
                unsigned char *bytes = (unsigned char *)&val;
                int ret = _PyLong_AsByteArray((PyLongObject *)v,
                                              bytes, sizeof(val),
                                              is_little, !is_unsigned);
                Py_DECREF(v);
                if (likely(!ret))
                    return val;
            }
#endif
            return (int) -1;
        }
    } else {
        int val;
        PyObject *tmp = __Pyx_PyNumber_IntOrLong(x);
        if (!tmp) return (int) -1;
        val = __Pyx_PyInt_As_int(tmp);
        Py_DECREF(tmp);
        return val;
    }
raise_overflow:
    PyErr_SetString(PyExc_OverflowError,
        "value too large to convert to int");
    return (int) -1;
raise_neg_overflow:
    PyErr_SetString(PyExc_OverflowError,
        "can't convert negative value to int");
    return (int) -1;
}

/* CheckBinaryVersion */
static int __Pyx_check_binary_version(void) {
    char ctversion[4], rtversion[4];
    PyOS_snprintf(ctversion, 4, "%d.%d", PY_MAJOR_VERSION, PY_MINOR_VERSION);
    PyOS_snprintf(rtversion, 4, "%s", Py_GetVersion());
    if (ctversion[0] != rtversion[0] || ctversion[2] != rtversion[2]) {
        char message[200];
        PyOS_snprintf(message, sizeof(message),
                      "compiletime version %s of module '%.100s' "
                      "does not match runtime version %s",
                      ctversion, __Pyx_MODULE_NAME, rtversion);
        return PyErr_WarnEx(NULL, message, 1);
    }
    return 0;
}

/* InitStrings */
static int __Pyx_InitStrings(__Pyx_StringTabEntry *t) {
    while (t->p) {
        #if PY_MAJOR_VERSION < 3
        if (t->is_unicode) {
            *t->p = PyUnicode_DecodeUTF8(t->s, t->n - 1, NULL);
        } else if (t->intern) {
            *t->p = PyString_InternFromString(t->s);
        } else {
            *t->p = PyString_FromStringAndSize(t->s, t->n - 1);
        }
        #else
        if (t->is_unicode | t->is_str) {
            if (t->intern) {
                *t->p = PyUnicode_InternFromString(t->s);
            } else if (t->encoding) {
                *t->p = PyUnicode_Decode(t->s, t->n - 1, t->encoding, NULL);
            } else {
                *t->p = PyUnicode_FromStringAndSize(t->s, t->n - 1);
            }
        } else {
            *t->p = PyBytes_FromStringAndSize(t->s, t->n - 1);
        }
        #endif
        if (!*t->p)
            return -1;
        ++t;
    }
    return 0;
}

static CYTHON_INLINE PyObject* __Pyx_PyUnicode_FromString(const char* c_str) {
    return __Pyx_PyUnicode_FromStringAndSize(c_str, (Py_ssize_t)strlen(c_str));
}
static CYTHON_INLINE char* __Pyx_PyObject_AsString(PyObject* o) {
    Py_ssize_t ignore;
    return __Pyx_PyObject_AsStringAndSize(o, &ignore);
}
static CYTHON_INLINE char* __Pyx_PyObject_AsStringAndSize(PyObject* o, Py_ssize_t *length) {
#if CYTHON_COMPILING_IN_CPYTHON && (__PYX_DEFAULT_STRING_ENCODING_IS_ASCII || __PYX_DEFAULT_STRING_ENCODING_IS_DEFAULT)
    if (
#if PY_MAJOR_VERSION < 3 && __PYX_DEFAULT_STRING_ENCODING_IS_ASCII
            __Pyx_sys_getdefaultencoding_not_ascii &&
#endif
            PyUnicode_Check(o)) {
#if PY_VERSION_HEX < 0x03030000
        char* defenc_c;
        PyObject* defenc = _PyUnicode_AsDefaultEncodedString(o, NULL);
        if (!defenc) return NULL;
        defenc_c = PyBytes_AS_STRING(defenc);
#if __PYX_DEFAULT_STRING_ENCODING_IS_ASCII
        {
            char* end = defenc_c + PyBytes_GET_SIZE(defenc);
            char* c;
            for (c = defenc_c; c < end; c++) {
                if ((unsigned char) (*c) >= 128) {
                    PyUnicode_AsASCIIString(o);
                    return NULL;
                }
            }
        }
#endif
        *length = PyBytes_GET_SIZE(defenc);
        return defenc_c;
#else
        if (__Pyx_PyUnicode_READY(o) == -1) return NULL;
#if __PYX_DEFAULT_STRING_ENCODING_IS_ASCII
        if (PyUnicode_IS_ASCII(o)) {
            *length = PyUnicode_GET_LENGTH(o);
            return PyUnicode_AsUTF8(o);
        } else {
            PyUnicode_AsASCIIString(o);
            return NULL;
        }
#else
        return PyUnicode_AsUTF8AndSize(o, length);
#endif
#endif
    } else
#endif
#if (!CYTHON_COMPILING_IN_PYPY) || (defined(PyByteArray_AS_STRING) && defined(PyByteArray_GET_SIZE))
    if (PyByteArray_Check(o)) {
        *length = PyByteArray_GET_SIZE(o);
        return PyByteArray_AS_STRING(o);
    } else
#endif
    {
        char* result;
        int r = PyBytes_AsStringAndSize(o, &result, length);
        if (unlikely(r < 0)) {
            return NULL;
        } else {
            return result;
        }
    }
}
static CYTHON_INLINE int __Pyx_PyObject_IsTrue(PyObject* x) {
   int is_true = x == Py_True;
   if (is_true | (x == Py_False) | (x == Py_None)) return is_true;
   else return PyObject_IsTrue(x);
}
static CYTHON_INLINE PyObject* __Pyx_PyNumber_IntOrLong(PyObject* x) {
#if CYTHON_USE_TYPE_SLOTS
  PyNumberMethods *m;
#endif
  const char *name = NULL;
  PyObject *res = NULL;
#if PY_MAJOR_VERSION < 3
  if (PyInt_Check(x) || PyLong_Check(x))
#else
  if (PyLong_Check(x))
#endif
    return __Pyx_NewRef(x);
#if CYTHON_USE_TYPE_SLOTS
  m = Py_TYPE(x)->tp_as_number;
  #if PY_MAJOR_VERSION < 3
  if (m && m->nb_int) {
    name = "int";
    res = PyNumber_Int(x);
  }
  else if (m && m->nb_long) {
    name = "long";
    res = PyNumber_Long(x);
  }
  #else
  if (m && m->nb_int) {
    name = "int";
    res = PyNumber_Long(x);
  }
  #endif
#else
  res = PyNumber_Int(x);
#endif
  if (res) {
#if PY_MAJOR_VERSION < 3
    if (!PyInt_Check(res) && !PyLong_Check(res)) {
#else
    if (!PyLong_Check(res)) {
#endif
      PyErr_Format(PyExc_TypeError,
                   "__%.4s__ returned non-%.4s (type %.200s)",
                   name, name, Py_TYPE(res)->tp_name);
      Py_DECREF(res);
      return NULL;
    }
  }
  else if (!PyErr_Occurred()) {
    PyErr_SetString(PyExc_TypeError,
                    "an integer is required");
  }
  return res;
}
static CYTHON_INLINE Py_ssize_t __Pyx_PyIndex_AsSsize_t(PyObject* b) {
  Py_ssize_t ival;
  PyObject *x;
#if PY_MAJOR_VERSION < 3
  if (likely(PyInt_CheckExact(b))) {
    if (sizeof(Py_ssize_t) >= sizeof(long))
        return PyInt_AS_LONG(b);
    else
        return PyInt_AsSsize_t(x);
  }
#endif
  if (likely(PyLong_CheckExact(b))) {
    #if CYTHON_USE_PYLONG_INTERNALS
    const digit* digits = ((PyLongObject*)b)->ob_digit;
    const Py_ssize_t size = Py_SIZE(b);
    if (likely(__Pyx_sst_abs(size) <= 1)) {
        ival = likely(size) ? digits[0] : 0;
        if (size == -1) ival = -ival;
        return ival;
    } else {
      switch (size) {
         case 2:
           if (8 * sizeof(Py_ssize_t) > 2 * PyLong_SHIFT) {
             return (Py_ssize_t) (((((size_t)digits[1]) << PyLong_SHIFT) | (size_t)digits[0]));
           }
           break;
         case -2:
           if (8 * sizeof(Py_ssize_t) > 2 * PyLong_SHIFT) {
             return -(Py_ssize_t) (((((size_t)digits[1]) << PyLong_SHIFT) | (size_t)digits[0]));
           }
           break;
         case 3:
           if (8 * sizeof(Py_ssize_t) > 3 * PyLong_SHIFT) {
             return (Py_ssize_t) (((((((size_t)digits[2]) << PyLong_SHIFT) | (size_t)digits[1]) << PyLong_SHIFT) | (size_t)digits[0]));
           }
           break;
         case -3:
           if (8 * sizeof(Py_ssize_t) > 3 * PyLong_SHIFT) {
             return -(Py_ssize_t) (((((((size_t)digits[2]) << PyLong_SHIFT) | (size_t)digits[1]) << PyLong_SHIFT) | (size_t)digits[0]));
           }
           break;
         case 4:
           if (8 * sizeof(Py_ssize_t) > 4 * PyLong_SHIFT) {
             return (Py_ssize_t) (((((((((size_t)digits[3]) << PyLong_SHIFT) | (size_t)digits[2]) << PyLong_SHIFT) | (size_t)digits[1]) << PyLong_SHIFT) | (size_t)digits[0]));
           }
           break;
         case -4:
           if (8 * sizeof(Py_ssize_t) > 4 * PyLong_SHIFT) {
             return -(Py_ssize_t) (((((((((size_t)digits[3]) << PyLong_SHIFT) | (size_t)digits[2]) << PyLong_SHIFT) | (size_t)digits[1]) << PyLong_SHIFT) | (size_t)digits[0]));
           }
           break;
      }
    }
    #endif
    return PyLong_AsSsize_t(b);
  }
  x = PyNumber_Index(b);
  if (!x) return -1;
  ival = PyInt_AsSsize_t(x);
  Py_DECREF(x);
  return ival;
}
static CYTHON_INLINE PyObject * __Pyx_PyInt_FromSize_t(size_t ival) {
    return PyInt_FromSize_t(ival);
}


#endif /* Py_PYTHON_H */
